// base-ui-components@3.1.2/dist/components/BaseChip/BaseChip.es.js
import { createId as n } from "../../utils/utils.es.js";
import "./BaseChip.vue_vue_type_style_index_0_scoped_a31dd8d0_lang.es.js";
import s from "../../_virtual/_plugin-vue2_normalizer.es.js";
const a = {
  name: "BaseChip",
  components: {
    BaseHoverBox: () => import("../BaseHoverBox/BaseHoverBox.es.js"),
    BaseIcon: () => import("../BaseIcon/BaseIcon.es.js")
  },
  model: {
    prop: "entry",
    event: "value-changed"
  },
  props: {
    entry: {
      type: String,
      required: !0
    },
    isLinked: {
      type: Boolean,
      default: !0
    },
    hoverBoxContent: {
      type: Object,
      default: () => ({})
    },
    chipActive: {
      type: Boolean,
      default: !1
    },
    isRemovable: {
      type: Boolean,
      default: !0
    },
    textStyling: {
      type: Object,
      default: () => ({})
    },
    editable: {
      type: Boolean,
      default: !1
    },
    assistiveText: {
      type: String,
      default: ""
    }
  },
  data() {
    return {
      entryInt: "",
      showInfoBox: !1
    };
  },
  computed: {
    hoverBoxEnabled() {
      return this.isLinked && !!Object.keys(this.hoverBoxContent).length;
    },
    internalId() {
      return n();
    }
  },
  watch: {
    entry: {
      handler(t) {
        t !== this.entryInt && (this.entryInt = t);
      },
      immediate: !0
    }
  },
  methods: {
    updateText(t) {
      const e = t.target.innerText;
      this.editable && e !== this.entryInt && (this.entryInt = e, this.$emit("value-changed", this.entryInt));
    },
    clickAction(t) {
      this.editable || (t.stopPropagation(), this.$emit("clicked"), this.isLinked && (this.$emit("hoverbox-active", !0), this.$refs.hoverBox.setPosition(t), this.showInfoBox = !this.showInfoBox));
    },
    moveBox(t) {
      this.hoverBoxEnabled && this.showInfoBox && this.$refs.hoverBox.setPosition(t);
    },
    hideBox() {
      this.hoverBoxEnabled && (this.$emit("hoverbox-active", !1), this.showInfoBox = !1);
    },
    onMouseDown(t) {
      this.$emit("mouse-down", t);
    },
    removeClicked() {
      this.$emit("remove-entry", this.entryInt);
    }
  }
};
var r = function() {
  var e = this, o = e._self._c;
  return o("div", { class: [
    "base-chip",
    { "base-chip__removable": e.isRemovable },
    { "base-chip__linked": e.isLinked },
    { "base-chip__active": e.chipActive }
  ] }, [o("div", { ref: "chipText", staticClass: "base-chip__text", style: e.textStyling, attrs: { contenteditable: e.editable ? "true" : !1, "aria-labelledby": e.assistiveText ? `${e.internalId}_aria-label` : !1, enterkeyhint: "search" }, on: { blur: e.updateText, keydown: function(i) {
    return !i.type.indexOf("key") && e._k(i.keyCode, "enter", 13, i.key, "Enter") ? null : (i.preventDefault(), e.updateText.apply(null, arguments));
  }, click: function(i) {
    return i.stopPropagation(), e.clickAction.apply(null, arguments);
  }, mousedown: e.onMouseDown, mousemove: e.moveBox, mouseleave: e.hideBox } }, [e._v(" " + e._s(e.entryInt) + " ")]), e.assistiveText ? o("span", { staticClass: "hide", attrs: { id: `${e.internalId}_aria-label` } }, [e._v(" " + e._s(e.assistiveText) + " ")]) : e._e(), e.isRemovable ? o("BaseIcon", { staticClass: "base-chip__icon", attrs: { name: "remove" }, nativeOn: { click: function(i) {
    return i.stopPropagation(), e.removeClicked.apply(null, arguments);
  } } }) : e._e(), o("base-hover-box", e._b({ ref: "hoverBox", class: { "base-chip__hover-box__hidden": !e.hoverBoxEnabled || !e.showInfoBox } }, "base-hover-box", e.hoverBoxContent, !1))], 1);
}, l = [], h = /* @__PURE__ */ s(
  a,
  r,
  l,
  !1,
  null,
  "a31dd8d0",
  null,
  null
);
const u = h.exports;
export {
  u as default
};
