// base-ui-components@3.1.2/dist/components/BaseBox/BaseBox.es.js
import "./BaseBox.vue_vue_type_style_index_0_scoped_f7db7b0e_lang.es.js";
import a from "../../_virtual/_plugin-vue2_normalizer.es.js";
const l = {
  name: "BaseBox",
  props: {
    renderElementAs: {
      type: String,
      default: "div"
    },
    boxSize: {
      type: Object,
      default: () => ({ width: "200px" })
    },
    boxRatio: {
      type: String,
      default: "100"
    },
    boxHover: {
      type: Boolean,
      default: !0
    },
    boxShadowSize: {
      type: String,
      default: "small",
      validator: (t) => ["small", "large"].includes(t)
    },
    disabled: {
      type: Boolean,
      default: !1
    },
    additionalAttributes: {
      type: Object,
      default: () => ({})
    }
  },
  methods: {
    clicked(t) {
      this.$emit("clicked", t);
    }
  }
};
var s = function() {
  var e = this, o = e._self._c;
  return o(e.renderElementAs, e._b({ tag: "compontent", class: [
    "base-box",
    "base-box-" + e.$props.boxRatio,
    "base-box-shadow-" + e.boxShadowSize,
    { "base-box-hover": e.boxHover && !e.disabled }
  ], style: e.boxSize, attrs: { disabled: e.disabled, type: e.renderElementAs === "button" ? "button" : "" }, on: { click: e.clicked } }, "compontent", e.additionalAttributes, !1), [e._t("default")], 2);
}, n = [], d = /* @__PURE__ */ a(
  l,
  s,
  n,
  !1,
  null,
  "f7db7b0e",
  null,
  null
);
const b = d.exports;
export {
  b as default
};
