// base-ui-components/dist/components/BaseFormGroups/BaseFormGroups.es.js
import l from "../BaseForm/BaseForm.es.js";
import { createId as f } from "../../utils/utils.es.js";
import "./BaseFormGroups.vue_vue_type_style_index_0_scoped_b3830d06_lang.es.js";
import s from "../../_virtual/_plugin-vue2_normalizer.es.js";
const u = {
  name: "BaseFormGroups",
  components: {
    BaseForm: l
  },
  props: {
    formFieldJson: {
      type: Object,
      required: !0
    },
    valueList: {
      type: Object,
      default: () => ({})
    },
    language: {
      type: String,
      default: "en"
    },
    availableLocales: {
      type: Array,
      default: () => []
    },
    formStyle: {
      type: Object,
      default: () => ({})
    },
    fieldsWithTabs: {
      type: Array,
      default: () => []
    },
    fieldIsLoading: {
      type: String,
      default: ""
    },
    dropDownLists: {
      type: Object,
      default: () => ({})
    },
    showErrorIcon: {
      type: Boolean,
      default: !1
    },
    clearable: {
      type: Boolean,
      default: !1
    },
    fieldProps: {
      type: Object,
      default: () => ({}),
      validator: (r) => Object.keys(r).length === 0 || Object.values(r).every((e) => Object.keys(e))
    }
  },
  computed: {
    formProps() {
      const r = {
        ...this.$props
      };
      return delete r.formFieldJson, r;
    },
    formFieldsGrouped() {
      const r = [], e = {};
      return Object.entries(this.formFieldJson).forEach(([o, n]) => {
        if (n["x-attrs"] && n["x-attrs"].form_group) {
          const i = n["x-attrs"].form_group - 1;
          r[i] ? r[i][o] = n : r[i] = {
            [o]: n
          };
        } else
          e[o] = n;
      }), Object.keys(e).length && r.push(e), r;
    },
    groupsId() {
      return f();
    }
  }
};
var a = function() {
  var e = this, o = e._self._c;
  return o("div", { staticClass: "base-form-groups" }, e._l(e.formFieldsGrouped, function(n, i) {
    return o("BaseForm", e._g(e._b({ key: `${e.groupsId}-${i}`, staticClass: "base-form-groups__group", attrs: { "form-field-json": n }, scopedSlots: e._u([{ key: "label-addition", fn: function({ fieldName: t }) {
      return [e._t("label-addition", null, { fieldName: t })];
    } }, { key: "pre-input-field", fn: function({ fieldName: t }) {
      return [e._t("pre-input-field", null, { fieldName: t })];
    } }, { key: "input-field-addition-before", fn: function({ fieldName: t }) {
      return [e._t("input-field-addition-before", null, { fieldName: t })];
    } }, { key: "input-field-inline-before", fn: function({ fieldName: t }) {
      return [e._t("input-field-inline-before", null, { fieldName: t })];
    } }, { key: "input-field-addition-after", fn: function({ fieldName: t }) {
      return [e._t("input-field-addition-after", null, { fieldName: t })];
    } }, { key: "post-input-field", fn: function({ fieldName: t }) {
      return [e._t("post-input-field", null, { fieldName: t })];
    } }, { key: "error-icon", fn: function() {
      return [e._t("error-icon")];
    }, proxy: !0 }, { key: "remove-icon", fn: function() {
      return [e._t("remove-icon")];
    }, proxy: !0 }, { key: "below-input", fn: function({ fieldName: t }) {
      return [e._t("below-input", null, { fieldName: t })];
    } }], null, !0) }, "BaseForm", e.formProps, !1), e.$listeners));
  }), 1);
}, d = [], p = /* @__PURE__ */ s(
  u,
  a,
  d,
  !1,
  null,
  "b3830d06",
  null,
  null
);
const b = p.exports;
export {
  b as default
};
