// base-ui-components/dist/utils/utils.es.js
import b from "@babel/runtime/helpers/typeof";
import g from "@babel/runtime/helpers/slicedToArray";
var d = globalThis, L = function() {
  return Math.random().toString(36).substr(2, 9);
}, N = function(t, r) {
  return t.split(".").reduce(function(n, e) {
    return n[e];
  }, r);
}, p = function(t) {
  var r = t.replace(/,? [0-9-]+/g, "");
  if (r.includes(",")) {
    var n = r.split(", ");
    return [n[0], n.splice(1).join()];
  }
  var e = r.split(" ");
  return [e.pop(), r];
}, T = function(t) {
  var r = arguments.length > 1 && arguments[1] !== void 0 ? arguments[1] : "", n = arguments.length > 2 && arguments[2] !== void 0 ? arguments[2] : !1, e = arguments.length > 3 && arguments[3] !== void 0 ? arguments[3] : null;
  return t.sort(function(s, i) {
    var a = s, o = i;
    if (r && (a = a[r], o = o[r]), e && (a = e(a), o = e(o)), n) {
      var c = "", v = "", h = p(a), f = g(h, 2);
      a = f[0], c = f[1];
      var l = p(o), m = g(l, 2);
      o = m[0], v = m[1], a.toLowerCase() === o.toLowerCase() && (a = c, o = v);
    }
    return a.toLowerCase() > o.toLowerCase() ? 1 : -1;
  });
}, x = function u(t) {
  var r = !1;
  if (t && b(t) === "object")
    if (t.length >= 0)
      t.forEach(function(e) {
        r = u(e) || r;
      });
    else {
      var n = Object.keys(t);
      n.forEach(function(e) {
        r = u(t[e]) || r;
      });
    }
  else
    r = t === 0 || !!t || r;
  return r;
}, y = function(t, r) {
  var n = 0;
  return function() {
    for (var e = arguments.length, s = new Array(e), i = 0; i < e; i++)
      s[i] = arguments[i];
    clearTimeout(n), n = setTimeout(function() {
      return r.apply(d, s);
    }, t);
  };
};
export {
  L as createId,
  y as debounce,
  N as extractNestedPropertyValue,
  p as getNameSortValue,
  x as hasData,
  T as sort
};
