// base-ui-components@3.1.2/dist/components/BaseMapLocations/BaseMapLocations.es.js
import n from "../BaseTextList/BaseTextList.es.js";
import "./BaseMapLocations.vue_vue_type_style_index_0_scoped_a5ebf4f2_lang.es.js";
import s from "../../_virtual/_plugin-vue2_normalizer.es.js";
const l = {
  name: "BaseMapLocations",
  components: {
    BaseMap: () => import("../BaseMap/BaseMap.es.js").then((t) => t.default || t),
    BaseTextList: n
  },
  props: {
    additionalMapStyles: {
      type: Object,
      default: () => ({ height: "368px" })
    },
    attribution: {
      type: String,
      default: "Source: <a href=https://openstreetmap.org/>OpenStreetMap contributors</a>"
    },
    attributionPosition: {
      type: String,
      default: "bottomright",
      validate(t) {
        return ["topright", "bottomleft", "bottomright"].includes(t);
      }
    },
    clusterSizes: {
      type: Object,
      default: () => ({ medium: 5, large: 20, xlarge: 100 }),
      validator: (t) => t.medium && t.large && t.xlarge
    },
    copyright: {
      type: String,
      default: "<a href=http://creativecommons.org/licenses/by-sa/3.0/>CC BY-SA 3.0</a>"
    },
    icon: {
      type: String,
      default: '<svg viewBox="0 0 70.866 70.866" xmlns="http://www.w3.org/2000/svg"><path d="m35.433 0a22.731 22.731 0 0 0-22.731 22.82 24.125 24.125 0 0 0 1.872 9.1814l19.611 38.063a1.3718 1.3718 0 0 0 2.496 0l19.611-38.063a22.249 22.249 0 0 0 1.872-9.1814 22.731 22.731 0 0 0-22.731-22.82zm0 32.858a10.216 10.216 0 1 1 10.216-10.216 10.241 10.241 0 0 1-10.216 10.216z" fill="#010101"/></svg>'
    },
    iconSize: {
      type: Number,
      default: 32
    },
    label: {
      type: String,
      default: "Addresses"
    },
    locations: {
      type: Array,
      default: () => [],
      validator: (t) => t.filter((e) => e.latLng || e.coordinates).length === t.length
    },
    markerPopups: {
      type: Boolean,
      default: !0
    },
    maxZoom: {
      type: Number,
      default: 18
    },
    options: {
      type: Object,
      default: () => ({})
    },
    url: {
      type: String,
      default: ""
    },
    tileLayerService: {
      type: String,
      default: "WMTS",
      validator(t) {
        return ["WMTS", "TMS", "WMS"].includes(t);
      }
    },
    zoom: {
      type: Number,
      default: 16
    }
  },
  data() {
    return {
      highlightedMarker: null,
      highlightedLocation: null,
      centeredMarker: null,
      initMap: !1
    };
  },
  computed: {
    observer() {
      return new IntersectionObserver((t, e) => {
        t.forEach((r) => {
          !r.isIntersecting || (e.unobserve(r.target), this.initMap = !0);
        });
      });
    },
    locationsFiltered() {
      return Array.from(new Set(this.locations.map(JSON.stringify))).map(JSON.parse);
    }
  },
  beforeDestroy() {
    this.observer.disconnect();
  },
  mounted() {
    this.observer.observe(this.$el);
  },
  methods: {
    highlightLocation(t) {
      this.highlightedLocation = t;
    },
    resetMarker() {
      this.highlightedMarker = null, this.centeredMarker = null;
    }
  }
};
var u = function() {
  var e = this, r = e._self._c;
  return r("div", { staticClass: "base-map-locations" }, [r("div", { staticClass: "base-map-locations__map", style: e.additionalMapStyles }, [e.initMap ? r("BaseMap", { style: e.additionalMapStyles, attrs: { attribution: e.attribution, "attribution-position": e.attributionPosition, "cluster-sizes": e.clusterSizes, copyright: e.copyright, "center-marker": e.centeredMarker, icon: e.icon, "icon-size": e.iconSize, "highlight-marker": e.highlightedMarker, marker: e.locationsFiltered, "marker-popups": e.markerPopups, "max-zoom": e.maxZoom, options: e.options, "tile-layer-service": e.tileLayerService, url: e.url, zoom: e.zoom }, on: { highlighted: e.highlightLocation } }) : e._e()], 1), e.label ? r("h2", { staticClass: "base-map-locations__label" }, [e._v(" " + e._s(e.label) + " ")]) : e._e(), r("div", { staticClass: "base-map-locations__list" }, [e._l(e.locationsFiltered, function(o, a) {
    return [o.latLng || o.coordinates ? r("BaseTextList", { key: a, class: [
      "base-map-locations__list__group",
      { "base-map-locations__list__group--highlight": a === e.highlightedLocation }
    ], attrs: { data: [o] }, nativeOn: { click: function(i) {
      e.centeredMarker = a;
    }, focusin: function(i) {
      e.highlightedMarker = a;
    }, mouseenter: function(i) {
      e.highlightedMarker = a;
    }, focusout: function(i) {
      return e.resetMarker.apply(null, arguments);
    }, mouseleave: function(i) {
      return e.resetMarker.apply(null, arguments);
    } } }) : e._e()];
  })], 2)]);
}, c = [], p = /* @__PURE__ */ s(
  l,
  u,
  c,
  !1,
  null,
  "a5ebf4f2",
  null,
  null
);
const d = p.exports;
export {
  d as default
};
