// base-ui-components@3.1.2/dist/components/BaseToggle/BaseToggle.es.js
import { createId as i } from "../../utils/utils.es.js";
import r from "../BaseIcon/BaseIcon.es.js";
import "./BaseToggle.vue_vue_type_style_index_0_scoped_73525b61_lang.es.js";
import h from "../../_virtual/_plugin-vue2_normalizer.es.js";
const f = {
  name: "BaseToggle",
  components: {
    BaseIcon: r
  },
  model: {
    prop: "checked",
    event: "clicked"
  },
  props: {
    name: {
      type: String,
      default: "baseToggle"
    },
    label: {
      type: String,
      default: "baseToggle"
    },
    disabled: {
      type: Boolean,
      default: !1
    },
    checked: {
      type: Boolean,
      default: !1
    },
    hideLabel: {
      type: Boolean,
      default: !1
    },
    bindSlotToState: {
      type: Boolean,
      default: !1
    },
    id: {
      type: String,
      default: ""
    }
  },
  data() {
    return {
      checkedInt: !1,
      animate: !1
    };
  },
  computed: {
    idInt() {
      return this.id || i();
    }
  },
  watch: {
    checked: {
      handler(a) {
        a !== this.checkedInt && (this.checkedInt = a);
      },
      immediate: !0
    },
    checkedInt(a) {
      this.$emit("clicked", a);
    }
  }
};
var m = function() {
  var e = this, t = e._self._c;
  return t("div", { class: [
    "base-toggle",
    { "base-toggle--checked": e.checkedInt },
    { "base-toggle--disabled": e.disabled }
  ], on: { focusin: function(c) {
    e.animate = !0;
  }, mouseover: function(c) {
    e.animate = !0;
  }, focusout: function(c) {
    e.animate = !1;
  }, mouseleave: function(c) {
    e.animate = !1;
  } } }, [t("label", { staticClass: "base-toggle__container", attrs: { for: `toggle-input-${e.idInt}` } }, [t("input", { directives: [{ name: "model", rawName: "v-model", value: e.checkedInt, expression: "checkedInt" }], staticClass: "base-toggle__input", attrs: { id: `toggle-input-${e.idInt}`, name: e.name, disabled: e.disabled, "aria-checked": e.checkedInt, "aria-disabled": e.disabled, value: "", type: "checkbox" }, domProps: { checked: e.checkedInt, checked: Array.isArray(e.checkedInt) ? e._i(e.checkedInt, "") > -1 : e.checkedInt }, on: { change: function(c) {
    var n = e.checkedInt, l = c.target, o = !!l.checked;
    if (Array.isArray(n)) {
      var d = "", s = e._i(n, d);
      l.checked ? s < 0 && (e.checkedInt = n.concat([d])) : s > -1 && (e.checkedInt = n.slice(0, s).concat(n.slice(s + 1)));
    } else
      e.checkedInt = o;
  } } }), t("div", { staticClass: "base-switch" }, [t("span", { class: ["base-switch__control", { "base-switch__control--animate": e.animate }] }, [e.checkedInt ? t("BaseIcon", { attrs: { title: e.hideLabel ? e.label : "", name: "check-mark" } }) : e._e(), e.checkedInt ? e._e() : t("BaseIcon", { attrs: { title: e.hideLabel ? e.label : "", name: "remove" } })], 1)]), e.hideLabel ? e._e() : t("span", { staticClass: "base-toggle__label" }, [e._v(" " + e._s(e.label) + " ")])]), !!e.$slots.default && e.checked && e.bindSlotToState || !!e.$slots.default && !e.bindSlotToState ? t("span", { staticClass: "base-toggle__subtext" }, [e._t("default")], 2) : e._e()]);
}, u = [], b = /* @__PURE__ */ h(
  f,
  m,
  u,
  !1,
  null,
  "73525b61",
  null,
  null
);
const I = b.exports;
export {
  I as default
};
