// base-ui-components@3.1.2/dist/components/BaseImageBox/BaseImageBox.es.js
import b from "../BaseBox/BaseBox.es.js";
import g from "../BaseIcon/BaseIcon.es.js";
import m from "../BaseImage/BaseImage.es.js";
import "./BaseImageBox.vue_vue_type_style_index_0_scoped_d0070e23_lang.es.js";
import x from "../../_virtual/_plugin-vue2_normalizer.es.js";
const h = {
  name: "BaseImageBox",
  components: {
    BaseCheckmark: () => import("../BaseCheckmark/BaseCheckmark.es.js").then((a) => a.default || a),
    BaseIcon: g,
    BaseBox: b,
    BaseImage: m
  },
  props: {
    title: {
      type: String,
      default: "No title"
    },
    subtext: {
      type: String,
      default: ""
    },
    showTitle: {
      type: Boolean,
      default: !0
    },
    showTitleOnHover: {
      type: Boolean,
      default: !1
    },
    imageUrl: {
      type: String,
      default: null
    },
    description: {
      type: String,
      default: null
    },
    additional: {
      type: String,
      default: null
    },
    icon: {
      type: String,
      default: ""
    },
    iconSize: {
      type: String,
      default: "xxlarge",
      validator: (a) => ["small", "medium", "large", "xlarge", "xxlarge"].includes(a)
    },
    playIcon: {
      type: Boolean,
      default: !1
    },
    selectable: {
      type: Boolean,
      default: !1
    },
    draggable: {
      type: Boolean,
      default: !1
    },
    selected: {
      type: Boolean,
      default: !1
    },
    boxSize: {
      type: Object,
      default: () => ({ width: "auto", height: "auto" })
    },
    boxText: {
      type: Array,
      default: () => []
    },
    lazyload: {
      type: Boolean,
      default: !1
    },
    imageFirst: {
      type: Boolean,
      default: !1
    },
    centerHeader: {
      type: Boolean,
      default: !1
    },
    renderElementAs: {
      type: String,
      default: "div"
    },
    linkTo: {
      type: [String, Object],
      default: "",
      validator: (a) => typeof a == "string" || a instanceof Object && Object.keys(a).includes("path")
    }
  },
  data() {
    return {
      selectedInt: !1,
      boxTextStyle: {},
      displayImage: !0,
      resizeTimeout: null
    };
  },
  computed: {
    imageShadowClass() {
      return this.imageFirst ? "base-image-box-img-third" : "base-image-box-img-half";
    }
  },
  watch: {
    selectedInt(a) {
      a !== this.selected && this.$emit("select-triggered", a);
    },
    selected(a) {
      this.selectedInt !== a && (this.selectedInt = a);
    },
    boxText() {
      this.calcTextHeight();
    }
  },
  mounted() {
    this.selectedInt = this.selected, window && window.addEventListener("resize", this.resizeTriggered), this.$nextTick(() => {
      this.calcTextHeight();
    });
  },
  destroyed() {
    window.removeEventListener("resize", this.resizeTriggered);
  },
  methods: {
    boxSelect() {
      this.selectable ? this.selectedInt = !this.selectedInt : this.$emit("clicked");
    },
    resizeTriggered() {
      this.resizeTimeout && (clearTimeout(this.resizeTimeout), this.resizeTimeout = null), this.resizeTimeout = setTimeout(() => {
        this.calcTextHeight();
      }, 500);
    },
    calcTextHeight() {
      if (!this.imageUrl && this.boxText.length && this.$refs.baseBox && this.$refs.boxText) {
        let a = 0;
        if (this.$refs.headerBox) {
          const r = this.$refs.headerBox, c = r.clientHeight, d = r.offsetTop;
          a = c + 2 * d;
        }
        const e = this.$refs.boxText, t = this.$refs.boxTextInner[0] || this.$refs.boxTextInner, s = this.$refs.baseBox.$el.clientHeight, i = window.getComputedStyle(t, null).getPropertyValue("line-height").replace("px", ""), n = window.getComputedStyle(e, null).getPropertyValue("margin-bottom").replace("px", "");
        let o = s - a - n;
        this.description && (o -= i);
        const l = Math.floor(o / i);
        this.boxTextStyle = {
          height: `${i * l}px`,
          "-webkit-line-clamp": l
        };
      }
    }
  }
};
var f = function() {
  var e = this, t = e._self._c;
  return t("BaseBox", { ref: "baseBox", attrs: { "box-size": e.boxSize, "render-element-as": e.renderElementAs, "additional-attributes": e.linkTo ? { to: e.linkTo } : {}, "box-ratio": "100" }, on: { clicked: e.boxSelect } }, [t("div", { class: [
    "base-image-box-content-wrapper",
    { "base-image-box-hover": (e.playIcon || e.showTitleOnHover) && !e.selectable },
    { "base-image-box-hover-show-title": e.showTitleOnHover && !e.selectable },
    { "base-image-box-selectable": e.selectable },
    { "base-image-box-draggable": e.draggable },
    { "base-image-box-selected": e.selectable && e.selectedInt }
  ] }, [t("div", { staticClass: "base-image-box-content" }, [e.showTitle ? t("div", { ref: "headerBox", class: [
    "base-image-box-header",
    { "base-image-box-header-centered": e.centerHeader }
  ] }, [t("div", { class: [
    "base-image-box-title",
    { "base-image-box-header-2-lines": !e.subtext }
  ], attrs: { title: e.title } }, [e._v(" " + e._s(e.title) + " ")]), e.subtext ? t("div", { staticClass: "base-image-box-subtext", attrs: { title: e.subtext } }, [e._v(" " + e._s(e.subtext) + " ")]) : e._e()]) : e._e(), t("div", { class: [
    "base-image-box-body",
    { "base-image-box-inner-shadow-bottom-top": e.selectable && e.imageUrl && !e.showTitle },
    {
      "base-image-box-inner-shadow-bottom": !e.icon && (!e.showTitleOnHover || e.showTitleOnHover && e.additional)
    },
    { "base-image-box-order-first": e.imageFirst },
    e.imageShadowClass
  ] }, [e.imageUrl && e.displayImage ? t("div", { class: ["base-image-box-img-wrapper"] }, [t("BaseImage", { ref: "image", class: [
    "base-image-box-image",
    { "base-image-box__image-second": !e.imageFirst },
    { "base-image-box-no-title": !e.showTitle }
  ], attrs: { alt: e.title, lazyload: e.lazyload, src: e.imageUrl }, on: { error: function(s) {
    e.displayImage = !1;
  } } })], 1) : e._e(), e.icon ? t("BaseIcon", { class: ["base-image-box-icon", "base-image-box-icon--" + e.iconSize], attrs: { name: e.icon } }) : e._e(), e.playIcon ? t("BaseIcon", { class: ["base-image-box-icon-play", "base-image-box-icon--" + e.iconSize], attrs: { name: "play" } }) : e._e(), !e.imageUrl || !e.displayImage ? t("div", { ref: "boxText", staticClass: "base-image-box__text-wrapper" }, [e._t("text", function() {
    return [t("div", { ref: "boxTextInner", staticClass: "base-image-box-text", style: e.boxTextStyle }, e._l(e.boxText, function(s, i) {
      return t("div", { key: i }, [e._v(" " + e._s(s) + " ")]);
    }), 0)];
  }, { text: e.boxText })], 2) : e._e(), t("div", { class: [
    "base-image-box-footer",
    { "base-image-box-footer-invert": e.icon }
  ] }, [e.$slots["footer-left"] ? t("div", { staticClass: "base-image-box-footer-left" }, [e._t("footer-left")], 2) : e._e(), t("div", { staticClass: "base-image-box-footer-body" }, [e.showTitleOnHover ? t("div", { staticClass: "base-image-box-footer-title bold", attrs: { title: e.title } }, [e._v(" " + e._s(e.title) + " ")]) : e._e(), e.description ? t("div", { class: [
    "base-image-box-footer-text",
    { bold: !e.additional }
  ], attrs: { title: e.description } }, [e._v(" " + e._s(e.description) + " ")]) : e._e(), e.additional ? t("div", { staticClass: "base-image-box-footer-text bold", attrs: { title: e.additional } }, [e._v(" " + e._s(e.additional) + " ")]) : e._e()]), e.$slots["footer-right"] || e.playIcon ? t("div", { staticClass: "base-image-box-footer-right" }, [e.playIcon ? t("BaseIcon", { staticClass: "base-image-box-icon-play-small", attrs: { name: "play" } }) : e._e(), e._t("footer-right")], 2) : e._e()])], 1)]), t("div", { staticClass: "base-image-box-features" }, [t("transition", { attrs: { name: "slide-fade" } }, [e.selectable ? t("BaseCheckmark", { staticClass: "base-image-box-checkbox", attrs: { checked: e.selectedInt, label: e.title, "mark-style": "checkbox", "check-box-size": "large" }, on: { clicked: e.boxSelect } }) : e._e()], 1)], 1)])]);
}, p = [], u = /* @__PURE__ */ x(
  h,
  f,
  p,
  !1,
  null,
  "d0070e23",
  null,
  null
);
const w = u.exports;
export {
  w as default
};
