// base-ui-components@3.1.2/dist/components/BaseDropDownList/BaseDropDownList.es.js
import c from "../../mixins/i18n.es.js";
import "./BaseDropDownList.vue_vue_type_style_index_0_scoped_2a2f4c3a_lang.es.js";
import f from "../../_virtual/_plugin-vue2_normalizer.es.js";
const u = {
  name: "BaseDropDownList",
  mixins: [
    c
  ],
  props: {
    dropDownOptions: {
      type: Array,
      default: () => []
    },
    identifierPropertyName: {
      type: String,
      default: "id"
    },
    labelPropertyName: {
      type: String,
      default: "value"
    },
    activeOption: {
      type: [Object, String],
      default: () => ({})
    },
    selectedOption: {
      type: [Object, String],
      default: () => ({})
    },
    listId: {
      type: String,
      default: ""
    },
    listBodyStyle: {
      type: Object,
      default: () => ({})
    },
    activeStyled: {
      type: Boolean,
      default: !0
    },
    selectStyled: {
      type: Boolean,
      default: !1
    },
    displayAsDropDown: {
      type: Boolean,
      default: !0
    },
    dropDownNoOptionsInfo: {
      type: String,
      default: ""
    },
    language: {
      type: String,
      default: ""
    },
    hasSubOptions: {
      type: Boolean,
      default: !1
    },
    useCustomOptionActiveBackgroundColor: {
      type: Boolean,
      default: !1
    }
  },
  data() {
    return {
      inputElement: null
    };
  },
  computed: {
    valueIsString() {
      return this.dropDownOptions && this.dropDownOptions.length && typeof this.getLangLabel(this.dropDownOptions[0][this.labelPropertyName], !0) == "string";
    },
    activeOptionIndex() {
      return this.hasSubOptions && (!this.activeOption || this.activeOption[this.identifierPropertyName] !== void 0) ? this.dropDownOptions.findIndex((e) => e[this.identifierPropertyName] === this.activeOption[this.identifierPropertyName]) : this.dropDownOptions.indexOf(this.activeOption);
    },
    noOptionsSlotHasData() {
      return !!this.$slots["no-options"];
    }
  },
  mounted() {
    const e = document.getElementsByTagName("input");
    if (e && e.length) {
      const t = Array.from(e);
      this.inputElement = t.find((i) => i.getAttribute("list") === this.listId), this.inputElement && this.inputElement.addEventListener("keydown", this.navigateOptions);
    }
  },
  destroyed() {
    this.inputElement && this.inputElement.removeEventListener("keydown", this.navigateOptions);
  },
  methods: {
    selected(e) {
      this.$emit("update:selected-option", e);
    },
    optionHasData(e) {
      return typeof e == "string" ? !!e : e && e.length ? !!e.length : !!(e && Object.keys(e).length);
    },
    navigateOptions(e, { activeOptionHeight: t = 0, activeOptionTop: i = 0 } = {}) {
      const { key: o } = e;
      if (["ArrowDown", "ArrowUp"].includes(o) && (!this.hasSubOptions || t) && this.$refs.option && this.$refs.option[this.activeOptionIndex]) {
        const a = this.$refs.option[this.activeOptionIndex], n = t || a.clientHeight, s = i || a.offsetTop;
        if (this.$parent.$refs.dropDownContainer)
          this.$parent.navigateOptions(e, {
            activeOptionHeight: n,
            activeOptionTop: s
          });
        else {
          const r = this.$refs.dropDownContainer, l = r.scrollTop, p = r.clientHeight, d = s + n < l || s > l + p;
          !this.hasSubOptions && !this.activeOptionIndex ? r.scrollTop = 0 : !this.hasSubOptions && this.activeOptionIndex === this.dropDownOptions.length - 1 ? r.scrollTo({
            top: s + n
          }) : e.key === "ArrowDown" ? d ? r.scrollTo({
            top: s
          }) : s + n > p + l && r.scrollTo({
            top: l + n
          }) : e.key === "ArrowUp" && (d ? r.scrollTop = s + n - p : l > s && (r.scrollTop -= n));
        }
      }
    }
  }
};
var h = function() {
  var t = this, i = t._self._c;
  return i("div", { ref: "dropDownContainer", staticClass: "base-drop-down-list__container", class: [
    "base-drop-down-list__container",
    { "base-drop-down-list__container-drop-down-style": t.displayAsDropDown }
  ] }, [t._t("before-list"), i("ul", { staticClass: "base-drop-down-list", style: t.listBodyStyle, attrs: { id: t.listId, "aria-activedescendant": t.activeOption ? t.activeOption[t.identifierPropertyName] : !1, role: "listbox" } }, [t._l(t.dropDownOptions, function(o, a) {
    return [t.optionHasData(o[t.labelPropertyName]) ? i("li", { key: a, ref: "option", refInFor: !0, class: [
      "base-drop-down-list__option",
      { "base-drop-down-list__option__hover": t.activeStyled },
      {
        "base-drop-down-list__option__selected": t.selectStyled && o === t.selectedOption
      },
      {
        "base-drop-down-list__option__active": t.activeStyled && a === t.activeOptionIndex
      },
      {
        "base-drop-down-list__option__active-custom": t.useCustomOptionActiveBackgroundColor && a === t.activeOptionIndex
      }
    ], attrs: { id: o[t.identifierPropertyName], value: t.valueIsString ? t.getLangLabel(o[t.labelPropertyName], !0) : o[t.identifierPropertyName], "aria-selected": t.selectStyled && o[t.identifierPropertyName] === t.selectedOption, role: "option", tabindex: "0" }, on: { keydown: function(n) {
      return !n.type.indexOf("key") && t._k(n.keyCode, "enter", 13, n.key, "Enter") ? null : t.selected(o);
    }, click: function(n) {
      return t.selected(o);
    } } }, [t._t("option", function() {
      return [t._v(" " + t._s(t.getLangLabel(o[t.labelPropertyName], !0)) + " ")];
    }, { option: o })], 2) : t._e()];
  }), !t.dropDownOptions.length && (t.noOptionsSlotHasData || t.dropDownNoOptionsInfo) ? i("div", { class: [
    "base-drop-down-list__option",
    "base-drop-down-list__no-options"
  ] }, [t.dropDownOptions.length ? t._e() : t._t("no-options", function() {
    return [t._v(" " + t._s(t.dropDownNoOptionsInfo) + " ")];
  })], 2) : t._e()], 2), t._t("after-list")], 2);
}, y = [], O = /* @__PURE__ */ f(
  u,
  h,
  y,
  !1,
  null,
  "2a2f4c3a",
  null,
  null
);
const v = O.exports;
export {
  v as default
};
