// base-ui-components/dist/components/BaseExpandBox/BaseExpandBox.es.js
import o from "../BaseBox/BaseBox.es.js";
import s from "../BaseButton/BaseButton.es.js";
import "./BaseExpandBox.vue_vue_type_style_index_0_scoped_ca63cad2_lang.es.js";
import a from "../../_virtual/_plugin-vue2_normalizer.es.js";
const i = {
  name: "BaseExpandBox",
  components: {
    BaseBox: o,
    BaseButton: s
  },
  props: {
    expand: {
      type: Boolean,
      default: !1
    },
    padding: {
      type: String,
      default: "small",
      validator(n) {
        return ["large", "small"].includes(n);
      }
    },
    showMoreText: {
      type: String,
      default: "Show more"
    },
    showLessText: {
      type: String,
      default: "Show less"
    },
    autoHeight: {
      type: Boolean,
      default: !1
    },
    maxCollapsedHeight: {
      type: Number,
      default: 300
    },
    showButtonHeight: {
      type: Number,
      default: 54
    }
  },
  data() {
    return {
      expandInt: !1,
      contentWidth: null,
      initialized: !1,
      showButton: !1
    };
  },
  computed: {
    maxCollapsedHeightInt() {
      return this.showButton ? this.maxCollapsedHeight - this.showButtonHeight : this.maxCollapsedHeight;
    },
    style() {
      return { "--max-collapsed-height": `${this.maxCollapsedHeightInt}px` };
    }
  },
  mounted() {
    this.init();
  },
  methods: {
    init() {
      this.expand && (this.showButton = this.contentInnerHeight() > this.contentHeight(), this.expandInt = !0), this.initialized = !0, this.boxResize().observe(this.$el);
    },
    boxResize() {
      return new ResizeObserver((n) => {
        const t = n[0].contentRect.width;
        this.contentWidth !== t && (this.expandInt || (this.showButton = this.contentInnerHeight() > this.contentHeight()), this.expand && this.expandInt && this.contentWidth !== null && (this.expandInt = !1)), this.contentWidth = t, this.expandInt || this.$emit("box-height", this.$el.offsetHeight);
      });
    },
    contentHeight() {
      return this.$el.querySelector(".base-expand-box-content").offsetHeight;
    },
    contentInnerHeight() {
      return this.$el.querySelector(".base-expand-box-content-inner > div").offsetHeight;
    },
    clicked() {
      this.expandInt = !this.expandInt, this.$emit("update:expand", this.expandInt);
    }
  }
};
var d = function() {
  var t = this, e = t._self._c;
  return e("base-box", { class: [
    "base-expand-box",
    "base-expand-box-padding-" + t.padding,
    { "base-expand-box-auto-height": t.autoHeight },
    { "base-expand-box-open": t.expandInt }
  ], style: t.style, attrs: { "box-ratio": "0", "box-size": {}, "box-hover": !1 } }, [e("div", { class: [
    "base-expand-box-content",
    { "base-expand-box-content-fade-out": !t.initialized || !t.expandInt && t.showButton }
  ] }, [e("div", { staticClass: "base-expand-box-content-inner" }, [e("div", [t._t("default")], 2)])]), t.showButton ? e("base-button", { class: [
    "base-expand-box-button",
    { "base-button-icon-rotate-180": t.expandInt }
  ], attrs: { text: t.expandInt ? t.showLessText : t.showMoreText, "has-background-color": !1, "align-text": "left", icon: "drop-down", "icon-position": "right" }, on: { clicked: t.clicked } }) : t._e(), t.$slots.footer ? e("div", { staticClass: "base-expand-box-footer" }, [e("div", { staticClass: "base-expand-box-footer-inner" }, [t._t("footer")], 2)]) : t._e()], 1);
}, h = [], l = /* @__PURE__ */ a(
  i,
  d,
  h,
  !1,
  null,
  "ca63cad2",
  null,
  null
);
const u = l.exports;
export {
  u as default
};
