// base-ui-components@3.1.2/dist/components/BaseExpandList/BaseExpandListRow.es.js
import l from "../BaseIcon/BaseIcon.es.js";
import d from "../../mixins/i18n.es.js";
import "./BaseExpandListRow.vue_vue_type_style_index_0_scoped_4f7de2da_lang.es.js";
import "./BaseExpandListRow.vue_vue_type_style_index_1_lang.es.js";
import "./BaseExpandListRow.vue_vue_type_style_index_2_lang.es.js";
import o from "../../_virtual/_plugin-vue2_normalizer.es.js";
const r = {
  name: "BaseExpandListRow",
  components: {
    BaseToggle: () => import("../BaseToggle/BaseToggle.es.js").then((t) => t.default || t),
    BaseButton: () => import("../BaseButton/BaseButton.es.js").then((t) => t.default || t),
    BaseExpandListRow: () => Promise.resolve().then(() => BaseExpandListRow$1).then((t) => t.default || t),
    BaseIcon: l
  },
  mixins: [d],
  props: {
    data: {
      type: Object,
      default: () => {
      }
    },
    edit: {
      type: Boolean,
      default: !1
    },
    multiple: {
      type: Boolean,
      default: !1
    },
    renderAs: {
      type: String,
      default: "div"
    },
    editHideText: {
      type: String,
      default: "Hide"
    },
    editShowText: {
      type: String,
      default: "Show"
    },
    controlType: {
      type: String,
      default: "button",
      validator: (t) => ["button", "toggle"].includes(t)
    },
    parentIndex: {
      type: [Number, String],
      default: 0
    },
    disabled: {
      type: Boolean,
      default: !1
    }
  },
  data() {
    return {
      expanded: !1,
      movable: !1,
      useSupportiveText: !0,
      dataInt: null,
      fadeOutLeft: !1,
      fadeOutRight: !0
    };
  },
  computed: {
    itemVisible: {
      set(t) {
        this.$set(this.dataInt, "hidden", !t);
      },
      get() {
        return !this.dataInt.hidden;
      }
    }
  },
  watch: {
    movable(t) {
      const e = this.$el.querySelector(".base-expand-item__handle");
      t && e.focus(), document.activeElement === e && this.supportiveText(this.movable ? "activated" : "activate"), this.useSupportiveText = !0;
    },
    dataInt: {
      handler(t) {
        !!t && JSON.stringify(t) !== JSON.stringify(this.data) && this.$emit("update:data", t);
      },
      deep: !0,
      immediate: !0
    },
    data: {
      handler(t) {
        JSON.stringify(t) !== JSON.stringify(this.dataInt) && (this.dataInt = JSON.parse(JSON.stringify(t)));
      },
      deep: !0,
      immediate: !0
    }
  },
  mounted() {
    if (this.$refs.listItemScrollable) {
      const t = this.$refs.listItemScrollable;
      t.addEventListener("scroll", () => this.scrollHandler(t));
    }
  },
  destroyed() {
    if (this.$refs.listItemScrollable) {
      const t = this.$refs.listItemScrollable;
      t.removeEventListener("scroll", () => this.scrollHandler(t));
    }
  },
  methods: {
    cancelMovable(t) {
      this.movable && (t.stopPropagation(), t.stopImmediatePropagation(), t.preventDefault(), t.cancelBubble = !1, this.movable = !1);
    },
    setVisibility(t) {
      this.itemVisible = t;
    },
    moveItem(t = "up ") {
      this.movable && (this.useSupportiveText = !1, this.movable = !1, this.$emit("sorted", {
        direction: t
      }));
    },
    expand() {
      const t = this.$vnode.key.toString().split("-");
      if (this.expanded) {
        this.expanded = !1, this.emitExpandedState(t.length > 1 ? t.slice(-1) : "");
        return;
      }
      if (!this.multiple) {
        const e = this.$parent.$refs.baseExpandListRow;
        e && e.forEach((i) => {
          i.expanded = !1;
        });
      }
      this.expanded = !0, this.emitExpandedState(t);
    },
    emitExpandedState(t) {
      this.$emit("expanded-state", t);
    },
    maxHeight(t) {
      const e = this.$el.querySelector(".base-expand-item").offsetHeight, i = e + e * this.dataInt.data.length;
      t.setAttribute("style", `max-height: ${i}px`);
    },
    closeRows(t) {
      t && t.forEach((e) => {
        e.expanded = !1;
      });
    },
    enter(t) {
      this.maxHeight(t);
    },
    afterEnter(t) {
      t.style.removeProperty("max-height");
    },
    leave(t) {
      const e = this.$el.offsetHeight;
      t.setAttribute("style", `max-height: ${e}px`);
    },
    beforeLeave() {
      this.multiple || this.closeRows(this.$refs.baseExpandListRow);
    },
    supportiveText(t = "activate") {
      !this.useSupportiveText || this.$emit("supportive", t);
    },
    scrollHandler(t) {
      this.fadeOutLeft = !!t.scrollLeft, this.fadeOutRight = t.scrollWidth - t.scrollLeft !== t.offsetWidth;
    }
  }
};
var p = function() {
  var e = this, i = e._self._c;
  return i(e.renderAs, { tag: "component", class: [{
    "base-expand": !e.edit,
    expanded: e.expanded
  }] }, [!e.edit && e.dataInt.label ? [i("button", { staticClass: "base-expand__head base-expand-item", attrs: { id: "base-expand-control-" + e._uid, "aria-expanded": e.expanded ? "true" : "false", "aria-controls": "base-expand-region-" + e._uid, type: "button" }, on: { click: e.expand } }, [i("span", { staticClass: "base-expand-item__col base-expand-item__label base-text-fade-out" }, [i("span", { staticClass: "base-expand__head__label" }, [e._v(e._s(e.dataInt.label))]), i("span", { staticClass: "base-expand__head__additional" }, [e._v(" (" + e._s(e.dataInt.count !== void 0 ? e.dataInt.count : e.dataInt.data.length) + ") ")])]), i("base-icon", { staticClass: "base-expand-item__col base-expand__head__icon", attrs: { name: "drop-down" } })], 1), i("transition", { attrs: { name: "expand" }, on: { enter: e.enter, "after-enter": e.afterEnter, leave: e.leave, "before-leave": e.beforeLeave } }, [i("ul", { directives: [{ name: "show", rawName: "v-show", value: e.expanded, expression: "expanded" }], staticClass: "base-expand__body", attrs: { id: "base-expand-region-" + e._uid, "aria-labelledby": "base-expand-control-" + e._uid, "aria-hidden": e.expanded ? "false" : "true" } }, e._l(e.dataInt.data, function(a, s) {
    return i("base-expand-list-row", { key: `${e.parentIndex}-${s}`, ref: "baseExpandListRow", refInFor: !0, attrs: { data: a, multiple: e.multiple, "render-as": "li" }, on: { "expanded-state": e.emitExpandedState }, scopedSlots: e._u([{ key: "content", fn: function(n) {
      return [e._t("content", null, { data: n.data })];
    } }], null, !0) });
  }), 1)])] : e._e(), !e.edit && e.dataInt.value ? [i("div", { class: [
    "base-expand-item",
    "base-expand-item--intend",
    {
      "base-text-fade-out-left": e.fadeOutLeft,
      "base-text-fade-out": e.fadeOutRight
    }
  ] }, [i("span", { ref: "listItemScrollable", staticClass: "base-expand-item__col base-expand-item__label" }, [e._t("content", function() {
    return [e._v(" " + e._s(e.dataInt.value) + " ")];
  }, { data: e.dataInt })], 2)])] : e._e(), e.edit ? [i("div", { class: ["base-expand-item", { "base-expand-item--movable": e.movable }] }, [i("span", { ref: "baseExpandItemHandle", staticClass: "base-expand-item__col base-expand-item__handle", attrs: { title: e.dataInt.label, tabindex: "0" }, on: { keydown: [function(a) {
    return !a.type.indexOf("key") && e._k(a.keyCode, "down", 40, a.key, ["Down", "ArrowDown"]) ? null : (a.preventDefault(), a.stopPropagation(), e.moveItem("down"));
  }, function(a) {
    return !a.type.indexOf("key") && e._k(a.keyCode, "up", 38, a.key, ["Up", "ArrowUp"]) ? null : (a.preventDefault(), a.stopPropagation(), e.moveItem("up"));
  }, function(a) {
    if (!a.type.indexOf("key") && e._k(a.keyCode, "enter", 13, a.key, "Enter"))
      return null;
    a.preventDefault(), a.stopPropagation(), e.movable = !e.movable;
  }], focus: function(a) {
    return e.supportiveText("activate");
  }, keyup: function(a) {
    return !a.type.indexOf("key") && e._k(a.keyCode, "esc", 27, a.key, ["Esc", "Escape"]) ? null : e.cancelMovable.apply(null, arguments);
  }, blur: e.cancelMovable } }, [i("base-icon", { attrs: { name: "drag-lines" } })], 1), i("div", { staticClass: "base-expand-item__col base-expand-item__label base-text-fade-out" }, [e.itemVisible ? e._e() : i("div", { staticClass: "base-expand-item__overlay" }), i("span", { class: [
    "base-expand__head__label",
    { "base-expand__head__label--disabled": !e.itemVisible }
  ] }, [e._v(" " + e._s(e.dataInt.label) + " ")]), i("span", { staticClass: "base-expand__head__additional" }, [e._v(" (" + e._s(e.dataInt.count !== void 0 ? e.dataInt.count : e.dataInt.data.length) + ") ")])]), i("div", { staticClass: "base-expand-item__col base-expand-item__controls" }, [e.controlType === "button" ? i("base-button", { attrs: { icon: e.itemVisible ? "eye" : "eye-hide", text: "", "has-background-color": !1, disabled: e.disabled, "aria-label": `${e.itemVisible ? e.editHideText : e.editShowText} ${e.dataInt.label}`, "icon-size": "large", "icon-position": "left" }, on: { clicked: function(a) {
    return e.setVisibility(!e.itemVisible);
  } } }) : e.controlType === "toggle" ? i("BaseToggle", { staticClass: "base-expand-item__toggle", attrs: { disabled: e.disabled, "hide-label": !0, label: `${e.editShowText} ${e.dataInt.label}` }, model: { value: e.itemVisible, callback: function(a) {
    e.itemVisible = a;
  }, expression: "itemVisible" } }) : e._e()], 1)])] : e._e()], 2);
}, u = [], c = /* @__PURE__ */ o(
  r,
  p,
  u,
  !1,
  null,
  "4f7de2da",
  null,
  null
);
const v = c.exports;
export {
  v as default
};
