// base-ui-components@3.1.2/dist/components/BaseMultilineTextInput/BaseMultilineTextInput.es.js
import n from "../BaseInput/BaseInput.es.js";
import r from "../BaseIcon/BaseIcon.es.js";
import { createId as s } from "../../utils/utils.es.js";
import "./BaseMultilineTextInput.vue_vue_type_style_index_0_scoped_521dc393_lang.es.js";
import l from "../../_virtual/_plugin-vue2_normalizer.es.js";
const o = {
  name: "BaseMultilineTextInput",
  components: {
    BaseInput: n,
    BaseIcon: r,
    BaseSwitchButton: () => import("../BaseSwitchButton/BaseSwitchButton.es.js").then((e) => e.default || e)
  },
  model: {
    prop: "input",
    event: "input"
  },
  props: {
    input: {
      type: [Object, String],
      default: ""
    },
    label: {
      type: String,
      default: ""
    },
    showLabel: {
      type: Boolean,
      default: !0
    },
    placeholder: {
      type: String,
      default: ""
    },
    tabs: {
      type: Array,
      default: () => ["default"]
    },
    tabLabels: {
      type: Array,
      default: () => []
    },
    activeTab: {
      type: String,
      default() {
        return this.tabs[0] || "default";
      }
    },
    tabsLegend: {
      type: String,
      default: "Radiogroup"
    },
    id: {
      type: [String, Number],
      default: ""
    },
    language: {
      type: String,
      default: ""
    },
    required: {
      type: Boolean,
      default: !1
    },
    invalid: {
      type: Boolean,
      default: !1
    },
    disabled: {
      type: Boolean,
      default: !1
    },
    errorMessage: {
      type: String,
      default: ""
    },
    showErrorIcon: {
      type: Boolean,
      default: !0
    },
    clearable: {
      type: Boolean,
      default: !1
    }
  },
  data() {
    return {
      activeTabInt: this.activeTab || "default",
      fieldContent: {},
      isActive: !1
    };
  },
  computed: {
    idInt() {
      return this.id || s();
    },
    switchTabs() {
      return this.tabs.map((e, t) => ({ value: e, label: this.tabLabels[t] || e }));
    },
    inputListeners() {
      return {
        ...this.$listeners,
        input: () => {
          this.$emit("input", this.emitFieldContent());
        }
      };
    },
    showFadeOut() {
      return this.$refs && this.$refs.textarea ? !this.isActive && this.$refs.textarea.scrollHeight > this.$refs.textareaWrapper.clientHeight : !this.isActive;
    }
  },
  watch: {
    input: {
      handler(e) {
        (typeof e == "object" && JSON.stringify(e) !== JSON.stringify(this.fieldContent) || typeof e == "string" && e !== this.fieldContent[this.activeTabInt]) && this.setFieldContent(e);
      },
      immediate: !0,
      deep: !0
    },
    activeTab(e) {
      this.activeTabInt = e;
    }
  },
  methods: {
    setFieldContent(e) {
      if (this.tabs.length < 2) {
        const t = this.activeTabInt || "default";
        this.$set(this.fieldContent, t, typeof e == "string" ? e : e[t]);
      } else
        this.tabs.forEach((t) => this.$set(
          this.fieldContent,
          t,
          e[t]
        ));
    },
    emitFieldContent() {
      return typeof this.input == "object" ? this.fieldContent : this.fieldContent[this.activeTabInt];
    },
    hasText(e) {
      return !!this.fieldContent[e];
    }
  }
};
var u = function() {
  var t = this, a = t._self._c;
  return a("BaseInput", t._g({ staticClass: "base-multiline-text-input", attrs: { id: t.idInt, "is-active": t.isActive, label: t.label, "show-label": t.showLabel, placeholder: t.placeholder, required: t.required, invalid: t.invalid, disabled: t.disabled, "show-error-icon": t.showErrorIcon, "error-message": t.errorMessage, clearable: t.clearable, "use-fade-out": !1 }, on: { "update:isActive": function(i) {
    t.isActive = i;
  }, "update:is-active": function(i) {
    t.isActive = i;
  } }, scopedSlots: t._u([{ key: "label-addition", fn: function() {
    return [a("div", { staticClass: "base-multiline-text-input__additions" }, [t._t("label-addition"), t.tabs && t.tabs[0] !== "default" ? a("BaseSwitchButton", { staticClass: "base-multiline-text-input__tabs", attrs: { options: t.switchTabs, label: t.tabsLegend, "active-tab": t.activeTab }, scopedSlots: t._u([{ key: "right-of-text", fn: function(i) {
      return [t.hasText(i.value) ? a("BaseIcon", { staticClass: "base-multiline-text-input__text-icon", attrs: { name: "text" } }) : t._e()];
    } }], null, !1, 3542795371), model: { value: t.activeTabInt, callback: function(i) {
      t.activeTabInt = i;
    }, expression: "activeTabInt" } }) : t._e()], 2)];
  }, proxy: !0 }, { key: "input", fn: function() {
    return [a("div", { ref: "textareaWrapper", class: [
      "base-multiline-text-input__textarea-wrapper",
      { "base-multiline-text-input__textarea-wrapper__fade-out": t.showFadeOut }
    ] }, [a("textarea", t._g({ directives: [{ name: "model", rawName: "v-model", value: t.fieldContent[t.activeTabInt], expression: "fieldContent[activeTabInt]" }], ref: "textarea", staticClass: "base-multiline-text-input__textarea", attrs: { id: t.idInt, required: t.required, "aria-required": t.required.toString(), "aria-describedby": t.idInt, "aria-invalid": t.invalid.toString(), disabled: t.disabled, "aria-disabled": t.disabled.toString(), placeholder: t.placeholder }, domProps: { value: t.fieldContent[t.activeTabInt] }, on: { keydown: function(i) {
      if (!i.type.indexOf("key") && t._k(i.keyCode, "tab", 9, i.key, "Tab"))
        return null;
      t.isActive = !1;
    }, input: function(i) {
      i.target.composing || t.$set(t.fieldContent, t.activeTabInt, i.target.value);
    } } }, t.inputListeners))])];
  }, proxy: !0 }, { key: "input-field-addition-before", fn: function() {
    return [t._t("input-field-addition-before")];
  }, proxy: !0 }, { key: "input-field-addition-after", fn: function() {
    return [t._t("input-field-addition-after")];
  }, proxy: !0 }, { key: "post-input-field", fn: function() {
    return [t._t("post-input-field")];
  }, proxy: !0 }, { key: "error-icon", fn: function() {
    return [t._t("error-icon")];
  }, proxy: !0 }, { key: "remove-icon", fn: function() {
    return [t._t("remove-icon")];
  }, proxy: !0 }], null, !0), model: { value: t.fieldContent[t.activeTabInt], callback: function(i) {
    t.$set(t.fieldContent, t.activeTabInt, i);
  }, expression: "fieldContent[activeTabInt]" } }, t.$listeners));
}, d = [], f = /* @__PURE__ */ l(
  o,
  u,
  d,
  !1,
  null,
  "521dc393",
  null,
  null
);
const y = f.exports;
export {
  y as default
};
