// base-ui-components/dist/components/BaseChipsInputField/BaseChipsInputField.es.js
import r from "vuedraggable";
import { createId as a, sort as o } from "../../utils/utils.es.js";
import d from "../BaseInput/BaseInput.es.js";
import p from "../../mixins/i18n.es.js";
import u from "../../mixins/navigateList.es.js";
import "./BaseChipsInputField.vue_vue_type_style_index_0_scoped_477631eb_lang.es.js";
import c from "../../_virtual/_plugin-vue2_normalizer.es.js";
const h = {
  name: "BaseChipsInputField",
  components: {
    BaseInput: d,
    BaseChip: () => import("../BaseChip/BaseChip.es.js").then((t) => t.default || t),
    Draggable: r
  },
  mixins: [
    p,
    u
  ],
  model: {
    prop: "input",
    event: "input"
  },
  props: {
    id: {
      type: String,
      default: ""
    },
    selectedList: {
      type: Array,
      default: () => []
    },
    input: {
      type: String,
      default: ""
    },
    label: {
      type: String,
      required: !0
    },
    showLabel: {
      type: Boolean,
      default: !0
    },
    placeholder: {
      type: String,
      default: null
    },
    allowUnknownEntries: {
      type: Boolean,
      default: !0
    },
    allowMultipleEntries: {
      type: Boolean,
      default: !0
    },
    addSelectedEntryDirectly: {
      type: Boolean,
      default: !1
    },
    sortable: {
      type: Boolean,
      default: !1
    },
    draggable: {
      type: Boolean,
      default: !1
    },
    alwaysLinked: {
      type: Boolean,
      default: !1
    },
    hoverboxContent: {
      type: Object,
      default: () => ({})
    },
    isLoading: {
      type: Boolean,
      default: !1
    },
    sortText: {
      type: String,
      default: "Sort A \u2013 Z"
    },
    sortName: {
      type: Boolean,
      default: !1
    },
    language: {
      type: String,
      default: ""
    },
    showInputBorder: {
      type: Boolean,
      default: !0
    },
    useFormFieldStyling: {
      type: Boolean,
      default: !0
    },
    dropDownListId: {
      type: String,
      default: ""
    },
    linkedListOption: {
      type: String,
      default: null
    },
    identifierPropertyName: {
      type: String,
      default: ""
    },
    labelPropertyName: {
      type: String,
      default: "label"
    },
    isStringArray: {
      type: Boolean,
      default: !1
    },
    displayChipsInline: {
      type: Boolean,
      default: !0
    },
    required: {
      type: Boolean,
      default: !1
    },
    invalid: {
      type: Boolean,
      default: !1
    },
    disabled: {
      type: Boolean,
      default: !1
    },
    errorMessage: {
      type: String,
      default: ""
    },
    showErrorIcon: {
      type: Boolean,
      default: !0
    },
    clearable: {
      type: Boolean,
      default: !1
    },
    loadable: {
      type: Boolean,
      default: !1
    },
    isActive: {
      type: Boolean,
      default: !1
    },
    inputClass: {
      type: String,
      default: ""
    },
    setFocusOnActive: {
      type: Boolean,
      default: !0
    },
    chipsEditable: {
      type: Boolean,
      default: !1
    },
    assistiveText: {
      type: Object,
      default: () => ({})
    },
    chipsRemovable: {
      type: Boolean,
      default: !0
    }
  },
  data() {
    return {
      selectedListInt: [],
      isActiveInt: !1,
      timeout: null,
      fired: !1,
      drag: !1,
      chipActiveForRemove: -1,
      inputInt: ""
    };
  },
  watch: {
    selectedList: {
      handler(t) {
        if (t) {
          const e = t.map((n) => ({
            ...this.isStringArray ? { [this.labelPropertyName]: n } : n,
            idInt: n[this.identifierPropertyName] || this.getIdInt(n)
          }));
          JSON.stringify(e) !== JSON.stringify(this.selectedListInt) && (this.selectedListInt = e);
        } else
          this.selectedListInt = [];
      },
      deep: !0,
      immediate: !0
    },
    isActiveInt(t) {
      t !== this.isActive && this.$emit("update:is-active", t);
    },
    isActive(t) {
      t !== this.isActiveInt && (this.isActiveInt = t);
    },
    input: {
      handler(t) {
        t !== this.inputInt && (this.inputInt = t);
      },
      immediate: !0
    },
    inputInt: {
      handler(t) {
        t !== this.input && this.$emit("input", t);
      },
      immediate: !0
    }
  },
  methods: {
    checkKeyEvent(t) {
      const { key: e } = t;
      if ((e === "Tab" && !(this.clearable && !!this.inputInt) || e === "Enter" && !this.allowMultipleEntries) && (this.isActiveInt = !1), this.chipsRemovable && (e === "Backspace" || e === "Delete"))
        e === "Backspace" && !this.fired && !this.inputInt && this.chipActiveForRemove < 0 ? this.chipActiveForRemove = this.selectedListInt.length - 1 : this.chipActiveForRemove >= 0 && !this.fired && !this.inputInt && (this.removeEntry(
          this.selectedListInt[this.chipActiveForRemove],
          this.chipActiveForRemove
        ), this.chipActiveForRemove = -1), this.fired = !0, this.timeout && (clearTimeout(this.timeout), this.timeout = null), this.timeout = setTimeout(() => {
          this.fired = !1;
        }, 200);
      else if (!this.inputInt && (t.key === "ArrowRight" || t.key === "ArrowLeft")) {
        const n = t.key === "ArrowRight", i = this.navigate(
          this.selectedListInt,
          n,
          this.chipActiveForRemove,
          !0
        );
        this.chipActiveForRemove = this.selectedListInt.indexOf(i);
      } else
        this.chipActiveForRemove = -1;
    },
    removeEntry(t, e) {
      this.selectedListInt.splice(e, 1), this.updateParentList(this.selectedListInt), this.$emit("removed", t);
      const n = this.$refs.baseInput.$el.getElementsByTagName("input"), i = Array.from(n).find((s) => s.id === this.id);
      i && i.focus();
    },
    addOption() {
      if (this.inputInt && this.allowUnknownEntries && this.addSelectedEntryDirectly) {
        const t = this.selectedListInt.find((e) => e[this.labelPropertyName] === this.inputInt);
        if (t)
          this.$emit("duplicate", t);
        else {
          const e = this.allowMultipleEntries ? this.selectedListInt.length : 0, n = {
            idInt: this.getIdInt(),
            [this.labelPropertyName]: this.inputInt
          };
          this.$set(this.selectedListInt, e, n), this.updateParentList(this.selectedListInt);
        }
        this.inputInt = "";
      }
    },
    modifyListEntry(t, e) {
      typeof this.selectedListInt[e][this.labelPropertyName] == "object" && this.selectedListInt[e][this.labelPropertyName][this.language] !== void 0 ? this.selectedListInt[e] = {
        [this.labelPropertyName]: {
          [this.language]: t
        }
      } : typeof this.selectedListInt[e][this.labelPropertyName] == "string" && (this.selectedListInt[e] = {
        [this.labelPropertyName]: t
      }), this.updateParentList(this.selectedListInt);
    },
    updateParentList(t) {
      let e = JSON.parse(JSON.stringify(t));
      this.isStringArray ? e = e.map((n) => n[this.labelPropertyName]) : t.length && (e = e.map((n) => (this.$delete(n, "idInt"), n))), this.$emit("update:selected-list", e);
    },
    getIdInt(t = null) {
      if (t) {
        const e = this.selectedListInt.find((n) => n[this.labelPropertyName] === t[this.labelPropertyName] || n[this.labelPropertyName] === t);
        if (e)
          return e.idInt;
      }
      return a();
    },
    sortSelectedList() {
      this.isActiveInt = !1, o(
        this.selectedListInt,
        this.labelPropertyName,
        this.sortName,
        this.language ? this.getLangLabel : null
      ), this.updateParentList(this.selectedListInt);
    },
    setDragElement(t, e) {
      const n = e.cloneNode(!0);
      n.id = "chip-inline-drag", n.style.position = "absolute", n.style.top = "-99999px", n.style.left = "-99999px", document.body.appendChild(n), t.setDragImage(n, 0, 0);
    },
    onDragEnd() {
      this.drag = !1;
      const t = document.getElementById("chip-inline-drag");
      t && t.parentNode.removeChild(t), JSON.stringify(this.selectedList) !== JSON.stringify(this.selectedListInt) && this.updateParentList(this.selectedListInt);
    },
    hoverBoxActive(t, e) {
      this.$emit("hoverbox-active", { value: t, option: e });
    }
  }
};
var f = function() {
  var e = this, n = e._self._c;
  return n("div", { staticClass: "base-chips-input-field" }, [n("BaseInput", e._g({ ref: "baseInput", attrs: { id: e.id, placeholder: e.allowMultipleEntries || !e.selectedListInt.length ? e.placeholder : "", label: e.label, "show-label": e.showLabel, "show-input-border": e.showInputBorder, "is-active": e.isActiveInt, "use-form-field-styling": e.useFormFieldStyling, "drop-down-list-id": e.dropDownListId, "linked-list-option": e.linkedListOption, "hide-input-field": !e.allowMultipleEntries && !!e.selectedListInt.length, invalid: e.invalid, required: e.required, disabled: e.disabled, clearable: e.clearable, loadable: e.loadable, "error-message": e.errorMessage, "show-error-icon": e.showErrorIcon, "is-loading": e.isLoading, "input-class": e.inputClass, "set-focus-on-active": e.setFocusOnActive }, on: { "update:isActive": function(i) {
    e.isActiveInt = i;
  }, "update:is-active": function(i) {
    e.isActiveInt = i;
  }, keydown: [function(i) {
    return !i.type.indexOf("key") && e._k(i.keyCode, "enter", 13, i.key, "Enter") ? null : (i.preventDefault(), e.addOption.apply(null, arguments));
  }, e.checkKeyEvent] }, scopedSlots: e._u([{ key: "label-addition", fn: function() {
    return [e._t("label-addition"), e.sortable ? n("button", { staticClass: "base-chips-input-field__sort", attrs: { "aria-label": `${e.label}. ${e.sortText.replace(/[—–-]/, "to")}`, type: "button" }, on: { click: function(i) {
      return e.sortSelectedList(e.selectedListInt);
    } } }, [e._v(" " + e._s(e.sortText) + " ")]) : e._e()];
  }, proxy: !0 }, { key: "pre-input-field", fn: function() {
    return [e._t("pre-input-field")];
  }, proxy: !0 }, { key: "input-field-addition-before", fn: function() {
    return [e._t("input-field-addition-before"), e.displayChipsInline ? n("div", { staticClass: "base-chips-input-field__chips" }, [e.draggable && !e.chipsEditable ? [n("draggable", { attrs: { "set-data": e.setDragElement, "force-fallback": !0, animation: 200, handle: ".base-chip__text" }, on: { start: function(i) {
      e.drag = !0;
    }, end: e.onDragEnd }, model: { value: e.selectedListInt, callback: function(i) {
      e.selectedListInt = i;
    }, expression: "selectedListInt" } }, [n("transition-group", { staticClass: "base-chips-input-field__chips-transition", attrs: { name: e.drag ? null : "flip-list", type: "transition" } }, [e._l(e.selectedListInt, function(i, s) {
      return [e._t("chip", function() {
        return [n("BaseChip", { key: e.allowMultipleEntries ? "chip-" + i.idInt : s, attrs: { id: i.idInt, entry: e.getLangLabel(i[e.labelPropertyName], !0), "hover-box-content": e.hoverboxContent, "is-linked": e.alwaysLinked || i[e.identifierPropertyName] === 0 || !!i[e.identifierPropertyName], "chip-active": e.chipActiveForRemove === s, "is-removable": e.chipsRemovable }, on: { "remove-entry": function(l) {
          return e.removeEntry(i, s);
        }, "hoverbox-active": function(l) {
          return e.hoverBoxActive(l, i);
        } } })];
      }, null, {
        entry: i,
        index: s,
        chipActiveForRemove: e.chipActiveForRemove,
        removeEntry: e.removeEntry
      })];
    })], 2)], 1)] : [e._l(e.selectedListInt, function(i, s) {
      return [e._t("chip", function() {
        return [n("BaseChip", { key: e.allowMultipleEntries ? "chip-" + i.idInt : s, attrs: { id: i.idInt, entry: e.getLangLabel(i[e.labelPropertyName], !0), "hover-box-content": e.hoverboxContent, editable: e.chipsEditable, "is-linked": e.alwaysLinked || i[e.identifierPropertyName] === 0 || !!i[e.identifierPropertyName], "chip-active": e.chipActiveForRemove === s, "assistive-text": e.assistiveText.selectedOption, "is-removable": e.chipsRemovable }, on: { "remove-entry": function(l) {
          return e.removeEntry(i, s);
        }, "value-changed": function(l) {
          return e.modifyListEntry(l, s);
        }, "hoverbox-active": function(l) {
          return e.hoverBoxActive(l, i);
        } } })];
      }, null, {
        entry: i,
        index: s,
        chipActiveForRemove: e.chipActiveForRemove,
        removeEntry: e.removeEntry
      })];
    })]], 2) : e._e()];
  }, proxy: !0 }, { key: "input-field-inline-before", fn: function() {
    return [e._t("input-field-inline-before")];
  }, proxy: !0 }, { key: "input-field-addition-after", fn: function() {
    return [e._t("input-field-addition-after")];
  }, proxy: !0 }, { key: "post-input-field", fn: function() {
    return [e._t("post-input-field")];
  }, proxy: !0 }, { key: "error-icon", fn: function() {
    return [e._t("error-icon")];
  }, proxy: !0 }, { key: "remove-icon", fn: function() {
    return [e._t("remove-icon")];
  }, proxy: !0 }, { key: "below-input", fn: function() {
    return [e._t("below-input")];
  }, proxy: !0 }], null, !0), model: { value: e.inputInt, callback: function(i) {
    e.inputInt = i;
  }, expression: "inputInt" } }, e.$listeners))], 1);
}, m = [], y = /* @__PURE__ */ c(
  h,
  f,
  m,
  !1,
  null,
  "477631eb",
  null,
  null
);
const B = y.exports;
export {
  B as default
};
