// base-ui-components@3.1.2/dist/components/BaseResultBoxSection/BaseResultBoxSection.es.js
import { extractNestedPropertyValue as l } from "../../utils/utils.es.js";
import d from "../../mixins/navigateList.es.js";
import u from "../BaseImageBox/BaseImageBox.es.js";
import c from "../../mixins/i18n.es.js";
import "./BaseResultBoxSection.vue_vue_type_style_index_0_scoped_f33de01b_lang.es.js";
import p from "../../_virtual/_plugin-vue2_normalizer.es.js";
const h = {
  name: "BaseResultBoxSection",
  components: {
    BaseImageBox: u,
    BaseLoader: () => import("../BaseLoader/BaseLoader.es.js").then((t) => t.default || t),
    BaseOptions: () => import("../BaseOptions/BaseOptions.es.js").then((t) => t.default || t),
    BaseButton: () => import("../BaseButton/BaseButton.es.js").then((t) => t.default || t),
    BasePagination: () => import("../BasePagination/BasePagination.es.js").then((t) => t.default || t),
    BaseBoxButton: () => import("../BaseBoxButton/BaseBoxButton.es.js").then((t) => t.default || t),
    BaseSelectOptions: () => import("../BaseSelectOptions/BaseSelectOptions.es.js").then((t) => t.default || t)
  },
  mixins: [c, d],
  model: {
    prop: "entryList",
    event: "entries-changed"
  },
  props: {
    entryList: {
      type: Array,
      default: () => []
    },
    showHeader: {
      type: Boolean,
      default: !0
    },
    headerText: {
      type: String,
      default: ""
    },
    showOptions: {
      type: Boolean,
      default: !0
    },
    optionsButtonText: {
      type: Object,
      default: () => ({
        show: "edit",
        hide: "editReturn"
      }),
      validator: (t) => ["show", "hide"].every((e) => Object.keys(t).includes(e))
    },
    optionsButtonIcon: {
      type: Object,
      default: () => ({
        show: "edit",
        hide: "remove"
      }),
      validator: (t) => ["show", "hide"].every((e) => Object.keys(t).includes(e))
    },
    boxBreakpoints: {
      type: Array,
      default: () => [
        [0, 2],
        [640, 4],
        [1024, 6]
      ],
      validator: (t) => t.every((e) => typeof e == "object" && e.length === 2 && e.every((s) => typeof s == "number"))
    },
    isLoading: {
      type: Boolean,
      default: !1
    },
    editMode: {
      type: Boolean,
      default: !1
    },
    editModeWhiteBackground: {
      type: Boolean,
      default: !1
    },
    selectedList: {
      type: Array,
      default: () => []
    },
    selectOptionsText: {
      type: Object,
      default: () => ({
        selectAll: "select all",
        selectNone: "select none",
        entriesSelected: "entries selected"
      })
    },
    messageText: {
      type: String,
      default: "Drag or Select"
    },
    messageSubtext: {
      type: String,
      default: "Drag'n Drop to reorder or select the relevant items and choose an action"
    },
    draggable: {
      type: Boolean,
      default: !1
    },
    usePagination: {
      type: Boolean,
      default: !1
    },
    maxRows: {
      type: Number,
      default: 5
    },
    useExpandMode: {
      type: Boolean,
      default: !1
    },
    expanded: {
      type: Boolean,
      default: !1
    },
    expandText: {
      type: Object,
      default: () => ({
        expand: "more objects",
        collapse: "collapse"
      })
    },
    total: {
      type: Number,
      default: null
    },
    currentPageNumber: {
      type: Number,
      default: null
    },
    maxShowMoreRows: {
      type: Number,
      default: 1
    },
    jumpToTop: {
      type: Boolean,
      default: !1
    },
    showActionButtonBoxes: {
      type: Boolean,
      default: !1
    },
    actionButtonsConfig: {
      type: Array,
      default: () => [{
        text: "delete",
        icon: "waste-bin",
        value: "delete",
        display: "all",
        disabled: !1
      }],
      validator: (t) => t.every((e) => {
        const s = ["text", "icon", "value"], i = Object.keys(e);
        return s.every((o) => i.includes(o)) && (["all", "top", "bottom"].includes(e.display) || !e.display);
      })
    },
    identifierPropertyName: {
      type: String,
      default: "id"
    },
    titlePropertyName: {
      type: String,
      default: "title"
    },
    usePaginationLinkElement: {
      type: [String, Boolean],
      default: !1,
      validator: (t) => typeof t == "boolean" && !t || typeof t == "string" && t
    },
    fetchDataExternally: {
      type: Boolean,
      default: !1
    },
    scrollToOffset: {
      type: Number,
      default: 0
    },
    initialItemsPerRow: {
      type: Number,
      default: 6
    },
    supportiveText: {
      type: Object,
      default: () => ({
        description: "Select items via space bar to carry out actions or use enter key to select an item for reordering. Use Tab key to navigate between items.",
        activated: "Item at position {pos} selected for reordering. Use arrow keys to order item.",
        moved: "Item moved to position {pos} of {total}"
      })
    },
    disableListElementFocus: {
      type: Boolean,
      default: !1
    }
  },
  data() {
    return {
      entryListInt: [],
      editModeActive: !1,
      currentPageNumberInt: 1,
      expandedInt: !1,
      wasExpanded: !1,
      itemsPerRow: null,
      initialBoxCalcDone: !1,
      selectedListInt: [],
      imageBoxesSelectable: !1,
      elementId: null,
      initialized: !1,
      movableElementId: null,
      currentSupportiveText: ""
    };
  },
  computed: {
    draggableComponent() {
      return this.draggable ? () => import("vuedraggable").then((t) => t.default || t) : "ul";
    },
    visibleBoxes: {
      get() {
        return this.editModeActive && (this.draggable || !this.usePagination) ? [...this.entryListInt] : !this.editModeActive && this.useExpandMode && !this.expandedInt && this.expandNeeded ? this.entryListInt.slice(0, this.itemsPerRow * this.maxShowMoreRows - 1) : this.usePagination && !this.fetchDataExternally ? this.entryListInt.slice(
          (this.currentPageNumberInt - 1) * this.visibleNumberOfItems,
          this.currentPageNumberInt * this.visibleNumberOfItems
        ) : this.fetchDataExternally ? this.entryList.slice(0, this.visibleNumberOfItems) : this.entryListInt;
      },
      set(t) {
        this.entryListInt = [...t];
      }
    },
    visibleNumberOfItems() {
      const t = this.itemsPerRow * this.maxRows;
      return this.editModeActive && this.showActionButtonBoxes ? t - this.actionButtonsConfig.length : !this.editModeActive && this.useExpandMode ? t - 1 : t;
    },
    pages() {
      return (this.total || this.entryListInt.length) && this.visibleNumberOfItems >= 0 ? Math.ceil((this.total || this.entryListInt.length) / this.visibleNumberOfItems) : 1;
    },
    expandNeeded() {
      return this.itemsPerRow * this.maxShowMoreRows < this.entryList.length;
    }
  },
  watch: {
    entryList: {
      handler(t) {
        JSON.stringify(t) !== JSON.stringify(this.entryListInt) && (this.entryListInt = [...t]);
      },
      immediate: !0,
      deep: !0
    },
    entryListInt: {
      handler(t) {
        JSON.stringify(t) !== JSON.stringify(this.entryList) && this.$emit("entries-changed", t);
      },
      deep: !0
    },
    selectedList: {
      handler(t) {
        JSON.stringify(t) !== JSON.stringify(this.selectedListInt) && (this.selectedListInt = JSON.parse(JSON.stringify(t)));
      },
      immediate: !0
    },
    selectedListInt(t) {
      JSON.stringify(t) !== JSON.stringify(this.selectedList) && this.$emit("update:selected-list", t);
    },
    pages: {
      handler(t) {
        this.currentPageNumberInt > t && (this.currentPageNumberInt = t);
      },
      immediate: !0
    },
    currentPageNumber: {
      handler(t) {
        t !== this.currentPageNumberInt && (this.currentPageNumberInt = t > this.pages && !this.fetchDataExternally ? this.pages : t);
      },
      immediate: !0
    },
    currentPageNumberInt: {
      handler(t) {
        t !== this.currentPageNumber && this.$emit("update:current-page-number", this.currentPageNumberInt);
      },
      immediate: !0
    },
    expanded: {
      handler(t) {
        t !== this.expandedInt && (this.expandedInt = t);
      },
      immediate: !0
    },
    expandedInt(t) {
      t || (this.currentPageNumberInt = 1), t !== this.expanded && this.$emit("update:expanded", t);
    },
    editModeActive(t) {
      setTimeout(() => {
        this.imageBoxesSelectable = t;
      }, 50), setTimeout(() => {
        t && this.$nextTick(() => {
          this.$refs.resultBoxItem && this.$refs.resultBoxItem.length && (this.$refs.resultBoxItem.forEach((s) => {
            s.getElementsByTagName("input")[0].setAttribute("tabindex", "-1");
          }), this.$refs.resultBoxItem.find((s) => s.id.includes(this.visibleBoxes[0].id)).focus({ preventScroll: !0 }));
        });
      }, 500), t ? this.currentSupportiveText = this.supportiveText.description : (this.movableElementId = null, this.currentSupportiveText = ""), t !== this.editMode && this.$emit("update:edit-mode", t), this.initialized && (t ? window.addEventListener("keyup", this.escEventHandler) : window.removeEventListener("keyup", this.escEventHandler));
    },
    editMode: {
      handler(t) {
        t !== this.editModeActive && (this.editModeActive = t && this.showOptions && this.showHeader);
      },
      immediate: !0
    },
    showOptions(t) {
      t || (this.editModeActive = !1);
    },
    showHeader(t) {
      t || (this.editModeActive = !1);
    }
  },
  created() {
    this.useExpandMode || (this.expandedInt = !0), this.itemsPerRow = this.initialItemsPerRow;
  },
  mounted() {
    this.elementId = this._uid, !this.initialBoxCalcDone && this.$refs.resultBoxesArea && this.calcBoxNumber(), window.addEventListener("resize", this.resizeBoxes), this.initialized = !0;
  },
  updated() {
    !this.initialBoxCalcDone && this.$refs.resultBoxesArea && this.calcBoxNumber();
  },
  methods: {
    optionsToggle(t) {
      this.editModeActive = t, t ? this.wasExpanded = this.expandedInt : (this.expandedInt = this.wasExpanded, this.selectedListInt = []);
    },
    selectAllTriggered(t) {
      if (t)
        this.selectedListInt = [.../* @__PURE__ */ new Set([
          ...this.selectedListInt,
          ...this.visibleBoxes.map((e) => this.getPropValue(this.identifierPropertyName, e))
        ])];
      else {
        const e = this.visibleBoxes.map((s) => this.getPropValue(this.identifierPropertyName, s));
        this.selectedListInt = this.selectedListInt.filter((s) => !e.includes(this.getPropValue(this.identifierPropertyName, s) || s));
      }
    },
    entrySelected(t, e) {
      e && !this.selectedListInt.includes(t) ? this.selectedListInt.push(t) : e || (this.selectedListInt = this.selectedListInt.filter((s) => s !== t));
    },
    isEntrySelected(t) {
      return this.selectedListInt.map((e) => this.getPropValue(this.identifierPropertyName, e) || e).includes(this.getPropValue(this.identifierPropertyName, t));
    },
    submitAction(t) {
      this.$emit("submit-action", t);
    },
    escEventHandler(t) {
      t.key === "Escape" && (this.movableElementId ? this.cancelDragMode() : (this.editModeActive = !1, this.currentSupportiveText = ""));
    },
    onEnterKey(t, e, s) {
      this.editModeActive && this.draggable ? (t.preventDefault(), this.movableElementId ? this.cancelDragMode() : (this.movableElementId = e.id, this.currentSupportiveText = this.supportiveText.activated.replace("{pos}", (s + 1).toString()))) : this.entryClicked(this.getPropValue(this.identifierPropertyName, e));
    },
    moveEntry(t, e) {
      this.currentSupportiveText = "";
      const { key: s } = t, i = ["ArrowRight", "ArrowDown"].includes(s), n = e + (i ? 1 : -1);
      if (this.isWithinArrayLimit(this.visibleBoxes, i, n)) {
        const r = this.visibleBoxes;
        [r[e], r[n]] = [r[n], r[e]], this.visibleBoxes = r, this.$nextTick(() => {
          document.getElementById(`li-${this.movableElementId}`).focus();
        }), this.currentSupportiveText = this.supportiveText.moved.replace("{pos}", n + 1).replace("{total}", this.visibleBoxes.length);
      }
    },
    cancelDragMode() {
      this.movableElementId = null, this.currentSupportiveText = "";
    },
    resizeBoxes() {
      this.resizeTimeout && (clearTimeout(this.resizeTimeout), this.resizeTimeout = null), this.resizeTimeout = setTimeout(() => {
        this.calcBoxNumber();
      }, 300);
    },
    calcBoxNumber() {
      const t = this.$refs.resultBoxesArea && this.$refs.resultBoxesArea.$el ? this.$refs.resultBoxesArea.$el : this.$refs.resultBoxesArea;
      if (t) {
        const e = t.clientWidth, s = [...this.boxBreakpoints];
        this.itemsPerRow = s.sort((n, r) => n[0] > r[0]).reduce((n, [r, a]) => e > r ? a : n, 0), this.$el.style.setProperty("--items-per-row", this.itemsPerRow);
        const i = `base-result-box-section__box-style-${this.elementId}`;
        let o = document.getElementById(i);
        o || (o = document.createElement("style"), o.id = i, this.$el.appendChild(o)), o.innerHTML = `
          .base-result-box-section__box-item-${this.elementId}:nth-child(n + ${this.itemsPerRow + 1}) {
            margin-top: var(--spacing-regular);
          }
          .base-result-box-section__box-item-${this.elementId}:not(:nth-child(${this.itemsPerRow}n)) {
            margin-right: var(--spacing-regular);
          }`, this.initialBoxCalcDone = !0, this.$emit("items-per-row-changed", this.itemsPerRow);
      }
    },
    setPage(t) {
      this.currentPageNumberInt = t, this.jumpToTop && window.scrollTo(0, this.$el.offsetTop - this.scrollToOffset);
    },
    entryClicked(t) {
      this.$emit("entry-clicked", { entryId: t });
    },
    getI18nString(t) {
      if (typeof t == "string")
        return this.getI18nTerm(t);
      const { string: e, count: s, variables: i } = { ...t };
      return this.getI18nTerm(e, s, i);
    },
    getPropValue(t, e) {
      return l(t, e);
    }
  }
};
var m = function() {
  var e = this, s = e._self._c;
  return s("div", { staticClass: "base-result-box-section" }, [e.isLoading ? s("div", { staticClass: "base-result-box-section__loading" }, [s("BaseLoader", { staticClass: "base-result-box-section__loader" })], 1) : e._e(), s("div", { staticClass: "base-result-box-section__container" }, [e.showHeader ? s("div", { staticClass: "base-result-box-section__header-row" }, [e.showOptions ? s("BaseOptions", { attrs: { "show-options": e.editModeActive, "options-button-icon": e.optionsButtonIcon, "options-button-text": e.optionsButtonText, "use-options-button-on": "always" }, on: { "update:show-options": e.optionsToggle }, scopedSlots: e._u([{ key: "beforeOptions", fn: function() {
    return [e._t("header", function() {
      return [e.headerText ? s("h3", { staticClass: "base-result-box-section__header" }, [e._v(" " + e._s(e.headerText) + " ")]) : e._e()];
    })];
  }, proxy: !0 }, { key: "options", fn: function() {
    return [e._t("optionButtons", function() {
      return [e._l(e.actionButtonsConfig, function(i) {
        return [i.display === "top" || i.display === "all" || !i.display ? s("BaseButton", { key: i.text, attrs: { text: e.getI18nString(i.text), icon: i.icon, "has-background-color": !1, disabled: i.disabled, "icon-size": "large", "button-style": "single" }, on: { clicked: function(o) {
          return e.submitAction(i.value);
        } } }) : e._e()];
      })];
    }, { submitAction: e.submitAction })];
  }, proxy: !0 }], null, !0) }) : [e._t("header", function() {
    return [e.headerText ? s("h3", { staticClass: "base-result-box-section__header" }, [e._v(" " + e._s(e.headerText) + " ")]) : e._e()];
  })]], 2) : e._e(), s("div", { class: [
    "base-result-box-section__background",
    {
      "base-result-box-section__background--white": e.editModeWhiteBackground && e.editModeActive
    }
  ] }, [e.editModeActive ? s("div", { key: e.headerText + "_messageArea", staticClass: "base-result-box-section__message-area" }, [s("div", { staticClass: "base-result-box-section__message-area-text" }, [e._v(" " + e._s(e.messageText) + " ")]), s("span", { staticClass: "base-result-box-section__message-area-subtext" }, [e._v(e._s(e.messageSubtext))]), e._t("optionsMessageAreaAfter")], 2) : e._e(), e.editModeActive ? s("BaseSelectOptions", { key: e.headerText + "_selectOptions", attrs: { "selected-number-text": e.getI18nString(e.selectOptionsText.entriesSelected), "select-text": e.getI18nString(e.selectOptionsText.selectAll), "deselect-text": e.getI18nString(e.selectOptionsText.selectNone), list: e.visibleBoxes, "selected-list": e.selectedListInt }, on: { selected: e.selectAllTriggered } }) : e._e(), s("span", { staticClass: "supportive-text", attrs: { "aria-live": "assertive" } }, [e._v(" " + e._s(e.currentSupportiveText) + " ")]), e.entryListInt.length ? [s(e.draggableComponent, { ref: "resultBoxesArea", tag: "component", staticClass: "base-result-box-section__boxes-container", attrs: { animation: e.draggable ? 150 : !1, tag: e.draggable ? "ul" : !1, draggable: e.editModeActive ? ".base-result-box-section__result-box-item" : !1, "aria-label": e.headerText, tabindex: "0", handle: ".base-result-box-section__result-box-item__draggable .base-image-box-content", "force-fallback": "true", role: "list" }, model: { value: e.visibleBoxes, callback: function(i) {
    e.visibleBoxes = i;
  }, expression: "visibleBoxes" } }, [e._l(e.visibleBoxes, function(i, o) {
    return s("li", { key: e.getPropValue(e.identifierPropertyName, i), ref: "resultBoxItem", refInFor: !0, class: [
      "base-result-box-section__box-item",
      "base-result-box-section__result-box-item",
      { "base-result-box-section__box-item__hidden": !e.initialBoxCalcDone },
      `base-result-box-section__box-item-${e.elementId}`,
      {
        "base-result-box-section__result-box-item__draggable": e.draggable && e.editModeActive
      },
      {
        "base-result-box-section__result-box-item__dragging": e.movableElementId === i.id
      }
    ], attrs: { id: `li-${i.id}`, tabindex: e.editModeActive || !e.disableListElementFocus ? 0 : -1, "aria-label": e.getPropValue(e.titlePropertyName, i), "aria-grabbed": (e.movableElementId === i.id).toString(), "aria-selected": e.editModeActive ? e.isEntrySelected(i).toString() : !1 }, on: { keydown: [function(n) {
      return !n.type.indexOf("key") && e._k(n.keyCode, "enter", 13, n.key, "Enter") ? null : e.onEnterKey(n, i, o);
    }, function(n) {
      if (!n.type.indexOf("key") && e._k(n.keyCode, "up", 38, n.key, ["Up", "ArrowUp"]) && e._k(n.keyCode, "down", 40, n.key, ["Down", "ArrowDown"]) && e._k(n.keyCode, "left", 37, n.key, ["Left", "ArrowLeft"]) && e._k(n.keyCode, "right", 39, n.key, ["Right", "ArrowRight"]) || "button" in n && n.button !== 0 || "button" in n && n.button !== 2)
        return null;
      n.preventDefault(), e.editModeActive && e.draggable && e.movableElementId && e.moveEntry(n, o);
    }, function(n) {
      if (!n.type.indexOf("key") && e._k(n.keyCode, "space", 32, n.key, [" ", "Spacebar"]))
        return null;
      n.preventDefault(), e.editModeActive && e.entrySelected(
        e.getPropValue(e.identifierPropertyName, i),
        !e.isEntrySelected(i)
      );
    }, function(n) {
      return !n.type.indexOf("key") && e._k(n.keyCode, "tab", 9, n.key, "Tab") ? null : e.cancelDragMode.apply(null, arguments);
    }] } }, [e._t("resultBox", function() {
      return [s("BaseImageBox", { key: e.getPropValue(e.identifierPropertyName, i), attrs: { selectable: e.imageBoxesSelectable, draggable: e.editModeActive && e.draggable, selected: e.isEntrySelected(i), "box-size": { width: "auto" }, "box-ratio": 100, title: e.getPropValue(e.titlePropertyName, i), subtext: i.subtext, description: i.description, "image-url": i.imageUrl, "box-text": i.text, lazyload: !0 }, on: { "select-triggered": function(n) {
        e.entrySelected(
          e.getPropValue(e.identifierPropertyName, i),
          n
        );
      }, clicked: function(n) {
        e.entryClicked(e.getPropValue(e.identifierPropertyName, i));
      } } })];
    }, { item: i, index: o, selectActive: e.editModeActive, isEntrySelected: e.isEntrySelected, entrySelected: e.entrySelected })], 2);
  }), e.showActionButtonBoxes && e.editModeActive ? e._t("actionButtons", function() {
    return [e._l(e.actionButtonsConfig, function(i) {
      return [i.display === "bottom" || i.display === "all" || !i.display ? s("BaseBoxButton", { key: i.value, class: [
        "base-result-box-section__box-item",
        `base-result-box-section__box-item-${e.elementId}`
      ], attrs: { text: e.getI18nString(i.text), "box-size": { width: "calc(25% - 8rem/19 - (8rem/19/2))", height: "100%" }, icon: i.icon, "box-style": "small", "box-type": "button" }, on: { clicked: function(o) {
        return e.submitAction(i.value);
      } } }) : e._e()];
    })];
  }, { itemsPerRow: e.itemsPerRow, elementId: e.elementId }) : e.useExpandMode && !e.editModeActive && e.expandNeeded && e.initialBoxCalcDone ? s("BaseBoxButton", { class: [
    "base-result-box-section__box-item",
    `base-result-box-section__box-item-${e.elementId}`
  ], attrs: { "box-size": { width: "calc(25% - 8rem/19 - (8rem/19/2))", height: "100%" }, icon: "", text: "", "box-type": "button" }, on: { clicked: function(i) {
    e.expandedInt = !e.expandedInt;
  } }, scopedSlots: e._u([{ key: "default", fn: function() {
    return [e.editModeActive ? e._e() : s("div", { staticClass: "base-result-box-section__expand-button__content" }, [e.expandedInt ? e._e() : s("span", { staticClass: "base-result-box-section__expand-button__content-number" }, [e._v(" " + e._s(`+${(e.total || e.entryList.length) - e.visibleBoxes.length}`) + " ")]), s("span", { class: [e.expandedInt ? "base-result-box-section__expand-button__content-text-expanded" : "base-result-box-section__expand-button__content-text-collapsed"] }, [e._v(" " + e._s(e.expandedInt ? e.expandText.collapse : e.expandText.expand) + " ")])])];
  }, proxy: !0 }], null, !1, 500459990) }) : e._e()], 2), e.usePagination && e.pages > 1 && (e.editModeActive && !e.draggable || !e.editModeActive && e.expandedInt) ? s("BasePagination", { key: "pagination-" + e.elementId, attrs: { total: e.pages, current: e.currentPageNumberInt, "use-link-element": e.usePaginationLinkElement }, on: { "set-page": e.setPage } }) : e._e()] : e._e()], 2)])]);
}, f = [], b = /* @__PURE__ */ p(
  h,
  m,
  f,
  !1,
  null,
  "f33de01b",
  null,
  null
);
const I = b.exports;
export {
  I as default
};
