// base-ui-components@3.1.2/dist/components/BaseMediaCarousel/BaseMediaCarousel.es.js
import l from "./BaseMediaCarouselItem.es.js";
import d from "../BaseIcon/BaseIcon.es.js";
import p from "../../mixins/popUpLock.es.js";
import "./BaseMediaCarousel.vue_vue_type_style_index_0_scoped_e14fdf5f_lang.es.js";
import u from "../../_virtual/_plugin-vue2_normalizer.es.js";
const c = {
  name: "BaseMediaCarousel",
  components: {
    BaseMediaCarouselItem: l,
    BaseIcon: d
  },
  mixins: [p],
  props: {
    items: {
      type: Array,
      default: () => []
    },
    initialSlide: {
      type: [Number, String],
      default: 0
    },
    allowDownload: {
      type: Boolean,
      default: !0
    },
    infoTexts: {
      type: Object,
      default: () => ({
        download: "Download",
        view: "View"
      })
    },
    showPreview: {
      type: Boolean,
      default: !1
    },
    swiperOptions: {
      type: Object,
      default: () => ({
        speed: 500,
        keyboard: {
          enabled: !0
        }
      })
    }
  },
  data() {
    return {
      showInt: this.showPreview,
      targetName: "mediaCarousel",
      swiper: null,
      swiperId: `base-media-carousel__swiper${this._uid}`,
      isMounted: !1
    };
  },
  watch: {
    showPreview(e) {
      this.showInt = e;
    }
  },
  mounted() {
    this.isMounted = !0;
  },
  updated() {
    this.$nextTick(() => {
      this.isMounted && this.showInt && this.swiper === null && (this.initSwiper(), this.$el.addEventListener("keyup", (e) => this.escapeEvent(e)), this.$el.addEventListener("keydown", (e) => this.tabEvents(e))), this.showInt || (this.swiper = null, this.$el.removeEventListener("keyup", (e) => this.escapeEvent(e)), this.$el.removeEventListener("keydown", (e) => this.tabEvents(e)));
    });
  },
  methods: {
    hide() {
      this.$emit("hide");
    },
    async initSwiper() {
      const { Swiper: e } = await import("swiper"), { Keyboard: t } = await import("swiper"), { Navigation: s } = await import("swiper"), i = {
        init: !1,
        initialSlide: this.initialSlide,
        navigation: {
          nextEl: ".swiper-button-next",
          prevEl: ".swiper-button-prev"
        },
        lazyPreloaderClass: "base-media-preview-preloader",
        threshold: 10,
        modules: [s, t]
      };
      this.swiper = new e(`#${this.swiperId}`, {
        ...this.swiperOptions,
        ...i
      }), this.swiper.on("slideChange", () => {
        this.mediaPause();
      }), this.swiper.on("transitionEnd", () => {
        this.swiper && this.$refs.baseMedia[this.swiper.activeIndex].$el.focus();
      }), this.swiper.on("init", () => {
        setTimeout(() => {
          window.dispatchEvent(new Event("resize"));
        }, 0);
      }), this.swiper.init(), this.$refs.baseMedia[this.swiper.activeIndex].$el.focus();
    },
    mediaPause() {
      const e = this.$refs.baseMedia[this.swiper.previousIndex];
      e && e.$refs.baseMediaVideo && e.$refs.baseMediaVideo.pause(), e && e.$refs.baseMediaAudio && e.$refs.baseMediaAudio.pause();
    },
    escapeEvent(e) {
      e.key === "Escape" && this.hide();
    },
    tabEvents(e) {
      if (e.shiftKey && e.key === "Tab") {
        e.preventDefault(), this.setFocus("prev");
        return;
      }
      e.key === "Tab" && (e.preventDefault(), this.setFocus("next"));
    },
    getFocusableItems() {
      const e = 'button, audio, video[tabindex="0"]', t = [];
      return this.$refs.baseMedia.forEach((s) => {
        const i = Array.from(s.$el.querySelectorAll(e));
        i.unshift(s.$el), t.push(i);
      }), t;
    },
    setFocus(e = "next") {
      const t = this.getFocusableItems();
      let s, i;
      if (t.forEach((a, o) => {
        a.forEach((r, n) => {
          r.matches(":focus") && (s = o, i = n);
        });
      }), e === "next") {
        if (t[s] && t[s][i + 1]) {
          t[s][i + 1].focus();
          return;
        }
        this.swiper.slideNext();
      }
      if (e === "prev") {
        if (t[s] && t[s][i - 1]) {
          t[s][i - 1].focus();
          return;
        }
        this.swiper.slidePrev();
      }
    },
    download(e) {
      this.$emit("download", e);
    }
  }
};
var h = function() {
  var t = this, s = t._self._c;
  return s("transition", { attrs: { name: "grow" } }, [t.showInt ? s("div", { ref: "mediaCarousel", staticClass: "base-media-carousel" }, [s("div", { staticClass: "base-media-carousel__background" }), s("button", { staticClass: "base-media-carousel__close", on: { click: t.hide } }, [s("base-icon", { attrs: { name: "remove" } })], 1), s("div", { staticClass: "swiper-container", attrs: { id: t.swiperId } }, [t.items.length > 1 ? [s("base-icon", { staticClass: "swiper-button swiper-button-prev", attrs: { name: "prev" } }), s("base-icon", { staticClass: "swiper-button swiper-button-next", attrs: { name: "next" } })] : t._e(), s("div", { staticClass: "swiper-wrapper" }, t._l(t.items, function(i, a) {
    return s("div", { key: a, staticClass: "swiper-slide" }, [s("base-media-carousel-item", { ref: "baseMedia", refInFor: !0, attrs: { autoplay: a === t.initialSlide, "additional-info": i.additionalInfo, "allow-download": t.allowDownload, "current-slide-info": t.items.length > 1 ? `${a + 1} / ${t.items.length}` : "", "display-name": i.title, "display-size": i.displaySize, "download-url": i.downloadUrl, "info-texts": t.infoTexts, "media-url": i.mediaUrl, "media-poster-url": i.mediaPosterUrl, "media-type": i.mediaType, orientation: i.orientation, previews: i.previews, "hls-start-level": i.hlsStartLevel, tabindex: "0" }, on: { download: t.download } })], 1);
  }), 0)], 2)]) : t._e()]);
}, w = [], f = /* @__PURE__ */ u(
  c,
  h,
  w,
  !1,
  null,
  "e14fdf5f",
  null,
  null
);
const $ = f.exports;
export {
  $ as default
};
