// base-ui-components/dist/components/BaseHoverBox/BaseHoverBox.es.js
import r from "../BaseImageBox/BaseImageBox.es.js";
import "./BaseHoverBox.vue_vue_type_style_index_0_scoped_846d1be6_lang.es.js";
import a from "../../_virtual/_plugin-vue2_normalizer.es.js";
const s = {
  name: "BaseHoverBox",
  components: {
    BaseImageBox: r
  },
  props: {
    title: {
      type: String,
      default: "No title"
    },
    subtext: {
      type: String,
      default: ""
    },
    showTitle: {
      type: Boolean,
      default: !0
    },
    boxText: {
      type: Array,
      default() {
        return [];
      }
    },
    imageUrl: {
      type: String,
      default: null
    },
    description: {
      type: String,
      default: null
    },
    selectable: {
      type: Boolean,
      default: !1
    },
    boxSize: {
      type: Object,
      default: () => ({ width: "212px", height: "212px" })
    }
  },
  data() {
    return {
      boxPosition: { top: 0, left: 0 }
    };
  },
  methods: {
    setPosition(t) {
      let e = t.layerX, o = t.layerY;
      const i = parseInt(this.boxSize.width.replace("px", ""), 10), n = this.$el, l = window.getComputedStyle(n, null).getPropertyValue("margin").replace("px", "");
      t.clientX + 3 * l + i > window.innerWidth && (e = t.layerX - i - 2 * l), t.clientY + 2 * l + i > window.innerHeight && (o = t.layerY - i - l), this.boxPosition = { top: `${o}px`, left: `${e}px` };
    }
  }
};
var p = function() {
  var e = this, o = e._self._c;
  return o("div", { staticClass: "base-hover-box", style: [e.boxSize, e.boxPosition] }, [o("base-image-box", e._b({}, "base-image-box", e.$props, !1))], 1);
}, d = [], x = /* @__PURE__ */ a(
  s,
  p,
  d,
  !1,
  null,
  "846d1be6",
  null,
  null
);
const m = x.exports;
export {
  m as default
};
