// base-ui-components/dist/components/BaseBreadCrumbs/BaseBreadCrumbs.es.js
import n from "../BaseLink/BaseLink.es.js";
import "./BaseBreadCrumbs.vue_vue_type_style_index_0_scoped_1de29a15_lang.es.js";
import o from "../../_virtual/_plugin-vue2_normalizer.es.js";
const l = {
  name: "BaseBreadCrumbs",
  components: { BaseLink: n },
  props: {
    routes: {
      type: Array,
      default: () => [{ route: "", label: "" }],
      validator: (t) => !t.some(
        (e) => e.route && (e.route.match(/^([a-z][a-z0-9+\-.]*:\/\/)/) || e.route.match(/^mailto:/) || e.route.match(/^tel:/))
      )
    },
    className: {
      type: String,
      default: "base-bread-crumbs"
    },
    renderLinkAs: {
      type: String,
      default: "RouterLink"
    }
  }
};
var u = function() {
  var e = this, r = e._self._c;
  return r("div", { class: e.className }, [e._l(e.routes, function(s, a) {
    return [r("BaseLink", { key: a, staticClass: "base-bread-crumbs__link", attrs: { "external-link-target": "_self", source: s.route, value: s.label, "render-link-as": e.renderLinkAs } }), a < e.routes.length - 1 ? r("span", { key: `${a}-separator`, staticClass: "base-bread-crumbs__separator" }, [e._v(">")]) : e._e()];
  })], 2);
}, c = [], m = /* @__PURE__ */ o(
  l,
  u,
  c,
  !1,
  null,
  "1de29a15",
  null,
  null
);
const p = m.exports;
export {
  p as default
};
