// base-ui-components@3.1.2/dist/components/BaseChipsBelow/BaseChipsBelow.es.js
import n from "vuedraggable";
import l from "../BaseIcon/BaseIcon.es.js";
import s from "../BaseChipsInput/BaseChipsInput.es.js";
import d from "../BaseChip/BaseChip.es.js";
import "./BaseChipsBelow.vue_vue_type_style_index_0_lang.es.js";
import p from "../../_virtual/_plugin-vue2_normalizer.es.js";
const c = {
  name: "BaseChipsBelow",
  components: {
    BaseChipsInput: s,
    Draggable: n,
    BaseChip: d,
    BaseIcon: l
  },
  model: {
    prop: "selectedList",
    event: "selected-changed"
  },
  props: {
    list: {
      type: Array,
      default: () => []
    },
    selectedList: {
      type: Array,
      default: () => []
    },
    id: {
      type: String,
      default: ""
    },
    label: {
      type: String,
      required: !0
    },
    showLabel: {
      type: Boolean,
      default: !0
    },
    placeholder: {
      type: String,
      default: null
    },
    dropDownNoOptionsInfo: {
      type: String,
      default: "No options available"
    },
    allowUnknownEntries: {
      type: Boolean,
      default: !1
    },
    allowMultipleEntries: {
      type: Boolean,
      default: !0
    },
    allowDynamicDropDownEntries: {
      type: Boolean,
      default: !1
    },
    addSelectedEntryDirectly: {
      type: Boolean,
      default: !0
    },
    additionalPropOptions: {
      type: Array,
      default: () => []
    },
    additionalPropPlaceholder: {
      type: String,
      default: "Select role(s)"
    },
    hoverboxContent: {
      type: Object,
      default: () => ({})
    },
    isLoading: {
      type: Boolean,
      default: !1
    },
    sortText: {
      type: String,
      default: "Sort A \u2013 Z"
    },
    sortName: {
      type: Boolean,
      default: !1
    },
    language: {
      type: String,
      default: ""
    },
    addNewChipText: {
      type: String,
      default: ""
    },
    identifierPropertyName: {
      type: String,
      default: "id"
    },
    labelPropertyName: {
      type: String,
      default: "label"
    },
    additionalPropertyName: {
      type: String,
      default: "roles"
    },
    required: {
      type: Boolean,
      default: !1
    },
    invalid: {
      type: Boolean,
      default: !1
    },
    disabled: {
      type: Boolean,
      default: !1
    },
    errorMessage: {
      type: String,
      default: ""
    },
    showErrorIcon: {
      type: Boolean,
      default: !1
    },
    clearable: {
      type: Boolean,
      default: !1
    },
    inputClass: {
      type: String,
      default: ""
    }
  },
  data() {
    return {
      chipsArray: [],
      selectedBelowListInt: [],
      chipActive: -1
    };
  },
  computed: {
    chipsInputProps() {
      const i = { ...this.$props };
      return delete i.language, delete i.additionalPropOptions, delete i.additionalPropPlaceholder, delete i.additionalPropertyName, i;
    },
    isDragDropCapable() {
      return window ? !(navigator.vendor && navigator.vendor.indexOf("Apple") > -1 && navigator.userAgent && navigator.userAgent.indexOf("CriOS") === -1) && "DragEvent" in window : !1;
    }
  },
  watch: {
    selectedList: {
      handler(i) {
        this.createInternalList(i);
      },
      immediate: !0
    }
  },
  methods: {
    addedEntry(i) {
      this.emitInternalList(i.map((e) => typeof e == "object" ? {
        ...e,
        [this.additionalPropertyName]: e[this.additionalPropertyName] || []
      } : {
        [this.labelPropertyName]: e,
        [this.additionalPropertyName]: e[this.additionalPropertyName] || []
      }));
    },
    removeEntry(i, e) {
      const a = this.selectedBelowListInt.splice(e, 1);
      this.$set(a, this.additionalPropertyName, {}), this.emitInternalList(this.selectedBelowListInt);
    },
    updateList(i, e) {
      this.emitInternalList(e);
    },
    updateAdditionalProperty(i, e) {
      this.$set(this.selectedBelowListInt[e], this.additionalPropertyName, i), this.emitInternalList(this.selectedBelowListInt);
    },
    createInternalList(i) {
      this.selectedBelowListInt = i.map((e, a) => typeof e == "object" ? {
        [this.additionalPropertyName]: [],
        idInt: this.identifierPropertyName && (e[this.identifierPropertyName] === 0 || e[this.identifierPropertyName]) ? e[this.identifierPropertyName] : e[this.labelPropertyName] + a,
        ...e
      } : {
        [this.labelPropertyName]: e,
        idInt: this.list.length + a,
        [this.additionalPropertyName]: []
      });
    },
    emitInternalList(i) {
      const e = [];
      i.forEach((a, t) => this.$set(e, t, { ...a })), e.forEach((a) => this.$delete(a, "idInt")), this.$emit("selected-changed", e);
    },
    modifyChipValue(i, e) {
      if (!i)
        this.selectedBelowListInt.splice(e, 1);
      else {
        const a = { ...this.selectedBelowListInt[e] };
        this.identifierPropertyName && this.$set(a, this.identifierPropertyName, ""), this.$set(a, this.labelPropertyName, i), this.$set(this.selectedBelowListInt, e, a);
      }
      this.emitInternalList(this.selectedBelowListInt);
    },
    fetchDropDownEntries(i) {
      this.$emit("fetch-dropdown-entries", i);
    },
    hoverBoxActive(i, e) {
      this.$emit("hoverbox-active", { value: i, entry: e });
    }
  }
};
var u = function() {
  var e = this, a = e._self._c;
  return a("div", { staticClass: "base-chips-below" }, [a("BaseChipsInput", e._b({ ref: "chipsInput", attrs: { sortable: !0, "is-loading": e.isLoading, "sort-text": e.sortText, "sort-name": e.sortName, "display-chips-inline": !1 }, on: { "selected-changed": e.addedEntry, "fetch-dropdown-entries": e.fetchDropDownEntries }, scopedSlots: e._u([{ key: "drop-down-entry", fn: function(t) {
    return [e._t("drop-down-entry", null, { item: t.item })];
  } }, { key: "label-addition", fn: function() {
    return [e._t("label-addition")];
  }, proxy: !0 }, { key: "input-field-addition-before", fn: function() {
    return [e._t("input-field-addition-before")];
  }, proxy: !0 }, { key: "input-field-addition-after", fn: function() {
    return [e._t("input-field-addition-after")];
  }, proxy: !0 }, { key: "post-input-field", fn: function() {
    return [e._t("post-input-field")];
  }, proxy: !0 }, { key: "error-icon", fn: function() {
    return [e._t("error-icon")];
  }, proxy: !0 }, { key: "remove-icon", fn: function() {
    return [e._t("remove-icon")];
  }, proxy: !0 }, { key: "no-options", fn: function() {
    return [e._t("no-options")];
  }, proxy: !0 }], null, !0), model: { value: e.selectedBelowListInt, callback: function(t) {
    e.selectedBelowListInt = t;
  }, expression: "selectedBelowListInt" } }, "BaseChipsInput", e.chipsInputProps, !1)), a("draggable", { attrs: { animation: 200, "force-fallback": !e.isDragDropCapable, "fallback-on-body": !e.isDragDropCapable, group: { name: "people" }, handle: ".base-chips-below-list-icon-wrapper" }, on: { end: function(t) {
    return e.updateList(t, e.selectedBelowListInt);
  } }, model: { value: e.selectedBelowListInt, callback: function(t) {
    e.selectedBelowListInt = t;
  }, expression: "selectedBelowListInt" } }, e._l(e.selectedBelowListInt, function(t, r) {
    return a("div", { key: "item" + t.idInt, staticClass: "base-chips-below-list-item", attrs: { name: t[e.labelPropertyName] }, on: { mousedown: function(o) {
      e.chipActive = r;
    } } }, [a("div", { key: "line" + t.idInt, staticClass: "base-chips-below-list-item-line" }, [a("div", { key: "iconwrapper" + t.idInt, staticClass: "base-chips-below-list-icon-wrapper" }, [a("base-icon", { key: "icon" + t.idInt, staticClass: "svg-icon base-chips-below-list-icon", attrs: { name: "drag-lines" } })], 1), a("div", { key: "chip-wrapper" + t.idInt, staticClass: "base-chips-below-list-item-chip-wrapper" }, [a("BaseChip", { key: "chip" + t.idInt, ref: "selectedChip", refInFor: !0, staticClass: "base-chips-input-chip", attrs: { id: "chips-below" + r, "is-linked": !t.edited && (t[e.identifierPropertyName] === 0 || !!t[e.identifierPropertyName]), "hover-box-content": e.hoverboxContent }, on: { "value-changed": function(o) {
      return e.modifyChipValue(o, r);
    }, "hoverbox-active": function(o) {
      return e.hoverBoxActive(o, t);
    }, "remove-entry": function(o) {
      return e.removeEntry(o, r);
    } }, model: { value: t[e.labelPropertyName], callback: function(o) {
      e.$set(t, e.labelPropertyName, o);
    }, expression: "entry[labelPropertyName]" } })], 1), a("BaseChipsInput", { key: "input_" + t.idInt, staticClass: "base-chips-below-chips-input", attrs: { id: `${e.id}_${e.additionalPropertyName}_${t[e.identifierPropertyName] || t.idInt}`, "show-label": !1, label: e.label + "-" + e.additionalPropertyName, list: e.additionalPropOptions, "show-input-border": !1, "allow-dynamic-drop-down-entries": !1, placeholder: e.additionalPropPlaceholder, "always-linked": !0, language: e.language, draggable: !0, "drop-down-no-options-info": e.dropDownNoOptionsInfo, "identifier-property-name": e.identifierPropertyName, "label-property-name": e.labelPropertyName }, on: { "selected-changed": function(o) {
      return e.updateAdditionalProperty(o, r);
    } }, model: { value: t[e.additionalPropertyName], callback: function(o) {
      e.$set(t, e.additionalPropertyName, o);
    }, expression: "entry[additionalPropertyName]" } })], 1)]);
  }), 0)], 1);
}, f = [], h = /* @__PURE__ */ p(
  c,
  u,
  f,
  !1,
  null,
  null,
  null,
  null
);
const I = h.exports;
export {
  I as default
};
