// base-ui-components@3.1.2/dist/components/BaseSearch/BaseSearch.es.js
import { createId as s } from "../../utils/utils.es.js";
import "./BaseSearch.vue_vue_type_style_index_0_scoped_635e5927_lang.es.js";
import "./BaseSearch.vue_vue_type_style_index_1_lang.es.js";
import a from "../../_virtual/_plugin-vue2_normalizer.es.js";
const r = {
  name: "BaseSearch",
  components: {
    BaseIcon: () => import("../BaseIcon/BaseIcon.es.js").then((t) => t.default || t),
    BaseChipsInputField: () => import("../BaseChipsInputField/BaseChipsInputField.es.js").then((t) => t.default || t),
    BaseInput: () => import("../BaseInput/BaseInput.es.js").then((t) => t.default || t),
    BaseDateInput: () => import("../BaseDateInput/BaseDateInput.es.js").then((t) => t.default || t)
  },
  model: {
    prop: "input",
    event: "input"
  },
  props: {
    input: {
      type: [String, Object],
      default: ""
    },
    selectedChips: {
      type: Array,
      default: () => []
    },
    placeholder: {
      type: [String, Object],
      default: "Search your works and events"
    },
    label: {
      type: String,
      default: "Search"
    },
    showPreInputIcon: {
      type: Boolean,
      default: !0
    },
    dropDownListId: {
      type: String,
      default: ""
    },
    id: {
      type: String,
      default: ""
    },
    type: {
      type: String,
      default: "text",
      validator: (t) => ["text", "chips", "chipssingle", "controlled", "date", "daterange"].includes(t)
    },
    linkedListOption: {
      type: String,
      default: null
    },
    loadable: {
      type: Boolean,
      default: !1
    },
    isLoading: {
      type: Boolean,
      default: !1
    },
    language: {
      type: String,
      default: "en",
      validator: (t) => !t || t.length === 2
    },
    identifierPropertyName: {
      type: String,
      default: "id"
    },
    labelPropertyName: {
      type: String,
      default: "label"
    },
    invalid: {
      type: Boolean,
      default: !1
    },
    showErrorIcon: {
      type: Boolean,
      default: !0
    },
    clearable: {
      type: Boolean,
      default: !0
    },
    isActive: {
      type: Boolean,
      default: !1
    },
    setFocusOnActive: {
      type: Boolean,
      default: !0
    },
    assistiveText: {
      type: Object,
      default: () => ({})
    },
    dateFieldDelay: {
      type: Number,
      default: 0
    }
  },
  data() {
    return {
      isActiveInt: !1,
      textInputInt: "",
      dateInputInt: "",
      selectedChipsInt: []
    };
  },
  computed: {
    inputComponent() {
      return this.type === "text" ? "BaseInput" : this.isFieldTypeChips ? "BaseChipsInputField" : this.type === "date" || this.type === "daterange" ? "BaseDateInput" : null;
    },
    inputInt: {
      set(t) {
        this.type === "date" ? (this.dateInputInt = t, this.$emit("input", this.dateInputInt)) : this.type === "daterange" ? (this.dateInputInt = { ...t }, this.$emit("input", this.dateInputInt)) : (this.textInputInt = t, this.$emit("input", this.textInputInt));
      },
      get() {
        return this.type === "date" ? this.dateInputInt.date_from || this.dateInputInt : this.type === "daterange" ? typeof this.dateInputInt == "object" ? this.dateInputInt : {
          date_from: this.dateInputInt,
          date_to: ""
        } : this.textInputInt;
      }
    },
    dateFieldType() {
      return this.type === "date" ? "single" : this.type === "daterange" ? "daterange" : !1;
    },
    languageInt() {
      return this.type === "date" || this.type === "daterange" ? ["de", "en", "fr"].includes(this.language) ? this.language : "en" : this.language;
    },
    isFieldTypeChips() {
      return this.type.includes("chips") || this.type === "controlled";
    },
    idInt() {
      return this.id || s();
    },
    placeholderInt() {
      return typeof this.placeholder == "string" ? this.placeholder : this.type.includes("date") ? this.placeholder.date : this.type === "controlled" || this.type === "chipssingle" ? this.placeholder.chips : this.placeholder[this.type];
    }
  },
  watch: {
    input: {
      handler(t) {
        t ? t !== this.inputInt && (this.inputInt = t) : (this.textInputInt = "", this.dateInputInt = this.type === "daterange" ? {
          date_from: "",
          date_to: ""
        } : "");
      },
      immediate: !0
    },
    inputInt(t) {
      t !== this.input && this.$emit("input", t);
    },
    selectedChips: {
      handler(t) {
        JSON.stringify(t) !== JSON.stringify(this.selectedChipsInt) && (this.selectedChipsInt = [...t]);
      },
      immediate: !0
    },
    selectedChipsInt(t) {
      JSON.stringify(t) !== JSON.stringify(this.selectedChips) && this.$emit("update:selected-chips", t);
    },
    isActive: {
      handler(t) {
        t !== this.isActiveInt && (this.isActiveInt = t);
      },
      immediate: !0
    },
    isActiveInt(t) {
      t !== this.isActive && this.$emit("update:is-active", t);
    }
  }
};
var p = function() {
  var e = this, n = e._self._c;
  return n("div", { staticClass: "base-search" }, [n(e.inputComponent, e._g({ tag: "component", staticClass: "base-search__input", attrs: { id: e.idInt, "selected-list": e.selectedChipsInt, "is-active": e.isActiveInt, type: e.dateFieldType, "show-label": !1, "use-form-field-styling": !1, "show-input-border": !1, label: e.label, placeholder: e.placeholderInt, "linked-list-option": e.linkedListOption, "drop-down-list-id": e.dropDownListId || (!1).toString(), "is-loading": e.isLoading, clearable: e.clearable, invalid: e.invalid, "show-error-icon": e.showErrorIcon, language: e.languageInt, "allow-unknown-entries": e.type === "chips", loadable: e.loadable, "chips-editable": e.type === "chips", "label-property-name": e.isFieldTypeChips ? e.labelPropertyName : !1, "identifier-property-name": e.isFieldTypeChips ? e.identifierPropertyName : !1, "set-focus-on-active": e.setFocusOnActive, "add-selected-entry-directly": !0, "assistive-text": e.isFieldTypeChips ? e.assistiveText : !1, "is-active-delay": e.dateFieldDelay, "allow-multiple-entries": e.isFieldTypeChips ? e.type !== "chipssingle" : !1, "chips-removable": e.type !== "chipssingle", "input-class": "base-search__input-field", "field-type": "search" }, on: { "update:selectedList": function(i) {
    e.selectedChipsInt = i;
  }, "update:selected-list": function(i) {
    e.selectedChipsInt = i;
  }, "update:isActive": function(i) {
    e.isActiveInt = i;
  }, "update:is-active": function(i) {
    e.isActiveInt = i;
  } }, scopedSlots: e._u([{ key: "pre-input-field", fn: function() {
    return [e._t("pre-input-field")];
  }, proxy: !0 }, { key: "input-field-addition-before", fn: function() {
    return [e._t("input-field-addition-before")];
  }, proxy: !0 }, { key: "input-field-inline-before", fn: function() {
    return [n("div", { class: [e.dateFieldType && e.showPreInputIcon ? "base-search__spacing-date" : "base-search__spacing"] }), e._t("input-field-inline-before", function() {
      return [e.showPreInputIcon ? n("BaseIcon", { class: [
        "base-search__magnifier-icon",
        { "base-search__magnifier-icon__date": !!e.dateFieldType },
        { "base-search__magnifier-icon__active": e.isActiveInt }
      ], attrs: { name: "magnifier" } }) : e._e()];
    })];
  }, proxy: !0 }, { key: "input-field-addition-after", fn: function() {
    return [e._t("input-field-addition-after")];
  }, proxy: !0 }, { key: "post-input-field", fn: function() {
    return [e._t("post-input-field"), n("div", { class: { "base-search__spacing": e.dateFieldType } })];
  }, proxy: !0 }, { key: "error-icon", fn: function() {
    return [e._t("error-icon")];
  }, proxy: !0 }, { key: "remove-icon", fn: function() {
    return [e._t("remove-icon")];
  }, proxy: !0 }, { key: "below-input", fn: function() {
    return [e._t("below-input")];
  }, proxy: !0 }], null, !0), model: { value: e.inputInt, callback: function(i) {
    e.inputInt = i;
  }, expression: "inputInt" } }, e.$listeners))], 1);
}, l = [], d = /* @__PURE__ */ a(
  r,
  p,
  l,
  !1,
  null,
  "635e5927",
  null,
  null
);
const h = d.exports;
export {
  h as default
};
