// base-ui-components@3.1.2/dist/components/BaseSwitchButton/BaseSwitchButton.es.js
import "./BaseSwitchButton.vue_vue_type_style_index_0_scoped_b5ca66e6_lang.es.js";
import i from "../../_virtual/_plugin-vue2_normalizer.es.js";
const r = {
  name: "BaseSwitchButton",
  model: {
    prop: "activeTab",
    event: "switch"
  },
  props: {
    options: {
      type: Array,
      default: () => [{ label: "tab", value: "tab" }]
    },
    activeTab: {
      type: String,
      default() {
        return this.options[0] ? this.options[0].value : "tab";
      }
    },
    label: {
      type: String,
      required: !0
    }
  },
  data() {
    return {
      selectedOption: this.activeTab
    };
  },
  computed: {
    optionIds() {
      return this.options.map((a) => this.generateId(a.value));
    }
  },
  watch: {
    selectedOption(a) {
      this.$emit("switch", a);
    },
    activeTab(a) {
      a !== this.selectedOption && (this.selectedOption = a);
    }
  },
  methods: {
    generateId(a) {
      return `${a}${Math.floor(Math.random() * 1e6).toString()}`;
    }
  }
};
var c = function() {
  var e = this, l = e._self._c;
  return l("fieldset", { staticClass: "base-switch-buttons" }, [l("legend", { staticClass: "hide" }, [e._v(" " + e._s(e.label) + " ")]), e._l(e.options, function(t, n) {
    return [l("label", { key: t.value + "label", class: [
      "base-switch-button__label",
      { "base-switch-button__label__active": t.value === e.selectedOption }
    ], attrs: { for: e.optionIds[n] } }, [l("input", { directives: [{ name: "model", rawName: "v-model", value: e.selectedOption, expression: "selectedOption" }], key: t.value + "input", class: ["hide", "base-switch-button__input"], attrs: { id: e.optionIds[n], tabindex: t.value === e.selectedOption ? 0 : -1, "aria-checked": t.value === e.selectedOption, name: e.label, type: "radio" }, domProps: { checked: t.value === e.selectedOption, value: t.value, checked: e._q(e.selectedOption, t.value) }, on: { keydown: function(s) {
      if (!s.type.indexOf("key") && e._k(s.keyCode, "enter", 13, s.key, "Enter"))
        return null;
      s.preventDefault();
    }, change: function(s) {
      e.selectedOption = t.value;
    } } }), e._v(" " + e._s(t.label) + " "), e._t("right-of-text", null, { value: t.value })], 2)];
  })], 2);
}, o = [], u = /* @__PURE__ */ i(
  r,
  c,
  o,
  !1,
  null,
  "b5ca66e6",
  null,
  null
);
const _ = u.exports;
export {
  _ as default
};
