// base-ui-components@3.1.2/dist/components/BasePopUp/BasePopUp.es.js
import s from "../BaseIcon/BaseIcon.es.js";
import i from "../../mixins/popUpLock.es.js";
import "./BasePopUp.vue_vue_type_style_index_0_scoped_37dab2a6_lang.es.js";
import n from "../../_virtual/_plugin-vue2_normalizer.es.js";
const a = {
  name: "BasePopUp",
  components: {
    BaseIcon: s,
    BaseButton: () => import("../BaseButton/BaseButton.es.js").then((o) => o.default || o),
    BaseLoader: () => import("../BaseLoader/BaseLoader.es.js").then((o) => o.default || o)
  },
  mixins: [i],
  props: {
    show: {
      type: Boolean,
      default: !1
    },
    title: {
      type: String,
      default: "Pop Up"
    },
    descriptionElementId: {
      type: String,
      default: ""
    },
    buttonLeftText: {
      type: String,
      default: "Cancel"
    },
    buttonRightText: {
      type: String,
      default: "Submit"
    },
    buttonLeftIcon: {
      type: String,
      default: "remove"
    },
    buttonRightIcon: {
      type: String,
      default: "check-mark"
    },
    buttonRightDisabled: {
      type: Boolean,
      default: !1
    },
    isLoading: {
      type: Boolean,
      default: !1
    },
    isOpenFocus: {
      type: String,
      default: ""
    },
    fullscreenOnMobile: {
      type: Boolean,
      default: !1
    }
  },
  data() {
    return {
      showInt: this.show,
      targetName: "popUpBody"
    };
  },
  watch: {
    show(o) {
      !this.showInt && !this.prevActiveElement && (this.prevActiveElement = document.activeElement), this.showInt = o;
    }
  },
  updated() {
    setTimeout(() => {
      this.showInt ? this.isOpenFocus !== "" && this.$el.querySelector(this.isOpenFocus) && this.$el.querySelector(this.isOpenFocus).focus() : this.prevActiveElement && (this.prevActiveElement.focus(), this.prevActiveElement = !1);
    }, 250);
  },
  mounted() {
    document.onkeyup = (o) => {
      const t = o || window.event;
      document.querySelector(".popup-box") && (t.keyCode === 27 || t.key === "Escape") && document.querySelector(".popup-box .base-popup__close-button").dispatchEvent(new Event("click"));
    };
  },
  methods: {
    close() {
      this.$emit("close"), this.showInt = !1;
    },
    buttonRight() {
      this.$emit("button-right");
    },
    buttonLeft() {
      this.$emit("button-left");
    }
  }
};
var p = function() {
  var t = this, e = t._self._c;
  return t.showInt ? e("div", { class: [
    "base-pop-up",
    { "base-pop-up--fullscreen-on-mobile": t.fullscreenOnMobile }
  ] }, [e("div", { staticClass: "base-pop-up-background" }), e("div", { ref: "popUpBody", staticClass: "popup-box", attrs: { role: "alertdialog", "aria-modal": "true", "aria-labelledby": "popup-title", "aria-describedby": "descriptionElementId" } }, [e("div", { staticClass: "popup-header" }, [e("div", { staticClass: "popup-title", attrs: { id: "popup-title" } }, [t._v(" " + t._s(t.title) + " ")]), e("button", { staticClass: "base-popup__close-button", attrs: { type: "button", "aria-label": "close pop up" }, on: { click: t.close } }, [e("base-icon", { staticClass: "popup-remove", attrs: { name: "remove" } })], 1)]), e("div", { staticClass: "popup-content" }, [t._t("default"), e("div", { staticClass: "popup-button-row" }, [t._t("button-row", function() {
    return [e("BaseButton", { staticClass: "base-popup-button", attrs: { id: "popup-left-button", text: t.buttonLeftText, icon: t.buttonLeftIcon, "icon-position": "right", "icon-size": "small" }, on: { clicked: t.buttonLeft } }), e("BaseButton", { staticClass: "base-popup-button", attrs: { id: "popup-right-button", text: t.buttonRightText, icon: t.isLoading ? "" : t.buttonRightIcon, "icon-position": "right", "icon-size": "small", disabled: t.buttonRightDisabled }, on: { clicked: t.buttonRight }, scopedSlots: t._u([{ key: "right-of-text", fn: function() {
      return [e("span", { directives: [{ name: "show", rawName: "v-show", value: t.isLoading, expression: "isLoading" }], staticClass: "base-popup-button-loader" }, [e("BaseLoader")], 1)];
    }, proxy: !0 }], null, !1, 288902964) })];
  })], 2)], 2)])]) : t._e();
}, l = [], u = /* @__PURE__ */ n(
  a,
  p,
  l,
  !1,
  null,
  "37dab2a6",
  null,
  null
);
const f = u.exports;
export {
  f as default
};
