// base-ui-components/dist/components/BaseInput/BaseInput.es.js
import d from "vue-click-outside";
import o from "../BaseIcon/BaseIcon.es.js";
import { createId as p } from "../../utils/utils.es.js";
import h from "../../mixins/i18n.es.js";
import "./BaseInput.vue_vue_type_style_index_0_scoped_00dfa997_lang.es.js";
import c from "../../_virtual/_plugin-vue2_normalizer.es.js";
const m = {
  name: "BaseInput",
  directives: {
    ClickOutside: d
  },
  components: {
    BaseIcon: o,
    BaseLoader: () => import("../BaseLoader/BaseLoader.es.js").then((e) => e.default || e)
  },
  mixins: [h],
  model: {
    prop: "input",
    event: "input"
  },
  props: {
    input: {
      type: [String, Number],
      default: ""
    },
    label: {
      type: String,
      required: !0
    },
    showLabel: {
      type: Boolean,
      default: !0
    },
    id: {
      type: String,
      default: ""
    },
    placeholder: {
      type: String,
      default: "Enter Text Here"
    },
    required: {
      type: Boolean,
      default: !1
    },
    invalid: {
      type: Boolean,
      default: !1
    },
    errorMessage: {
      type: String,
      default: ""
    },
    showErrorIcon: {
      type: Boolean,
      default: !0
    },
    fieldType: {
      type: String,
      default: "text",
      validator: (e) => ["text", "number", "password", "email", "url", "search"].includes(e)
    },
    dropDownListId: {
      type: String,
      default: ""
    },
    linkedListOption: {
      type: String,
      default: null
    },
    hideInputField: {
      type: Boolean,
      default: !1
    },
    isActive: {
      type: Boolean,
      default: null
    },
    showInputBorder: {
      type: Boolean,
      default: !0
    },
    useFormFieldStyling: {
      type: Boolean,
      default: !0
    },
    clearable: {
      type: Boolean,
      default: !1
    },
    loadable: {
      type: Boolean,
      default: !1
    },
    isLoading: {
      type: Boolean,
      default: !1
    },
    useFadeOut: {
      type: Boolean,
      default: !0
    },
    inputClass: {
      type: String,
      default: ""
    },
    language: {
      type: String,
      default: "",
      validator: (e) => !e || e.length === 2
    },
    setFocusOnActive: {
      type: Boolean,
      default: !0
    },
    disabled: {
      type: Boolean,
      default: !1
    },
    minLength: {
      type: Number,
      default: null
    },
    maxLength: {
      type: Number,
      default: null
    },
    decimals: {
      type: Number,
      default: null
    },
    decimalSeparator: {
      type: String,
      default: "."
    },
    min: {
      type: Number,
      default: null
    },
    max: {
      type: Number,
      default: null
    },
    validationTexts: {
      type: Object,
      default: () => ({
        min: "Value must be greater than or equal to {value}.",
        max: "Value must be less than or equal to {value}.",
        minLength: "Text must be at least {value} character(s) long.",
        maxLength: "Text cannot be longer than {value} characters."
      }),
      validator: (e) => ["min", "max", "minLength", "maxLength"].every((t) => Object.keys(e).includes(t))
    }
  },
  data() {
    return {
      isActiveInt: !1,
      inputInt: "",
      previousInput: "",
      errorMessageInt: "",
      invalidInt: ""
    };
  },
  computed: {
    idInt() {
      return this.id || p();
    },
    showLabelRow() {
      const e = this.$slots["label-addition"], t = !!e && !!e.length && e.some((i) => i.tag);
      return this.showLabel || t;
    },
    showRemoveIcon() {
      return this.clearable && !!this.inputInt;
    },
    inputListeners() {
      return {
        ...this.$listeners,
        input: (e) => {
          let { value: t } = e.target;
          if (this.errorMessageInt = "", this.invalidInt = !1, this.fieldType === "number") {
            const i = this.decimals ? `\\${this.decimalSeparator}` : "", n = new RegExp(`[^e0-9${i}\\+-]`, "g"), a = this.decimals && this.decimals !== Number("-1") ? `{0,${this.decimals}}` : "*", s = this.allowNegativeNumber ? "-?" : "", u = this.decimals && this.decimals !== Number("-1") ? "-" : "", r = new RegExp(`^((${s}[0-9]*${i}?([0-9]${a})(e(${u}|\\+)?[0-9]*)?)|-)$`, "g");
            if (t = t.replace(",", this.decimalSeparator), t = t.replace(".", this.decimalSeparator), t = t.replace(n, ""), t = this.removeMultipleChars(t, this.decimalSeparator), t = this.removeMultipleChars(t, "e"), this.inputInt = t, this.decimals && !Number.isNaN(Number(t)) && (t = this.roundDecimals(t), this.inputInt = t), t !== "" && !t.match(r)) {
              this.inputInt = this.previousInput;
              return;
            }
            if (this.min && t && Number(this.stringToFloat(t)) < this.min) {
              this.errorMessageInt = this.validationTexts.min.replace("{value}", this.min.toString()), this.invalidInt = !0;
              return;
            }
            if (this.max && Number(this.stringToFloat(t)) > this.max) {
              this.errorMessageInt = this.validationTexts.max.replace("{value}", this.max.toString()), this.invalidInt = !0;
              return;
            }
            if (Number(this.stringToFloat(t)) === 1 / 0 && (t = 0, this.inputInt = t), this.inputIsNaN(t))
              return;
            this.previousInput = t, t = this.stringToFloat(t);
          }
          if (this.fieldType !== "number") {
            if (this.minLength && t && t.length < this.minLength) {
              this.errorMessageInt = this.validationTexts.minLength.replace("{value}", this.minLength.toString()), this.invalidInt = !0;
              return;
            }
            if (this.maxLength && t.length > this.maxLength) {
              this.errorMessageInt = this.validationTexts.maxLength.replace("{value}", this.maxLength.toString()), this.invalidInt = !0;
              return;
            }
          }
          this.$emit("input", t);
        },
        blur: (e) => {
          const { value: t } = e.target;
          if (this.fieldType === "number") {
            if (t === "" || Number.isNaN(Number(this.stringToFloat(t)))) {
              this.inputInt = "", this.previousInput = "";
              return;
            }
            this.inputInt = this.translateFloat(Number(this.stringToFloat(t))), this.previousInput = this.inputInt;
          }
        }
      };
    },
    inputElement() {
      if (window) {
        if (this.$refs && this.$refs.input)
          return this.$refs.input;
        const e = document.getElementById(this.idInt);
        if (e)
          return e;
        console.warn("BaseInput: you did not assign the same id to the BaseInputComponent and the input element!");
      }
      return null;
    },
    allowNegativeNumber() {
      return this.min === null || this.min < 0;
    }
  },
  watch: {
    input: {
      handler(e) {
        const t = this.fieldType === "number" ? this.translateFloat(e) : e;
        t !== this.inputInt && (this.inputInt = t, this.previousInput = t, this.triggerInputEvent());
      },
      immediate: !0
    },
    inputInt(e) {
      let t = e;
      if (this.fieldType === "number") {
        if (this.inputIsNaN(t))
          return;
        t = this.stringToFloat(t);
      }
      t !== this.input && this.$emit("input", t);
    },
    isActive(e) {
      e !== this.isActiveInt && (this.isActiveInt = e);
    },
    isActiveInt(e) {
      this.inputElement && e && this.setFocusOnActive && this.inputElement.focus(), e !== this.isActive && this.$emit("update:is-active", e);
    },
    errorMessage(e) {
      e !== this.errorMessageInt && (this.errorMessageInt = e);
    },
    invalid(e) {
      e !== this.invalidInt && (this.invalidInt = e);
    },
    invalidInt(e) {
      e !== this.invalid && this.$emit("update:invalid", e);
    }
  },
  mounted() {
    this.max && Number(this.stringToFloat(this.input)) > this.max && (this.inputInt = this.max);
  },
  methods: {
    blurInput(e) {
      this.setFieldState(!1), this.$emit("keydown", e);
    },
    clickedInside(e) {
      this.disabled || (this.setFieldState(!0), this.$emit("click-input-field", e));
    },
    clickedOutsideInput(e) {
      this.setFieldState(!1), this.$emit("clicked-outside", e);
    },
    removeInput() {
      this.inputInt = "", this.inputElement && this.inputElement.focus();
    },
    setFieldState(e) {
      this.isActiveInt = e;
    },
    handleInputTab(e) {
      (!this.showRemoveIcon || e.shiftKey) && this.setFieldState(!1), this.$emit("keydown", e);
    },
    translateFloat(e) {
      return e == null ? "" : e.toString().replace(".", this.decimalSeparator);
    },
    stringToFloat(e) {
      return e ? Number(e.toString().replace(this.decimalSeparator, ".")) : "";
    },
    inputIsNaN(e) {
      const t = this.decimalSeparator, i = new RegExp(`^(-0|-?(([0-9]*\\${t}([0]*)?)?|(\\${t}([0-9]*)?)?))$`, "g");
      return e.length && (e.toString().match(i) || Number.isNaN(this.stringToFloat(e)));
    },
    triggerInputEvent() {
      setTimeout(() => {
        this.$refs.input && this.$refs.input.dispatchEvent(new Event("input"));
      }, 0);
    },
    roundDecimals(e) {
      const t = e.split(this.decimalSeparator);
      return t[1] && t[1].length > this.decimals ? this.translateFloat(Number(Number(e).toFixed(this.decimals))) : e;
    },
    removeMultipleChars(e, t) {
      const i = new RegExp(`\\${t}`, "g");
      return e.replace(i, (n, a, s) => s.indexOf(n) === a ? n : "");
    }
  }
};
var f = function() {
  var t = this, i = t._self._c;
  return i("div", { staticClass: "base-input" }, [i("div", { class: ["base-input__label-row", { hide: !t.showLabelRow }] }, [i("label", { class: ["base-input__label", { hide: !t.showLabel }], attrs: { for: t.idInt } }, [t._v(" " + t._s(t.getLangLabel(t.label)) + " ")]), i("div", { staticClass: "base-input__label-spacer" }), t._t("label-addition")], 2), i("div", { directives: [{ name: "click-outside", rawName: "v-click-outside", value: t.clickedOutsideInput, expression: "clickedOutsideInput" }], class: [
    "base-input__input-frame",
    { "base-input__input-frame__border": t.showInputBorder },
    { "base-input__input-frame__disabled": t.disabled },
    { "base-input__input-frame__invalid": t.invalidInt }
  ], on: { focusin: t.clickedInside, click: t.clickedInside } }, [i("div", { class: [
    "base-input__input-container",
    {
      "base-input__input-container__is-active": t.isActiveInt && t.useFormFieldStyling
    }
  ] }, [t._t("pre-input-field"), i("div", { class: [
    "base-input__input-line-container",
    { "base-input__input-line-container__wrap": !t.hideInputField }
  ] }, [t._t("input-field-addition-before"), i("div", { staticClass: "base-input__input-line" }, [t._t("input-field-inline-before"), i("div", { class: [
    "base-input__input-wrapper",
    {
      "base-input__input-wrapper__fade-out": t.useFadeOut && !t.isActiveInt && !t.hideInputField
    }
  ] }, [t._t("input", function() {
    return [(t.fieldType === "number" ? "text" : t.fieldType) === "checkbox" ? i("input", t._g({ directives: [{ name: "model", rawName: "v-model", value: t.inputInt, expression: "inputInt" }], ref: "input", class: [
      t.inputClass,
      "base-input__input",
      { "base-input__input__hidden": t.hideInputField }
    ], attrs: { id: t.idInt, placeholder: t.placeholder, list: t.dropDownListId || !1, disabled: t.disabled, "aria-disabled": t.disabled.toString(), "aria-activedescendant": t.linkedListOption, "aria-describedby": t.idInt, "aria-required": t.required.toString(), required: t.required, "aria-invalid": t.invalidInt.toString(), minlength: t.minLength, maxlength: t.maxLength, inputmode: t.fieldType === "number" && t.allowNegativeNumber ? "decimal" : null, autocomplete: "off", type: "checkbox" }, domProps: { checked: Array.isArray(t.inputInt) ? t._i(t.inputInt, null) > -1 : t.inputInt }, on: { keydown: function(n) {
      return !n.type.indexOf("key") && t._k(n.keyCode, "tab", 9, n.key, "Tab") ? null : t.handleInputTab.apply(null, arguments);
    }, change: function(n) {
      var a = t.inputInt, s = n.target, u = !!s.checked;
      if (Array.isArray(a)) {
        var r = null, l = t._i(a, r);
        s.checked ? l < 0 && (t.inputInt = a.concat([r])) : l > -1 && (t.inputInt = a.slice(0, l).concat(a.slice(l + 1)));
      } else
        t.inputInt = u;
    } } }, t.inputListeners)) : (t.fieldType === "number" ? "text" : t.fieldType) === "radio" ? i("input", t._g({ directives: [{ name: "model", rawName: "v-model", value: t.inputInt, expression: "inputInt" }], ref: "input", class: [
      t.inputClass,
      "base-input__input",
      { "base-input__input__hidden": t.hideInputField }
    ], attrs: { id: t.idInt, placeholder: t.placeholder, list: t.dropDownListId || !1, disabled: t.disabled, "aria-disabled": t.disabled.toString(), "aria-activedescendant": t.linkedListOption, "aria-describedby": t.idInt, "aria-required": t.required.toString(), required: t.required, "aria-invalid": t.invalidInt.toString(), minlength: t.minLength, maxlength: t.maxLength, inputmode: t.fieldType === "number" && t.allowNegativeNumber ? "decimal" : null, autocomplete: "off", type: "radio" }, domProps: { checked: t._q(t.inputInt, null) }, on: { keydown: function(n) {
      return !n.type.indexOf("key") && t._k(n.keyCode, "tab", 9, n.key, "Tab") ? null : t.handleInputTab.apply(null, arguments);
    }, change: function(n) {
      t.inputInt = null;
    } } }, t.inputListeners)) : i("input", t._g({ directives: [{ name: "model", rawName: "v-model", value: t.inputInt, expression: "inputInt" }], ref: "input", class: [
      t.inputClass,
      "base-input__input",
      { "base-input__input__hidden": t.hideInputField }
    ], attrs: { id: t.idInt, placeholder: t.placeholder, list: t.dropDownListId || !1, disabled: t.disabled, "aria-disabled": t.disabled.toString(), "aria-activedescendant": t.linkedListOption, "aria-describedby": t.idInt, "aria-required": t.required.toString(), required: t.required, "aria-invalid": t.invalidInt.toString(), minlength: t.minLength, maxlength: t.maxLength, inputmode: t.fieldType === "number" && t.allowNegativeNumber ? "decimal" : null, autocomplete: "off", type: t.fieldType === "number" ? "text" : t.fieldType }, domProps: { value: t.inputInt }, on: { keydown: function(n) {
      return !n.type.indexOf("key") && t._k(n.keyCode, "tab", 9, n.key, "Tab") ? null : t.handleInputTab.apply(null, arguments);
    }, input: function(n) {
      n.target.composing || (t.inputInt = n.target.value);
    } } }, t.inputListeners))];
  }, null, {
    id: t.idInt
  })], 2), t.showRemoveIcon ? i("button", { staticClass: "base-input__remove-icon-wrapper", on: { keydown: function(n) {
    return !n.type.indexOf("key") && t._k(n.keyCode, "tab", 9, n.key, "Tab") ? null : t.blurInput.apply(null, arguments);
  }, click: function(n) {
    return n.stopPropagation(), t.removeInput.apply(null, arguments);
  } } }, [t._t("remove-icon", function() {
    return [i("BaseIcon", { staticClass: "base-input__remove-icon", attrs: { name: "remove", title: "Clear input" } })];
  })], 2) : t._e(), t.loadable ? i("div", { staticClass: "base-input__loader" }, [i("BaseLoader", { attrs: { hide: !t.isLoading } })], 1) : t._e(), t._t("input-field-addition-after")], 2)], 2), t.showErrorIcon && t.invalidInt ? i("div", { staticClass: "base-input__error-icon-wrapper" }, [t._t("error-icon", function() {
    return [i("BaseIcon", { staticClass: "base-input__error-icon", attrs: { name: "attention" } })];
  })], 2) : t._e(), t._t("post-input-field")], 2)]), i("div", [t._t("below-input")], 2), t.invalidInt && t.errorMessageInt ? i("div", { staticClass: "base-input__invalid-message" }, [t._v(" " + t._s(t.getLangLabel(t.errorMessageInt)) + " ")]) : t._e()]);
}, b = [], g = /* @__PURE__ */ c(
  m,
  f,
  b,
  !1,
  null,
  "00dfa997",
  null,
  null
);
const N = g.exports;
export {
  N as default
};
