// base-ui-components@3.1.2/dist/components/BaseTextList/BaseTextList.es.js
import i from "../../mixins/i18n.es.js";
import "./BaseTextList.vue_vue_type_style_index_0_scoped_f121c850_lang.es.js";
import r from "../../_virtual/_plugin-vue2_normalizer.es.js";
const c = {
  name: "BaseTextList",
  components: {
    BaseLink: () => import("../BaseLink/BaseLink.es.js").then((s) => s.default || s)
  },
  mixins: [i],
  props: {
    data: {
      type: Array,
      default: () => []
    },
    renderLabelAs: {
      type: String,
      default: "div"
    },
    labelMarginBottom: {
      type: Boolean,
      default: !1
    },
    cols2: {
      type: Boolean,
      default: !1
    },
    autoHeight: {
      type: Boolean,
      default: !1
    },
    renderLinkAs: {
      type: String,
      default: "RouterLinnk"
    },
    listType: {
      type: String,
      default: "horizontal",
      validate: (s) => ["horizontal", "vertical"].includes(s)
    }
  },
  methods: {
    containKeys(s, t) {
      for (const l of s)
        if (!Object.keys(l).includes(t))
          return !1;
      return !0;
    },
    isChip(s) {
      return !!(s.source && s.type);
    }
  }
};
var d = function() {
  var t = this, l = t._self._c;
  return l("div", { class: [
    "base-text-list",
    {
      "base-text-list-2-cols": t.cols2,
      "base-text-list-2-cols-single-content": t.cols2 && t.data.length === 1 && typeof t.data[0].data == "string"
    }
  ] }, t._l(t.data, function(e, o) {
    return l("div", { key: o, staticClass: "base-text-list-group" }, [e.label ? l(t.renderLabelAs, { tag: "component", class: ["base-text-list-label", { "base-text-list-label-mb": t.labelMarginBottom }] }, [t._v(" " + t._s(t.getLangLabel(e.label)) + " ")]) : t._e(), typeof e.data == "string" ? l("p", { class: [
      "base-text-list-content",
      "base-text-list-content-pre-line",
      { "base-text-list-2-cols": t.data.length === 1 }
    ] }, [t._v(t._s(e.data))]) : typeof e.data == "object" && typeof e.data[0] == "string" ? l("ul", { staticClass: "base-text-list-content" }, t._l(e.data, function(a, n) {
      return l("li", { key: n }, [t._v(" " + t._s(a) + " ")]);
    }), 0) : typeof e.data == "object" ? [t.containKeys([].concat(e.data), "label") ? t._e() : [l("div", { staticClass: "base-text-list-content" }, [t._l([].concat(e.data), function(a, n) {
      return [l("BaseLink", { key: n, class: [{ "base-link--chip-text-list": t.isChip(a) }], attrs: { source: a.source, tooltip: a.additional, type: a.type, url: a.url, value: a.value }, scopedSlots: t._u([{ key: "tooltip", fn: function() {
        return [t._t("tooltip", null, { data: a.additional })];
      }, proxy: !0 }], null, !0) }), t._v(t._s(e.data.length && n !== e.data.length - 1 && !t.isChip(a) ? ", " : "") + " ")];
    })], 2)], t.containKeys([].concat(e.data), "label") ? [typeof e.data == "object" ? l("dl", { class: [
      "base-text-list-content",
      "base-text-list-content--" + t.listType
    ] }, [t._l([].concat(e.data), function(a, n) {
      return [l("dt", { key: "l" + n, staticClass: "base-text-list__content__label" }, [a.label ? [t._v(" " + t._s(t.getLangLabel(a.label)) + ": ")] : t._e()], 2), l("dd", { key: "v" + n, staticClass: "base-text-list__content__label base-text-list__content__value" }, [l("BaseLink", { attrs: { "render-link-as": t.renderLinkAs, source: a.source, tooltip: a.additional, type: a.type, url: a.url, value: a.value } }, [t._t("tooltip")], 2)], 1)];
    })], 2) : t._e()] : t._e()] : t._e()], 2);
  }), 0);
}, u = [], _ = /* @__PURE__ */ r(
  c,
  d,
  u,
  !1,
  null,
  "f121c850",
  null,
  null
);
const b = _.exports;
export {
  b as default
};
