// base-ui-components@3.1.2/dist/components/BaseOptions/BaseOptions.es.js
import { debounce as r } from "../../utils/utils.es.js";
import a from "../BaseButton/BaseButton.es.js";
import h from "../../mixins/i18n.es.js";
import "./BaseOptions.vue_vue_type_style_index_0_scoped_f7cd57a9_lang.es.js";
import p from "../../_virtual/_plugin-vue2_normalizer.es.js";
const l = {
  name: "BaseOptions",
  components: {
    BaseButton: a
  },
  mixins: [h],
  props: {
    showOptions: {
      type: Boolean,
      default: !1
    },
    optionsHidden: {
      type: Boolean,
      default: !1
    },
    alignOptions: {
      type: String,
      default: "right",
      validator: (i) => ["left", "right"].includes(i)
    },
    useOptionsButtonOn: {
      type: String,
      default: "always",
      validator: (i) => ["always", "mobile", "never"].includes(i)
    },
    optionsButtonText: {
      type: Object,
      default: () => ({
        show: "options",
        hide: "return"
      }),
      validator: (i) => "show" in i && "hide" in i
    },
    optionsButtonIcon: {
      type: Object,
      default: () => ({
        show: "options-menu",
        hide: "remove"
      }),
      validator: (i) => "show" in i && "hide" in i
    },
    showAfterOptionsBelow: {
      type: Boolean,
      default: !0
    },
    optionsConfig: {
      type: Array,
      default: () => [{
        text: "delete",
        icon: "waste-bin",
        value: "delete"
      }],
      validator: (i) => i.every((t) => {
        const s = ["text", "icon", "value"], o = Object.keys(t);
        return s.every((e) => o.includes(e));
      })
    }
  },
  data() {
    return {
      showOptionsInt: !1,
      isMobile: !0,
      rowWidth: 4e3,
      beforeOptionsWidth: 0,
      afterOptionsWidth: 0,
      optionsButtonWidth: 0,
      remainingActionsWidth: 4e3,
      actionButtonsWidth: 0,
      timeout: null,
      resizeObserver: null,
      mutationObserver: null
    };
  },
  computed: {
    useOptionsButton() {
      return this.useOptionsButtonOn === "always" || this.useOptionsButtonOn === "mobile" && this.isMobile;
    },
    showOptionsInline() {
      return this.remainingActionsWidth - this.actionButtonsWidth > 0 || !this.useOptionsButton;
    },
    showAfterOptionsInline() {
      return this.afterSlotHasData && (!this.useOptionsButton && this.remainingActionsWidth > this.actionButtonsWidth || this.useOptionsButton && (!this.showAfterOptionsBelow || this.remainingActionsWidth > 0));
    },
    wrapActions() {
      return this.rowWidth <= this.actionButtonsWidth;
    },
    wrapHeaderActions() {
      return this.rowWidth - this.beforeOptionsWidth < this.actionButtonsWidth;
    },
    beforeSlotHasData() {
      return !!this.$slots.beforeOptions;
    },
    afterSlotHasData() {
      return !!this.$slots.afterOptions;
    }
  },
  watch: {
    showOptionsInt(i) {
      this.showOptions !== i && this.$emit("update:show-options", i);
    },
    showOptions: {
      handler(i) {
        this.showOptionsInt !== i && this.useOptionsButton && (this.showOptionsInt = i);
      },
      immediate: !0
    },
    useOptionsButton: {
      handler(i) {
        i || (this.showOptionsInt = !0);
      },
      immediate: !0
    },
    isMobile(i) {
      this.useOptionsButtonOn === "mobile" && i && (this.showOptionsInt = !1);
    }
  },
  mounted() {
    this.initObserver();
  },
  updated() {
    this.calcFixedElementWidth(), this.showOptionsInt && this.$refs.actions && this.calcOptionsWidth();
  },
  beforeDestroy() {
    this.resizeObserver && this.resizeObserver.unobserve(this.$refs.optionsRow), this.mutationObserver && this.afterSlotHasData && this.mutationObserver.disconnect();
  },
  methods: {
    initObserver() {
      const i = new ResizeObserver(r(50, this.resizeActions));
      if (i.observe(this.$refs.optionsRow), this.resizeObserver = i, this.afterSlotHasData) {
        const t = new MutationObserver(this.calcOptionsWidth);
        t.observe(this.$refs.afterOptions, {
          childList: !0,
          subtree: !0
        }), this.mutationObserver = t;
      }
    },
    optionTriggered(i) {
      this.$emit("option-triggered", i);
    },
    resizeActions() {
      this.timeout && (clearTimeout(this.timeout), this.timeout = null), this.timeout = setTimeout(() => {
        this.calcIsMobile();
      }, 200), this.calcOptionsWidth();
    },
    calcIsMobile() {
      this.isMobile = window.innerWidth < 640;
    },
    calcOptionsWidth() {
      const { optionsRow: i } = this.$refs;
      i && (this.rowWidth = i.clientWidth), this.remainingActionsWidth = i ? this.rowWidth - this.beforeOptionsWidth - this.afterOptionsWidth - this.optionsButtonWidth : 0;
      const t = this.$refs.actions;
      t && (this.actionButtonsWidth = t.clientWidth);
    },
    calcFixedElementWidth() {
      this.beforeSlotHasData && this.$refs.beforeOptions && (this.beforeOptionsWidth = this.$refs.beforeOptions.clientWidth), this.afterSlotHasData && this.$refs.afterOptions && (this.afterOptionsWidth = this.$refs.afterOptions.clientWidth);
      const i = this.$refs.optionsButton;
      this.useOptionsButton && i && (this.optionsButtonWidth = i.$el.clientWidth);
    }
  }
};
var u = function() {
  var t = this, s = t._self._c;
  return s("div", { staticClass: "base-options" }, [s("div", { ref: "optionsRow", class: [
    "base-options__row",
    `base-options__row-${t.alignOptions}`,
    {
      "base-options__row-wrap": !t.showOptionsInline || !t.useOptionsButton && t.remainingActionsWidth < t.actionButtonsWidth
    }
  ] }, [t.beforeSlotHasData ? s("div", { ref: "beforeOptions", staticClass: "base-options__before" }, [t._t("beforeOptions")], 2) : t._e(), t.alignOptions === "right" ? s("div", { staticClass: "base-options__spacer" }) : t._e(), !t.optionsHidden && t.showOptionsInline && t.showOptionsInt ? s("div", { ref: "actions", class: [
    "base-options__options-inline",
    { "base-options__options-inline-wrap": t.wrapActions },
    { "base-options__options-inline-left": t.alignOptions === "left" }
  ] }, [t._t("options", function() {
    return [t.optionsConfig.length ? t._l(t.optionsConfig, function(o, e) {
      return s("BaseButton", { key: o.text + "_" + e, attrs: { text: o.text, icon: o.icon, "has-background-color": !1, "icon-size": "large", "button-style": "single" }, on: { clicked: function(n) {
        return t.optionTriggered(o.value);
      } } });
    }) : t._e()];
  })], 2) : t._e(), !t.optionsHidden && t.useOptionsButton ? s("BaseButton", { ref: "optionsButton", class: [{ "base-options__options-button-left": t.alignOptions === "left" }], attrs: { text: t.showOptionsInt ? t.getI18nTerm(t.optionsButtonText.hide) : t.getI18nTerm(t.optionsButtonText.show), icon: t.showOptionsInt ? t.optionsButtonIcon.hide : t.optionsButtonIcon.show }, on: { clicked: function(o) {
    t.showOptionsInt = !t.showOptionsInt;
  } } }) : t._e(), t.alignOptions === "left" ? s("div", { staticClass: "base-options__spacer base-options__spacer-left" }) : t._e(), t.showAfterOptionsInline && t.afterSlotHasData ? s("div", { ref: "afterOptions", staticClass: "base-options__after-inline" }, [t._t("afterOptions")], 2) : t._e()], 1), s("transition", { attrs: { name: "slide-fade-options" } }, [!t.optionsHidden && t.showOptionsInt && !t.showOptionsInline ? s("div", { staticClass: "base-options__below" }, [t._t("options", function() {
    return [t.optionsConfig.length ? t._l(t.optionsConfig, function(o, e) {
      return s("BaseButton", { key: o.text + "_" + e, attrs: { text: o.text, icon: o.icon, "has-background-color": !1, "icon-size": "large", "button-style": "single" }, on: { clicked: function(n) {
        return t.optionTriggered(o.value);
      } } });
    }) : t._e()];
  })], 2) : t._e()]), !t.showAfterOptionsInline && t.afterSlotHasData ? s("div", { staticClass: "base-options__after" }, [t._t("afterOptions")], 2) : t._e()], 1);
}, d = [], c = /* @__PURE__ */ p(
  l,
  u,
  d,
  !1,
  null,
  "f7cd57a9",
  null,
  null
);
const m = c.exports;
export {
  m as default
};
