// base-ui-components@3.1.2/dist/components/BaseBoxButton/BaseBoxButton.es.js
import s from "../BaseIcon/BaseIcon.es.js";
import a from "../BaseBox/BaseBox.es.js";
import i from "../BaseBoxTooltip/BaseBoxTooltip.es.js";
import "./BaseBoxButton.vue_vue_type_style_index_0_scoped_73af657a_lang.es.js";
import n from "../../_virtual/_plugin-vue2_normalizer.es.js";
const l = {
  name: "BaseBoxButton",
  components: {
    BaseIcon: s,
    BaseBox: a,
    BaseBoxTooltip: i
  },
  props: {
    renderElementAs: {
      type: String,
      default: "div"
    },
    boxStyle: {
      type: String,
      default: "large",
      validator(o) {
        return ["large", "small"].includes(o);
      }
    },
    icon: {
      type: String,
      default: null
    },
    showPlus: {
      type: Boolean,
      default: !1
    },
    text: {
      type: String,
      default: ""
    },
    subtext: {
      type: String,
      default: ""
    },
    boxSize: {
      type: Object,
      default: () => ({ width: "25%" })
    },
    boxRatio: {
      type: String,
      default: "100"
    },
    disabled: {
      type: Boolean,
      default: !1
    },
    showTooltip: {
      type: Boolean,
      default: !1
    }
  },
  data() {
    return {
      showSubtext: !0
    };
  },
  mounted() {
    this.setShowSubtext(), window.addEventListener("resize", () => {
      this.setShowSubtext();
    });
  },
  methods: {
    clicked(o) {
      this.$emit("clicked", o);
    },
    onTooltip(o) {
      this.$emit("on-tooltip", o);
    },
    setShowSubtext() {
      const o = this.$refs.baseBoxSubtext;
      if (o) {
        const t = o.offsetTop + o.offsetHeight, e = this.$refs.baseBox.$el.offsetHeight;
        this.showSubtext = e >= t;
      }
    }
  }
};
var b = function() {
  var t = this, e = t._self._c;
  return e("base-box", { ref: "baseBox", staticClass: "base-box-button", attrs: { "render-element-as": t.renderElementAs, "box-size": t.boxSize, "box-ratio": t.boxRatio, disabled: t.disabled }, on: { clicked: t.clicked } }, [t.boxStyle === "large" ? e("div", { staticClass: "button-box-content" }, [e("div", { staticClass: "button-box-center" }, [e("div", { staticClass: "button-box-image-row" }, [e("div", { staticClass: "button-box-plus-container" }, [t.showPlus ? e("base-icon", { staticClass: "button-box-plus", attrs: { name: "plus", alt: "add" } }) : t._e()], 1), e("div", { staticClass: "button-box-icon-container" }, [t.icon ? e("base-icon", { staticClass: "button-box-icon", attrs: { name: t.icon } }) : t._e()], 1)]), e("div", { staticClass: "button-box-text" }, [t._v(" " + t._s(t.text) + " ")])]), e("div", { ref: "baseBoxSubtext", class: ["button-box-subtext", { "base-button-box__subtext-hidden": !t.showSubtext }] }, [t._v(" " + t._s(t.subtext) + " ")]), t.showTooltip ? e("BaseBoxTooltip", { on: { clicked: t.onTooltip } }) : t._e()], 1) : e("div", { staticClass: "button-box-content-small" }, [t.icon ? e("base-icon", { staticClass: "button-box-icon-small", attrs: { name: t.icon } }) : t._e(), e("div", { staticClass: "button-box-text-small" }, [t._v(" " + t._s(t.text) + " ")])], 1), t._t("default")], 2);
}, r = [], u = /* @__PURE__ */ n(
  l,
  b,
  r,
  !1,
  null,
  "73af657a",
  null,
  null
);
const m = u.exports;
export {
  m as default
};
