// base-ui-components@3.1.2/dist/components/BaseLink/BaseLink.es.js
import o from "vue-click-outside";
import "./BaseLink.vue_vue_type_style_index_0_scoped_22bc0e40_lang.es.js";
import a from "../../_virtual/_plugin-vue2_normalizer.es.js";
const r = {
  name: "BaseLink",
  components: {
    BaseIcon: () => import("../BaseIcon/BaseIcon.es.js").then((e) => e.default || e),
    BaseTooltip: () => import("../BaseTooltip/BaseTooltip.es.js").then((e) => e.default || e),
    BaseLoader: () => import("../BaseLoader/BaseLoader.es.js").then((e) => e.default || e)
  },
  directives: {
    ClickOutside: o
  },
  props: {
    renderLinkAs: {
      type: String,
      default: "RouterLink"
    },
    externalLinkTarget: {
      type: String,
      default: "_blank",
      validator: (e) => e === "_blank" || e === "_self"
    },
    tooltip: {
      type: Array,
      default: () => []
    },
    tooltipAsync: {
      type: Array,
      default: () => []
    },
    tooltipStyles: {
      type: Object,
      default: () => ({})
    },
    source: {
      type: String,
      default: ""
    },
    url: {
      type: String,
      default: ""
    },
    value: {
      type: String,
      default: ""
    },
    type: {
      type: String,
      default: ""
    },
    spaceAfter: {
      type: Boolean,
      default: !1
    }
  },
  data() {
    return {
      isLoading: !1,
      scrollResizeTimeout: null,
      showTooltip: !1
    };
  },
  computed: {
    vueRouterAvailable() {
      return !!this.$router;
    },
    renderAs() {
      let e = "span";
      return this.hasValidUrl && (e = "a"), this.source && (e = this.vueRouterAvailable ? this.renderLinkAs : "a"), this.source && this.type && (e = "button"), e;
    },
    hasValidUrl() {
      return this.url && (this.url.match(/^([a-z][a-z0-9+\-.]*:\/\/)/) || this.url.match(/^mailto:/) || this.url.match(/^tel:/));
    }
  },
  watch: {
    tooltip(e) {
      e && (this.isLoading = !1, this.showTooltip = !0);
    }
  },
  mounted() {
    window.addEventListener("scroll", this.scrollResize), window.addEventListener("resize", this.scrollResize);
  },
  destroyed() {
    window.removeEventListener("scroll", this.scrollResize), window.removeEventListener("resize", this.scrollResize);
  },
  methods: {
    chipClicked() {
      this.$emit("chip-clicked", { source: this.source, type: this.type });
    },
    async tooltipClicked() {
      if (this.tooltip.length) {
        this.showTooltip = !this.showTooltip;
        return;
      }
      this.tooltipAsync.length && (this.isLoading = !0, this.$emit("tooltip-clicked", this.tooltipAsync));
    },
    closeTooltip() {
      !this.showTooltip || (this.showTooltip = !1);
    },
    scrollResize() {
      this.scrollResizeTimeout && (clearTimeout(this.scrollResizeTimeout), this.scrollResizeTimeout = null), this.scrollResizeTimeout = setTimeout(() => {
        this.showTooltip && this.closeTooltip();
      }, 100);
    }
  }
};
var n = function() {
  var t = this, i = t._self._c;
  return i(t.renderAs, { directives: [{ name: "click-outside", rawName: "v-click-outside", value: () => t.closeTooltip(), expression: "() => closeTooltip()" }], tag: "component", class: [
    "base-link",
    {
      "base-link--external": t.hasValidUrl,
      "base-link--internal": t.source && !t.type,
      "base-link--chip": t.source && t.type,
      "base-link--tooltip": t.tooltip.length || t.tooltipAsync.length,
      "base-link--active": t.showTooltip,
      "base-link--space-after": t.spaceAfter
    }
  ], attrs: { tabindex: t.source || t.hasValidUrl || t.source && t.type || t.tooltip.length ? 0 : null, role: t.tooltip.length ? "button" : null, href: t.source || t.hasValidUrl ? t.source || t.url : null, to: t.vueRouterAvailable && t.source && !t.type ? "/" + t.source : null, target: t.hasValidUrl && t.externalLinkTarget === "_blank" ? "_blank" : null, title: t.source || t.hasValidUrl || t.source && t.type ? t.value : null }, on: { click: function(l) {
    t.source && t.type && t.chipClicked;
  } } }, [!t.tooltip.length && !t.tooltipAsync.length ? [t._v(" " + t._s(t.value) + " ")] : t._e(), t.tooltip.length && !t.source || t.tooltipAsync.length && !t.source ? [i("span", { staticClass: "base-link__label", on: { keydown: function(l) {
    if (!l.type.indexOf("key") && t._k(l.keyCode, "enter", 13, l.key, "Enter"))
      return null;
    l.preventDefault(), t.tooltip.length && t.tooltipClicked;
  }, click: t.tooltipClicked } }, [t._v(" " + t._s(t.value) + " "), i("span", { ref: "icon", staticClass: "base-link__icon" }, [i("BaseIcon", { directives: [{ name: "show", rawName: "v-show", value: !t.isLoading, expression: "!isLoading" }], attrs: { name: "information" } })], 1), t.isLoading ? i("span", { staticClass: "base-link__loader" }, [t.isLoading ? i("BaseLoader") : t._e()], 1) : t._e()]), t.showTooltip ? i("BaseTooltip", { attrs: { "attach-to": t.$refs.icon, styles: t.tooltipStyles } }, [t._t("tooltip", function() {
    return [t.tooltip.label ? i("span", { staticClass: "base-tooltip__label" }, [t._v(" " + t._s(t.tooltip.label) + " ")]) : t._e(), t._l(t.tooltip, function(l, s) {
      return i("div", { key: s, staticClass: "base-tooltip__row" }, [t._v(" " + t._s(l.label) + ": "), l.url ? [i("a", { staticClass: "base-link--external", attrs: { href: l.url, title: l.value } }, [t._v(" " + t._s(l.value) + " ")])] : [t._v(t._s(l.value))]], 2);
    })];
  }, { item: t.tooltip })], 2) : t._e()] : t._e()], 2);
}, c = [], u = /* @__PURE__ */ a(
  r,
  n,
  c,
  !1,
  null,
  "22bc0e40",
  null,
  null
);
const f = u.exports;
export {
  f as default
};
