// base-ui-components@3.1.2/dist/components/BaseDetailBox/BaseDetailBox.es.js
import o from "../BaseBox/BaseBox.es.js";
import l from "../../mixins/i18n.es.js";
import "./BaseDetailBox.vue_vue_type_style_index_0_scoped_c88db47e_lang.es.js";
import s from "../../_virtual/_plugin-vue2_normalizer.es.js";
const n = {
  name: "BaseDetailBox",
  components: {
    BaseBox: o
  },
  mixins: [
    l
  ],
  props: {
    data: {
      type: Array,
      default: () => []
    },
    valuePropertyName: {
      type: String,
      default: "value"
    },
    labelPropertyName: {
      type: String,
      default: "label"
    },
    language: {
      type: String,
      default: "en"
    }
  }
};
var i = function() {
  var e = this, a = e._self._c;
  return a("base-box", { staticClass: "base-detail-box", attrs: { "box-size": { width: "100%" }, "box-hover": !1, "box-ratio": "0" } }, [a("dl", { staticClass: "base-detail-box-dl" }, [e._l(e.data, function(t, r) {
    return [a("dt", { key: "dt" + r }, [e._v(" " + e._s(e.getLangLabel(t[e.labelPropertyName], !0)) + " ")]), a("dd", { key: "dd" + r }, [e._v(" " + e._s(t[e.valuePropertyName]) + " ")])];
  })], 2)]);
}, d = [], p = /* @__PURE__ */ s(
  n,
  i,
  d,
  !1,
  null,
  "c88db47e",
  null,
  null
);
const b = p.exports;
export {
  b as default
};
