// base-ui-components/dist/components/BaseExpandRow/BaseExpandRow.es.js
import n from "../BaseIcon/BaseIcon.es.js";
import s from "../BaseCheckmark/BaseCheckmark.es.js";
import "./BaseExpandRow.vue_vue_type_style_index_0_scoped_42c293df_lang.es.js";
import d from "../../_virtual/_plugin-vue2_normalizer.es.js";
const l = {
  name: "BaseExpandRow",
  components: {
    BaseCheckmark: s,
    BaseIcon: n
  },
  props: {
    expandable: {
      type: Boolean,
      default: !0
    },
    title: {
      type: String,
      default: ""
    },
    subtitle: {
      type: String,
      default: ""
    },
    icon: {
      type: String,
      default: ""
    },
    bodyHasBackground: {
      type: Boolean,
      default: !0
    },
    isExpanded: {
      type: Boolean,
      default: !1
    },
    isSelectable: {
      type: Boolean,
      default: !1
    },
    isSelected: {
      type: Boolean,
      default: !1
    }
  },
  data() {
    return {
      id: null,
      isExpandedInternal: !1,
      isSelectedInternal: !1
    };
  },
  computed: {
    hasIconSlot() {
      return !!this.$slots.icon;
    }
  },
  watch: {
    isExpanded(t) {
      this.isExpandedInternal = t;
    },
    isSelected(t) {
      this.isSelectedInternal = t;
    }
  },
  created() {
    this.id = this._uid, this.isSelectedInternal = this.isSelected, this.isExpandedInternal = this.isExpanded;
  },
  methods: {
    clicked() {
      this.isExpandedInternal = !this.isExpandedInternal, this.$emit("expanded", this.isExpandedInternal);
    },
    checkboxClicked() {
      this.isSelectedInternal = !this.isSelectedInternal, this.$emit("selected", this.isSelectedInternal);
    }
  }
};
var i = function() {
  var e = this, a = e._self._c;
  return a("div", { class: [
    "base-expand-row",
    { "base-expand-row--expandable": e.expandable },
    { "base-expand-row--expanded": e.isExpandedInternal }
  ] }, [a("div", { staticClass: "base-expand-row-header", attrs: { id: "base-expand-row-" + e.id, "aria-expanded": e.expandable ? e.isExpandedInternal.toString() : null } }, [e.isSelectable ? a("BaseCheckmark", { key: e.id + "checkmark", staticClass: "base-expand-row-checkbox", attrs: { title: "checkbox", "mark-style": "checkbox", checked: e.isSelected }, on: { clicked: e.checkboxClicked } }) : e._e(), a(e.expandable ? "button" : "div", { tag: "component", class: ["base-expand-row-button", { selectable: e.isSelectable }], attrs: { type: e.expandable ? "button" : null }, on: { click: function(c) {
    e.expandable && e.clicked();
  } } }, [e.icon || e.hasIconSlot ? a("div", { staticClass: "base-expand-row-icon" }, [e._t("icon", function() {
    return [a("BaseIcon", { attrs: { name: e.icon, title: "open" } })];
  })], 2) : e._e(), a("div", { staticClass: "base-expand-row-title-subtitle-wrapper" }, [a("div", { staticClass: "base-expand-row-title base-text-fade-out" }, [e._v(" " + e._s(e.title) + " ")]), e.subtitle ? a("div", { staticClass: "base-expand-row-subtitle base-text-fade-out" }, [e._v(" " + e._s(e.subtitle) + " ")]) : e._e()]), e.expandable ? a("BaseIcon", { staticClass: "base-expand-row-collapse-icon", attrs: { name: "drop-down", title: "open" } }) : e._e()], 1)], 1), a("div", { class: ["base-expand-row-body", { "base-expand-row-body-bg": e.bodyHasBackground }], attrs: { role: "region", "aria-labelledby": "base-expand-row-" + e.id, "aria-hidden": e.isExpandedInternal ? "false" : "true" } }, [e._t("default")], 2)]);
}, o = [], r = /* @__PURE__ */ d(
  l,
  i,
  o,
  !1,
  null,
  "42c293df",
  null,
  null
);
const h = r.exports;
export {
  h as default
};
