// base-ui-components/dist/components/BaseAutocompleteInput/BaseAutocompleteInput.es.js
import o from "../BaseInput/BaseInput.es.js";
import r from "../BaseDropDownList/BaseDropDownList.es.js";
import { createId as s } from "../../utils/utils.es.js";
import a from "../../mixins/navigateList.es.js";
import "./BaseAutocompleteInput.vue_vue_type_style_index_0_scoped_532361b5_lang.es.js";
import l from "../../_virtual/_plugin-vue2_normalizer.es.js";
const p = {
  name: "BaseAutocompleteInput",
  components: {
    BaseInput: o,
    BaseDropDownList: r
  },
  mixins: [
    a
  ],
  model: {
    prop: "input",
    event: "input"
  },
  props: {
    input: {
      type: [String, Number],
      default: ""
    },
    list: {
      type: Array,
      default: () => []
    },
    label: {
      type: String,
      required: !0
    },
    showLabel: {
      type: Boolean,
      default: !0
    },
    id: {
      type: String,
      default: ""
    },
    placeholder: {
      type: String,
      default: "Enter Text Here"
    },
    required: {
      type: Boolean,
      default: !1
    },
    invalid: {
      type: Boolean,
      default: !1
    },
    errorMessage: {
      type: String,
      default: ""
    },
    showErrorIcon: {
      type: Boolean,
      default: !0
    },
    isActive: {
      type: Boolean,
      default: null
    },
    showInputBorder: {
      type: Boolean,
      default: !0
    },
    useFormFieldStyling: {
      type: Boolean,
      default: !0
    },
    clearable: {
      type: Boolean,
      default: !1
    },
    loadable: {
      type: Boolean,
      default: !1
    },
    isLoading: {
      type: Boolean,
      default: !1
    },
    language: {
      type: String,
      default: ""
    },
    disabled: {
      type: Boolean,
      default: !1
    },
    identifierPropertyName: {
      type: String,
      default: "id"
    },
    labelPropertyName: {
      type: String,
      default: "label"
    },
    dropDownNoOptionsInfo: {
      type: String,
      default: "No options available"
    },
    dynamicFetch: {
      type: Boolean,
      default: !0
    }
  },
  data() {
    return {
      inputInt: null,
      activeOptionIndex: -1,
      isActiveInt: !1
    };
  },
  computed: {
    inputListeners() {
      return {
        ...this.$listeners,
        input: () => {
        }
      };
    },
    listInt() {
      return this.optionsIsListOfStrings ? this.list.map((e) => ({
        [this.labelPropertyName]: e,
        [this.identifierPropertyName]: `${e}-${s()}`
      })) : this.list;
    },
    filteredListInt() {
      return this.dynamicFetch ? this.listInt : this.listInt.filter((e) => e[this.labelPropertyName].toLowerCase().includes(this.inputInt.toLowerCase()));
    },
    optionsIsListOfStrings() {
      return !!this.list && !!this.list.length && typeof this.list[0] == "string";
    },
    activeOption: {
      set(e) {
        this.activeOptionIndex = this.listInt.indexOf(e);
      },
      get() {
        return this.listInt[this.activeOptionIndex];
      }
    }
  },
  watch: {
    input: {
      handler(e) {
        e !== this.inputInt && (this.inputInt = e);
      },
      immediate: !0
    },
    inputInt(e) {
      e !== this.input && this.$emit("input", e), this.dynamicFetch && this.$emit("fetch-dropdown-entries", { value: e });
    },
    isActive: {
      handler(e) {
        e !== this.isActiveInt && (this.isActiveInt = e);
      },
      immediate: !0
    },
    isActiveInt(e) {
      e || (this.activeOptionIndex = -1), JSON.stringify(e) !== JSON.stringify(this.isActive) && this.$emit("update:is-active", e);
    }
  },
  methods: {
    onArrowKey(e) {
      this.listInt.length && (this.activeOption = this.navigate(
        this.listInt,
        e.key === "ArrowDown",
        this.activeOptionIndex,
        !1,
        !0
      ));
    },
    onEnter() {
      this.activeOption ? this.selectOption(this.activeOption) : this.isActiveInt = !1;
    },
    onKeydown(e) {
      const { key: t } = e;
      ["Tab", "Enter"].includes(t) || (this.isActiveInt = !0);
    },
    toggleDropDown() {
      this.isActiveInt = !this.isActiveInt;
    },
    selectOption(e) {
      if (this.inputInt = e[this.labelPropertyName], this.optionsIsListOfStrings)
        this.$emit("selected", this.inputInt);
      else {
        const t = this.list.find((n) => n[this.identifierPropertyName] === e[this.identifierPropertyName]);
        this.$emit("selected", t);
      }
      this.isActiveInt = !1;
    }
  }
};
var u = function() {
  var t = this, n = t._self._c;
  return n("div", { staticClass: "base-autocomplete-input" }, [n("BaseInput", t._g({ staticClass: "base-autocomplete-input__input-field", attrs: { id: t.id, "is-active": t.isActiveInt, label: t.label, "show-label": t.showLabel, placeholder: t.placeholder, required: t.required, invalid: t.invalid, "error-message": t.errorMessage, "show-error-icon": t.showErrorIcon, "use-form-field-styling": t.useFormFieldStyling, "show-input-border": t.showInputBorder, clearable: t.clearable, loadable: t.loadable, "is-loading": t.isLoading, language: t.language, disabled: t.disabled, "drop-down-list-id": `${t.id}-list-identifier`, "linked-list-option": t.activeOption ? t.activeOption[t.identifierPropertyName] : null }, on: { "update:isActive": function(i) {
    t.isActiveInt = i;
  }, "update:is-active": function(i) {
    t.isActiveInt = i;
  }, keydown: [function(i) {
    return !i.type.indexOf("key") && t._k(i.keyCode, "enter", 13, i.key, "Enter") ? null : (i.preventDefault(), t.onEnter.apply(null, arguments));
  }, function(i) {
    return !i.type.indexOf("key") && t._k(i.keyCode, "up", 38, i.key, ["Up", "ArrowUp"]) && t._k(i.keyCode, "down", 40, i.key, ["Down", "ArrowDown"]) ? null : (i.preventDefault(), t.onArrowKey.apply(null, arguments));
  }, t.onKeydown] }, scopedSlots: t._u([{ key: "below-input", fn: function() {
    return [t.isActiveInt ? n("BaseDropDownList", { staticClass: "base-autocomplete-input__drop-down", attrs: { "drop-down-options": t.filteredListInt, "active-option": t.activeOption, "identifier-property-name": t.identifierPropertyName, "label-property-name": t.labelPropertyName, "list-id": `${t.id}-list-identifier`, language: t.language, "drop-down-no-options-info": t.dropDownNoOptionsInfo }, on: { "update:activeOption": function(i) {
      t.activeOption = i;
    }, "update:active-option": function(i) {
      t.activeOption = i;
    }, "update:selected-option": t.selectOption }, nativeOn: { click: function(i) {
      i.stopPropagation();
    }, touchstart: function(i) {
      i.stopPropagation();
    } }, scopedSlots: t._u([{ key: "option", fn: function({ option: i }) {
      return [t._t("drop-down-entry", null, { item: i })];
    } }], null, !0) }) : t._e()];
  }, proxy: !0 }, { key: "label-addition", fn: function() {
    return [t._t("label-addition")];
  }, proxy: !0 }, { key: "pre-input-field", fn: function() {
    return [t._t("pre-input-field")];
  }, proxy: !0 }, { key: "input-field-addition-before", fn: function() {
    return [t._t("input-field-addition-before")];
  }, proxy: !0 }, { key: "input-field-inline-before", fn: function() {
    return [t._t("input-field-inline-before")];
  }, proxy: !0 }, { key: "input-field-addition-after", fn: function() {
    return [t._t("input-field-addition-after")];
  }, proxy: !0 }, { key: "post-input-field", fn: function() {
    return [t._t("post-input-field")];
  }, proxy: !0 }, { key: "error-icon", fn: function() {
    return [t._t("error-icon")];
  }, proxy: !0 }, { key: "remove-icon", fn: function() {
    return [t._t("remove-icon")];
  }, proxy: !0 }], null, !0), model: { value: t.inputInt, callback: function(i) {
    t.inputInt = i;
  }, expression: "inputInt" } }, t.inputListeners))], 1);
}, d = [], f = /* @__PURE__ */ l(
  p,
  u,
  d,
  !1,
  null,
  "532361b5",
  null,
  null
);
const v = f.exports;
export {
  v as default
};
