// base-ui-components@3.1.2/dist/components/BaseButton/BaseButton.es.js
import { createId as a } from "../../utils/utils.es.js";
import i from "../BaseIcon/BaseIcon.es.js";
import l from "../BaseBoxTooltip/BaseBoxTooltip.es.js";
import "./BaseButton.vue_vue_type_style_index_0_scoped_19e5830f_lang.es.js";
import r from "../../_virtual/_plugin-vue2_normalizer.es.js";
const s = {
  name: "BaseButton",
  components: {
    BaseIcon: i,
    BaseBoxTooltip: l
  },
  props: {
    text: {
      type: String,
      default: "Submit"
    },
    icon: {
      type: String,
      default: null
    },
    hideIcon: {
      type: Boolean,
      default: !1
    },
    iconPosition: {
      type: String,
      default: "left",
      validator(e) {
        return ["top", "right", "left"].includes(e);
      }
    },
    active: {
      type: Boolean,
      default: !1
    },
    buttonStyle: {
      type: String,
      default: "single",
      validator(e) {
        return ["single", "row", "secondary"].includes(e);
      }
    },
    iconColored: {
      type: Boolean,
      default: !1
    },
    iconSize: {
      type: String,
      default: "small",
      validator(e) {
        return ["small", "large"].includes(e);
      }
    },
    iconTitle: {
      type: String,
      default: ""
    },
    disabled: {
      type: Boolean,
      default: !1
    },
    buttonType: {
      type: String,
      default: "button"
    },
    alignText: {
      type: String,
      default: "center",
      validator(e) {
        return ["center", "left", "right"].includes(e);
      }
    },
    hasBackgroundColor: {
      type: Boolean,
      default: !0
    },
    showTooltip: {
      type: Boolean,
      default: !1
    },
    description: {
      type: String,
      default: ""
    },
    buttonTextWrap: {
      type: Boolean,
      default: !0
    }
  },
  data() {
    return {
      internalId: a()
    };
  },
  methods: {
    clicked(e) {
      this.$emit("clicked", e);
    }
  }
};
var c = function() {
  var t = this, n = t._self._c;
  return n("button", { class: [
    "base-button",
    `base-button-${t.buttonStyle}`,
    `base-button-icon-${t.iconPosition}`,
    `base-button-align-${t.alignText}`,
    { "base-button-background": t.hasBackgroundColor },
    { "base-button-active": t.active }
  ], attrs: { disabled: t.disabled, "aria-disabled": t.disabled, "aria-describedby": t.internalId, type: t.buttonType }, on: { click: function(o) {
    return o.preventDefault(), t.clicked.apply(null, arguments);
  } } }, [t.icon ? n("base-icon", { class: [
    "base-button-icon",
    "base-button-icon-" + t.iconSize,
    { "base-button-icon-colored": t.iconColored },
    { "base-button-icon-hide": t.hideIcon }
  ], attrs: { name: t.icon, title: t.iconTitle } }) : t._e(), t._t("left-of-text"), t._t("text", function() {
    return [t.text ? n("span", { class: ["base-button-text", { "base-button-text__nowrap": !t.buttonTextWrap }] }, [t._v(" " + t._s(t.text) + " ")]) : t._e()];
  }), t._t("right-of-text"), t.showTooltip ? n("BaseBoxTooltip", { on: { clicked: t.clicked } }) : t._e(), t.description ? n("span", { staticClass: "hide", attrs: { id: t.internalId } }, [t._v(" " + t._s(t.description) + " ")]) : t._e()], 2);
}, u = [], d = /* @__PURE__ */ r(
  s,
  c,
  u,
  !1,
  null,
  "19e5830f",
  null,
  null
);
const m = d.exports;
export {
  m as default
};
