// base-ui-components/dist/components/BaseEntrySelector/BaseEntrySelector.es.js
import i from "../../mixins/i18n.es.js";
import "./BaseEntrySelector.vue_vue_type_style_index_0_scoped_3eb2b813_lang.es.js";
import l from "../../_virtual/_plugin-vue2_normalizer.es.js";
const o = {
  name: "BaseEntrySelector",
  components: {
    BaseDropDown: () => import("../BaseDropDown/BaseDropDown.es.js").then((t) => t.default || t),
    BaseLoader: () => import("../BaseLoader/BaseLoader.es.js").then((t) => t.default || t),
    BaseMenuList: () => import("../BaseMenuList/BaseMenuList.es.js").then((t) => t.default || t),
    BaseOptions: () => import("../BaseOptions/BaseOptions.es.js").then((t) => t.default || t),
    BasePagination: () => import("../BasePagination/BasePagination.es.js").then((t) => t.default || t),
    BaseSearch: () => import("../BaseSearch/BaseSearch.es.js").then((t) => t.default || t),
    BaseSelectOptions: () => import("../BaseSelectOptions/BaseSelectOptions.es.js").then((t) => t.default || t)
  },
  mixins: [i],
  props: {
    entries: {
      type: Array,
      default: () => []
    },
    entriesTotal: {
      type: Number,
      default: null
    },
    entriesPerPage: {
      type: Number,
      default: 10
    },
    entriesSelectable: {
      type: Boolean,
      default: !1
    },
    maxSelectedEntries: {
      type: Number,
      default: 0
    },
    height: {
      type: String,
      default: ""
    },
    isLoading: {
      type: Boolean,
      default: !1
    },
    activeEntry: {
      type: Number,
      default: -1
    },
    language: {
      type: String,
      default: "en"
    },
    optionsHidden: {
      type: Boolean,
      default: !1
    },
    entryTypes: {
      type: Array,
      default: () => []
    },
    entryTypesConfig: {
      type: Object,
      default: () => ({
        label: "filter by type",
        default: null,
        valuePropertyName: "value"
      }),
      validator: (t) => !t || ["label", "default", "valuePropertyName"].every((e) => Object.keys(t).includes(e))
    },
    sortOptions: {
      type: Array,
      default: () => []
    },
    sortConfig: {
      type: Object,
      default: () => ({
        label: "sort entries",
        default: null,
        valuePropertyName: "value"
      }),
      validator: (t) => !t || Object.keys(t).every((e) => [
        "label",
        "default",
        "valuePropertyName"
      ].includes(e))
    },
    entrySelectorText: {
      type: Object,
      default: () => ({
        noEntriesTitle: "No matching entries found",
        noEntriesSubtext: "Please adjust your search criteria.",
        options: {
          show: "options",
          hide: "options"
        },
        search: "Search",
        selectAll: "Select All",
        selectNone: "Select None",
        entriesSelected: "entries selected",
        maxEntriesReached: "Maximum Number Exceeded!"
      }),
      validator: (t) => [
        "noEntriesTitle",
        "noEntriesSubtext",
        "options",
        "search",
        "selectAll",
        "selectNone",
        "entriesSelected",
        "maxEntriesReached"
      ].every((e) => Object.keys(t).includes(e)) && ["show", "hide"].every((e) => Object.keys(t.options).includes(e))
    },
    useSearch: {
      type: Boolean,
      default: !0
    },
    showOptionsRow: {
      type: Boolean,
      default: !0
    }
  },
  data() {
    return {
      filterString: "",
      filterType: {},
      sortParam: {},
      headHasShadow: !1,
      pageNumber: 1,
      showOptions: !1,
      selectedEntries: [],
      timeout: null
    };
  },
  computed: {
    calcStyle() {
      return this.height ? { height: this.height } : {};
    },
    pageTotal() {
      return this.entriesPerPage ? Math.ceil(this.entriesTotal / this.entriesPerPage) : 0;
    },
    selectedListIds() {
      return [...new Set(this.selectedEntries.map((t) => t.id))];
    },
    dropDownElementsCount() {
      return this.sortOptions && this.sortOptions.length && this.entryTypes && this.entryTypes.length ? 2 : 1;
    },
    selectableEntries() {
      return this.entries.filter((t) => !t.disabled);
    }
  },
  watch: {
    entriesSelectable: {
      handler(t) {
        t !== this.showOptions && (this.showOptions = t);
      },
      immediate: !0
    },
    showOptions(t) {
      t || (this.selectedEntries = []), t !== this.entriesSelectable && this.$emit("update:entries-selectable", this.showOptions);
    },
    selectedEntries() {
      this.$emit("selected-changed", this.selectedEntries);
    }
  },
  created() {
    this.sortParam = this.sortConfig.default || (this.sortOptions.length ? this.sortOptions[0] : {}), this.filterType = this.entryTypesConfig.default || (this.entryTypes.length ? this.entryTypes[0] : {});
  },
  mounted() {
    this.fetchEntries(), this.$refs.body.addEventListener("scroll", this.scroll);
  },
  beforeDestroy() {
    this.$refs.body.removeEventListener("scroll", this.scroll);
  },
  methods: {
    changeAllSelectState(t) {
      if (t)
        this.selectedEntries = this.selectedEntries.concat(this.selectableEntries), this.selectedEntries = [...new Set(this.selectedEntries)];
      else {
        const e = this.entries.map((s) => s.id);
        this.selectedEntries = this.selectedEntries.filter((s) => !e.includes(s.id));
      }
    },
    fetchEntries() {
      this.timeout && (clearTimeout(this.timeout), this.timeout = null), this.timeout = setTimeout(() => {
        this.$emit("fetch-entries", {
          page: this.pageNumber,
          query: this.filterString,
          sort: this.sortParam,
          type: this.filterType
        });
      }, 600);
    },
    filterEntries(t, e) {
      e === "type" ? this.fetchEntries() : e === "title" && (this.timeout && (clearTimeout(this.timeout), this.timeout = null), this.timeout = setTimeout(() => {
        (t.length === 0 || t.length > 2) && this.fetchEntries();
      }, 600)), this.pageNumber = 1;
    },
    scroll() {
      if (this.$refs.body.scrollTop) {
        this.headHasShadow = !0;
        return;
      }
      this.headHasShadow = !1;
    },
    selectEntry(t) {
      t.selected ? this.selectedEntries.push(this.entries[t.index]) : this.selectedEntries = this.selectedEntries.filter((e) => e.id !== this.entries[t.index].id);
    },
    setPage(t) {
      this.pageNumber = t, this.fetchEntries();
    },
    entryClicked(t) {
      this.$emit("entry-clicked", this.entries[t].id);
    }
  }
};
var a = function() {
  var e = this, s = e._self._c;
  return s("div", { staticClass: "base-entry-selector", style: e.calcStyle }, [s("div", { ref: "head", class: [
    "base-entry-selector__head",
    { "base-entry-selector__head--shadow": e.headHasShadow },
    { "base-entry-selector__head--padding": e.useSearch }
  ] }, [e._t("head", function() {
    return [e.useSearch ? s("BaseSearch", { class: [
      "base-entry-selector__head__search-bar",
      { "base-entry-selector__head__search-bar--margin-large": !e.showOptionsRow }
    ], attrs: { "show-image": !0, placeholder: e.getI18nTerm(e.entrySelectorText.search) }, on: { input: function(n) {
      return e.filterEntries(n, "title");
    } }, model: { value: e.filterString, callback: function(n) {
      e.filterString = n;
    }, expression: "filterString" } }) : e._e()];
  }), e.showOptionsRow ? s("div", { staticClass: "base-entry-selector__options" }, [s("BaseOptions", { ref: "baseOptions", attrs: { "show-options": e.showOptions, "options-hidden": e.optionsHidden, "use-options-button-on": "always", "show-after-options-below": !0, "options-button-icon": {
    show: "options-menu",
    hide: "options-menu"
  }, "options-button-text": e.entrySelectorText.options, "align-options": "left" }, on: { "update:showOptions": function(n) {
    e.showOptions = n;
  }, "update:show-options": function(n) {
    e.showOptions = n;
  } }, scopedSlots: e._u([{ key: "afterOptions", fn: function() {
    return [s("div", { staticClass: "base-entry-selector__dropdowns" }, [e._t("after-options", function() {
      return [e.sortOptions.length ? s("BaseDropDown", { ref: "baseDropDown", staticClass: "base-entry-selector__dropdowns__dropdown", style: { maxWidth: `${100 / e.dropDownElementsCount}%` }, attrs: { id: `${e.sortConfig.label}-sort-drop-down`, label: e.sortConfig.label, options: e.sortOptions, "with-spacing": !1, "align-drop-down": e.entryTypes.length ? "left" : "right", "value-prop": e.sortConfig.valuePropertyName }, on: { "value-selected": e.fetchEntries }, model: { value: e.sortParam, callback: function(n) {
        e.sortParam = n;
      }, expression: "sortParam" } }) : e._e(), e.entryTypes.length ? s("BaseDropDown", { ref: "baseDropDown", staticClass: "base-entry-selector__dropdowns__dropdown", style: { maxWidth: `${100 / e.dropDownElementsCount}%` }, attrs: { id: `${e.entryTypesConfig.label}-types-drop-down`, label: e.entryTypesConfig.label, options: e.entryTypes, language: e.language, "with-spacing": !1, "value-prop": e.entryTypesConfig.valuePropertyName, "align-drop-down": "right" }, on: { "value-selected": function(n) {
        return e.filterEntries(n, "type");
      } }, model: { value: e.filterType, callback: function(n) {
        e.filterType = n;
      }, expression: "filterType" } }) : e._e()];
    })], 2)];
  }, proxy: !0 }, { key: "options", fn: function() {
    return [e._t("option-actions")];
  }, proxy: !0 }], null, !0) })], 1) : e._e(), e.showOptions ? s("BaseSelectOptions", { ref: "selectOptions", attrs: { "select-text": e.getI18nTerm(e.entrySelectorText.selectAll), "selected-number-text": e.getI18nTerm(e.entrySelectorText.entriesSelected), "deselect-text": e.getI18nTerm(e.entrySelectorText.selectNone), list: e.selectableEntries, "selected-list": e.selectedEntries, "select-all-disabled": !!e.maxSelectedEntries && !(e.selectableEntries.length < e.maxSelectedEntries - e.selectedListIds.length || !e.selectableEntries.some((n) => !e.selectedListIds.includes(n.id))) }, on: { selected: e.changeAllSelectState }, scopedSlots: e._u([{ key: "selectedText", fn: function() {
    return [e._v(" " + e._s(`${e.selectedListIds.length}${e.maxSelectedEntries ? `/${e.maxSelectedEntries}` : ""} ${e.getI18nTerm(e.entrySelectorText.entriesSelected)}`) + " "), !!e.maxSelectedEntries && e.selectedListIds.length >= e.maxSelectedEntries ? s("span", [e._v(" " + e._s(`(${e.getI18nTerm(e.entrySelectorText.maxEntriesReached)})`) + " ")]) : e._e()];
  }, proxy: !0 }], null, !1, 1235622275) }) : e._e()], 2), s("div", { ref: "body", staticClass: "base-entry-selector__body" }, [e.isLoading ? s("div", { staticClass: "loading-area" }, [s("BaseLoader", { class: { "base-entry-selector__loader__center": e.entries.length < 4 } })], 1) : e._e(), e._t("entries", function() {
    return [e.entries.length ? s("BaseMenuList", { key: "menu-list", ref: "menuList", staticClass: "base-entry-selector__body__entries", attrs: { "select-active": e.showOptions, list: e.entries, "active-entry": e.activeEntry, "selected-list": e.selectedListIds }, on: { clicked: e.entryClicked, selected: e.selectEntry }, scopedSlots: e._u([{ key: "entry-text-content", fn: function({ item: n }) {
      return [e._t("entry-text-content", null, { item: n })];
    } }, { key: "entry-right-side-elements", fn: function({ isSelected: n, item: r }) {
      return [e._t("entry-right-side-elements", null, { isSelected: n, selectActive: e.showOptions, item: r })];
    } }, { key: "thumbnails", fn: function({ item: n }) {
      return [e._t("thumbnails", null, { item: n })];
    } }], null, !0) }) : e.isLoading ? e._e() : s("div", { staticClass: "base-entry-selector__no-entries" }, [s("p", { staticClass: "base-entry-selector__no-entries__title" }, [e._v(" " + e._s(e.getI18nTerm(e.entrySelectorText.noEntriesTitle)) + " ")]), s("p", { staticClass: "base-entry-selector__no-entries__subtext" }, [e._v(" " + e._s(e.getI18nTerm(e.entrySelectorText.noEntriesSubtext)) + " ")])])];
  }, { entries: e.entries, selectEntry: e.selectEntry })], 2), e.pageTotal > 1 ? s("BasePagination", { ref: "pagination", attrs: { total: e.pageTotal, current: e.pageNumber }, on: { "set-page": e.setPage } }) : e._e()], 1);
}, c = [], d = /* @__PURE__ */ l(
  o,
  a,
  c,
  !1,
  null,
  "3eb2b813",
  null,
  null
);
const y = d.exports;
export {
  y as default
};
