// base-ui-components/dist/components/BaseMenuList/BaseMenuList.es.js
import n from "vuedraggable";
import l from "../BaseMenuEntry/BaseMenuEntry.es.js";
import "./BaseMenuList.vue_vue_type_style_index_0_scoped_2121ef26_lang.es.js";
import o from "../../_virtual/_plugin-vue2_normalizer.es.js";
const c = {
  name: "BaseMenuList",
  components: {
    BaseMenuEntry: l,
    Draggable: n
  },
  props: {
    selectActive: {
      type: Boolean,
      default: !1
    },
    list: {
      type: Array,
      default: () => [],
      validator: (t) => t.every((e) => Object.keys(e).includes("id"))
    },
    activeEntry: {
      type: Number,
      default: -1
    },
    selectedList: {
      type: Array,
      default: () => []
    },
    dragName: {
      type: String,
      default: "menuEntry"
    }
  },
  data() {
    return {
      entryProps: [],
      dragging: !1,
      dragAndDropCapable: !1,
      isDraggable: !0,
      resizeTimeout: null,
      dragImg: null
    };
  },
  watch: {
    list() {
      this.setInternalVar();
    },
    activeEntry(t) {
      this.entryProps.map((e) => this.$set(e, "active", !1)), t >= 0 && this.entryProps[t] && this.$set(this.entryProps[t], "active", !0);
    },
    selectActive(t) {
      t ? this.setInternalVar() : this.entryProps.forEach((e) => this.$set(e, "selected", !1));
    },
    selectedList() {
      this.setInternalVar();
    }
  },
  created() {
    this.setInternalVar();
  },
  mounted() {
    this.isDraggable = !this.isMobile(), this.dragAndDropCapable = "DragEvent" in window, window.addEventListener("resize", () => {
      clearTimeout(this.resizeTimeout), this.resizeTimeout = setTimeout(() => {
        this.isDraggable = !this.isMobile();
      }, 250);
    }), typeof this.$refs.draggable._sortable.nativeDraggable < "u" ? this.$refs.draggable._sortable.nativeDraggable = this.dragAndDropCapable : console.warn('The option "nativeDraggable" in sortableJS is missing. Please check for changes https://github.com/SortableJS/sortablejs.');
  },
  methods: {
    activateItem(t) {
      this.$emit("clicked", t);
    },
    selectItem(t, e) {
      this.$set(this.entryProps[t], "selected", e), this.$emit("selected", { index: t, selected: e });
    },
    setInternalVar() {
      this.entryProps = this.list.map((t) => ({
        selected: t.selected || this.selectedList.includes(t.id),
        active: t.active || !1,
        error: t.error || !1
      })), this.entryProps.length && this.activeEntry >= 0 && this.$set(this.entryProps[this.activeEntry], "active", !0);
    },
    getDragImage(t) {
      const e = t.item.getElementsByClassName("base-menu-entry-icon")[0];
      if (e) {
        const r = `${e.clientHeight * 2}px`, s = e.cloneNode(!0);
        s.id = "drag-icon", s.style.height = r, s.style.maxHeight = r, s.style.width = r, s.style.backgroundColor = "white", s.style.position = "absolute", s.style.top = "-99999px", s.style.left = "-99999px", this.dragImg = s;
      } else
        this.dragImg = null;
    },
    dragStart() {
      this.dragging = !0;
    },
    dragEnd() {
      this.dragging = !1;
      const t = document.getElementById("drag-icon");
      t && t.parentNode.removeChild(t);
    },
    modifyDragItem(t) {
      this.dragImg && (document.body.appendChild(this.dragImg), typeof DataTransfer.prototype.setDragImage == "function" && t.setDragImage(this.dragImg, 0, 0)), t.setData("draggable", "");
    },
    isMobile() {
      return window.innerWidth < 640;
    }
  }
};
var d = function() {
  var e = this, r = e._self._c;
  return r("draggable", { ref: "draggable", staticClass: "base-menu-list", attrs: { value: e.list, sort: !1, disabled: !e.isDraggable || e.selectActive, group: { name: e.dragName, pull: "clone", put: !1 }, "set-data": e.modifyDragItem, "force-fallback": !e.dragAndDropCapable, "fallback-on-body": !e.dragAndDropCapable, tag: "ul" }, on: { choose: e.getDragImage, start: e.dragStart, end: e.dragEnd } }, e._l(e.list, function(s, a) {
    return r("li", { key: s.id || s.title, staticClass: "base-menu-list__list-entry" }, [r("base-menu-entry", { ref: "menuEntry", refInFor: !0, attrs: { "entry-id": s.id, title: s.title, "is-active": e.entryProps[a].active, "is-selected": e.entryProps[a].selected, "is-disabled": s.disabled, icon: s.icon, description: s.description, "is-selectable": !0, "select-active": e.selectActive }, on: { clicked: function(i) {
      return e.activateItem(a);
    }, selected: function(i) {
      return e.selectItem(a, i);
    } }, scopedSlots: e._u([{ key: "text-content", fn: function() {
      return [e._t("entry-text-content", null, { item: s })];
    }, proxy: !0 }, { key: "right-side-elements", fn: function({ isSelected: i }) {
      return [e._t("entry-right-side-elements", null, { isSelected: i, item: s })];
    } }, { key: "thumbnails", fn: function() {
      return [e._t("thumbnails", null, { item: s })];
    }, proxy: !0 }], null, !0) })], 1);
  }), 0);
}, g = [], u = /* @__PURE__ */ o(
  c,
  d,
  g,
  !1,
  null,
  "2121ef26",
  null,
  null
);
const y = u.exports;
export {
  y as default
};
