// base-ui-components/dist/components/BaseForm/BaseForm.es.js
import c from "../BaseIcon/BaseIcon.es.js";
import p from "../../mixins/i18n.es.js";
import m from "../BaseFormFieldCreator/BaseFormFieldCreator.es.js";
import "./BaseForm.vue_vue_type_style_index_0_scoped_fff90047_lang.es.js";
import "./BaseForm.vue_vue_type_style_index_1_lang.es.js";
import h from "../../_virtual/_plugin-vue2_normalizer.es.js";
const _ = [
  "errorMessage",
  "invalid",
  "isActive",
  "isLoading",
  "linkedListOption",
  "activeTab",
  "validationTexts"
], g = {
  name: "BaseForm",
  components: {
    BaseFormFieldCreator: m,
    BaseIcon: c
  },
  mixins: [p],
  props: {
    formFieldJson: {
      type: Object,
      required: !0
    },
    valueList: {
      type: Object,
      default: () => ({})
    },
    language: {
      type: String,
      default: "en"
    },
    availableLocales: {
      type: Array,
      default: () => []
    },
    formId: {
      type: String,
      default: ""
    },
    formStyle: {
      type: Object,
      default: () => ({})
    },
    fieldsWithTabs: {
      type: Array,
      default: () => []
    },
    fieldIsLoading: {
      type: String,
      default: ""
    },
    dropDownLists: {
      type: Object,
      default: () => ({})
    },
    showErrorIcon: {
      type: Boolean,
      default: !1
    },
    clearable: {
      type: Boolean,
      default: !1
    },
    fieldProps: {
      type: Object,
      default: () => ({}),
      validator: (e) => Object.keys(e).length === 0 || Object.values(e).every((t) => Object.keys(t))
    },
    renderHeaderAs: {
      type: String,
      default: "div"
    },
    validationTexts: {
      type: Object,
      default: () => ({
        text: {
          min: "Value must be greater than or equal to {value}.",
          max: "Value must be less than or equal to {value}.",
          minLength: "Text must be at least {value} character(s) long.",
          maxLength: "Text cannot be longer than {value} characters."
        }
      }),
      validator: (e) => ["text"].every((t) => Object.keys(e).includes(t)) && ["min", "max", "minLength", "maxLength"].every((t) => Object.keys(e.text).includes(t))
    }
  },
  data() {
    return {
      multiplyParams: null,
      fetchingAutocompleteFor: "",
      currentInputString: "",
      valueListInt: {}
    };
  },
  computed: {
    formFieldsHalf() {
      return this.formFieldListInt.filter((e) => e["x-attrs"] && e["x-attrs"].field_format === "half");
    },
    formFieldListInt() {
      return Object.keys(this.formFieldJson).filter((e) => !this.formFieldJson[e].$ref && !(this.formFieldJson[e]["x-attrs"] && this.formFieldJson[e]["x-attrs"].hidden)).map((e) => ({ name: e, ...this.formFieldJson[e] })).sort((e, t) => e["x-attrs"] && t["x-attrs"] && e["x-attrs"].order > t["x-attrs"].order ? 1 : -1);
    },
    formHeader() {
      const e = this.formFieldListInt.find((t) => t["x-attrs"] && t["x-attrs"].form_group_title);
      return e ? e["x-attrs"].form_group_title : "";
    }
  },
  watch: {
    valueList: {
      handler(e) {
        Object.keys(this.valueListInt).some((r) => JSON.stringify(this.valueListInt[r]) !== JSON.stringify(e[r])) && this.initializeValueObject();
      },
      deep: !0
    },
    formFieldJson: {
      handler() {
        this.valueListInt = {}, this.initializeValueObject();
      },
      immediate: !0
    }
  },
  updated() {
    if (this.multiplyParams && this.multiplyParams.name) {
      const e = this.$refs[this.multiplyParams.name];
      e.length && setTimeout((t) => {
        const r = e[t.index].getElementsByTagName("textarea").length ? e[t.index].getElementsByTagName("textarea") : e[t.index].getElementsByTagName("input");
        r.length && r[0].focus();
      }, 50, this.multiplyParams), this.multiplyParams = null;
    }
  },
  methods: {
    fetchAutocomplete(e) {
      this.currentInputString = e.value, this.fetchingAutocompleteFor = e.name, this.$emit("fetch-autocomplete", e);
    },
    allowMultiply(e) {
      return e.type === "array" && (!e["x-attrs"] || !e["x-attrs"].field_type || e["x-attrs"] && !["chips", "chips-below"].includes(e["x-attrs"].field_type));
    },
    multiplyButtonsInline(e) {
      return !["group", "multiline"].includes(e["x-attrs"].field_type);
    },
    multiplyField(e, t) {
      const r = this.getInitialFieldValue(e.items);
      !t && t !== 0 ? this.valueListInt[e.name].push(r) : this.valueListInt[e.name].splice(t + 1, 0, r), this.multiplyParams = {
        index: !t && t !== 0 ? this.valueListInt[e.name].length - 1 : t + 1,
        name: e.name
      }, this.propagateValueListChanges();
    },
    removeField(e, t) {
      const r = this.valueListInt[e.name];
      r && r.length > 1 ? r.splice(t, 1) : this.$set(r, t, this.getInitialFieldValue(e.items)), this.propagateValueListChanges();
    },
    propagateValueListChanges() {
      this.$emit("values-changed", this.valueListInt);
    },
    isHalfFieldSecond(e) {
      const t = this.formFieldsHalf.indexOf(e);
      return t > 0 && !!(t % 2);
    },
    getFieldName(e) {
      return e.title || this.getI18nTerm(`form.${e.name}` || e.name);
    },
    formFieldComponentProps(e, t, r) {
      const { name: i } = e;
      let o = this.fieldProps[i] ? JSON.parse(JSON.stringify(this.fieldProps[i])) : {};
      const a = r >= 0;
      if (a) {
        const s = Object.entries(o).filter(([l]) => _.includes(l));
        if (s.length) {
          const l = s.reduce((u, [f, d]) => Object.keys(d).includes(r.toString()) ? {
            ...u,
            [f]: d[r.toString()]
          } : (delete o[f], u), {});
          o = {
            ...o,
            ...l
          };
        }
      }
      const n = a ? `${t}_${r}` : t;
      return {
        field: e,
        label: o && o.label ? o.label : this.getFieldName(e),
        fieldProps: o,
        showLabel: !this.allowMultiply(e) || !this.multiplyButtonsInline(e) || r === 0,
        dropDownList: this.dropDownLists[i],
        secondaryDropdown: this.dropDownLists[`${i}_secondary`],
        language: this.language,
        availableLocales: this.availableLocales,
        sortText: this.getI18nTerm("form.sort") || "Sort",
        fieldKey: `${i}_${n}_${this.formId}`,
        fieldValue: a ? this.valueListInt[i][r] : this.valueListInt[i],
        autocompleteLoading: this.fieldIsLoading === i,
        fieldGroupParams: this.formFieldListInt.some((s) => s["x-attrs"] && s["x-attrs"].field_type === "group") ? this.$props : null,
        clearable: this.clearable,
        showErrorIcon: this.showErrorIcon,
        validationTexts: o.validationTexts || this.validationTexts
      };
    },
    setFieldValue(e, t, r) {
      this.currentInputString = "", r >= 0 ? this.$set(this.valueListInt[t], r, JSON.parse(JSON.stringify(e))) : this.$set(this.valueListInt, t, e && JSON.parse(JSON.stringify(e))), this.$emit("values-changed", this.valueListInt);
    },
    initializeValueObject() {
      this.formFieldListInt.forEach((e) => {
        this.$set(
          this.valueListInt,
          e.name,
          this.getInitialFieldValue(e)
        );
      });
    },
    getInitialFieldValue(e) {
      const t = this.valueList[e.name], { type: r } = e, i = e["x-attrs"] && e["x-attrs"].field_type ? e["x-attrs"].field_type : void 0;
      if (["number", "integer"].includes(r))
        return t || "";
      if (i && i.includes("chips") && r === "object")
        return t && Object.keys(t).length ? [].concat(t) : [];
      if (r === "array")
        return typeof t == "object" && t && t.length ? [].concat(t) : !e["x-attrs"] || !e["x-attrs"].field_type || i && !i.includes("chips") ? [].concat(this.getInitialFieldValue(e.items)) : [];
      if (r === "object") {
        const o = {};
        return Object.keys(e.properties).forEach((a) => {
          this.$set(o, a, this.getInitialFieldValue(e.properties[a]));
        }), { ...o, ...t };
      }
      return r === "boolean" ? t || !1 : typeof t == "string" ? t : "";
    },
    checkFieldContent(e) {
      let t = !1;
      return e && typeof e == "object" ? e.length >= 0 ? e.forEach((r) => {
        t = this.checkFieldContent(r) || t;
      }) : Object.keys(e).forEach((i) => {
        t = this.checkFieldContent(e[i]) || t;
      }) : t = e === 0 || !!e || t, t;
    }
  }
};
var y = function() {
  var t = this, r = t._self._c;
  return r("div", { staticClass: "base-form", style: t.formStyle }, [t.formHeader ? r(t.renderHeaderAs, { tag: "component", staticClass: "base-form__header" }, [t._v(" " + t._s(t.formHeader) + " ")]) : t._e(), r("div", { staticClass: "base-form__body" }, [t._l(t.formFieldListInt, function(i, o) {
    return [t.allowMultiply(i) ? [t._l(t.valueListInt[i.name], function(a, n) {
      return r("div", { key: `${i.name}_${o}_${n}_${t.formId}_wrapper`, ref: i.name, refInFor: !0, class: [
        "base-form-field",
        { "base-form-field__multiple__inline": t.multiplyButtonsInline(i) },
        { "base-form-field__multiple__inline--has-title": n === 0 },
        t.formFieldsHalf.indexOf(i) >= 0 ? "base-form-field-half" : "base-form-field-full",
        { "base-form-field-left-margin": t.isHalfFieldSecond(i) }
      ] }, [r("BaseFormFieldCreator", t._b({ key: `${i.name}_${o}_${n}_${t.formId}`, staticClass: "base-form-field__multiple__inline-element", on: { "field-value-changed": function(s) {
        t.setFieldValue(
          s,
          i.name,
          n,
          i["x-attrs"] ? i["x-attrs"].equivalent : ""
        );
      }, "fetch-autocomplete": t.fetchAutocomplete, "subform-input": function(s) {
        return t.setFieldValue(s, i.name, n);
      } }, scopedSlots: t._u([{ key: "label-addition", fn: function({ fieldName: s, groupNames: l }) {
        return [t._t("label-addition", null, { fieldName: s, groupNames: l, index: n })];
      } }, { key: "pre-input-field", fn: function({ fieldName: s, groupNames: l }) {
        return [t._t("pre-input-field", null, { fieldName: s, groupNames: l, index: n })];
      } }, { key: "input-field-addition-before", fn: function({ fieldName: s, groupNames: l }) {
        return [t._t("input-field-addition-before", null, { fieldName: s, groupNames: l, index: n })];
      } }, { key: "input-field-inline-before", fn: function({ fieldName: s, groupNames: l }) {
        return [i["x-attrs"] && i["x-attrs"].text_before ? r("span", { staticClass: "base-form-field__text-before" }, [t._v(" " + t._s(i["x-attrs"].text_before) + " ")]) : t._e(), t._t("input-field-inline-before", null, { fieldName: s, groupNames: l, index: n })];
      } }, { key: "input-field-addition-after", fn: function({ fieldName: s, groupNames: l }) {
        return [i["x-attrs"] && i["x-attrs"].text_after ? r("span", { staticClass: "base-form-field__text-after" }, [t._v(" " + t._s(i["x-attrs"].text_after) + " ")]) : t._e(), t._t("input-field-addition-after", null, { fieldName: s, groupNames: l, index: n })];
      } }, { key: "post-input-field", fn: function({ fieldName: s, groupNames: l }) {
        return [t._t("post-input-field", null, { fieldName: s, groupNames: l, index: n })];
      } }, { key: "error-icon", fn: function() {
        return [t._t("error-icon")];
      }, proxy: !0 }, { key: "remove-icon", fn: function() {
        return [t._t("remove-icon")];
      }, proxy: !0 }, { key: "below-input", fn: function({ fieldName: s, groupNames: l }) {
        return [t._t("below-input", null, { fieldName: s, groupNames: l, index: n })];
      } }], null, !0) }, "BaseFormFieldCreator", t.formFieldComponentProps(i, o, n), !1)), !t.multiplyButtonsInline(i) && (t.checkFieldContent(t.valueList[i.name]) || t.valueListInt[i.name].length > 1) ? r("div", { key: `${o}_button_${n}_${t.formId}`, staticClass: "group-add" }, [r("button", { staticClass: "field-group-button", attrs: { type: "button" }, on: { click: function(s) {
        return s.preventDefault(), t.removeField(i, n);
      } } }, [r("span", [t._v(" " + t._s(t.valueListInt[i.name].length === 1 ? t.getI18nTerm("form.clearField") || "Clear" : t.getI18nTerm("form.removeField", -1, { fieldType: t.getFieldName(i) })) + " ")]), r("span", [r("base-icon", { staticClass: "field-group-icon", attrs: { name: "remove" } })], 1)])]) : t._e(), t.multiplyButtonsInline(i) ? [r("div", { staticClass: "base-form__inline-icons" }, [t.checkFieldContent(t.valueList[i.name]) || t.valueListInt[i.name].length > 1 ? r("base-icon", { staticClass: "base-form__inline-icon", attrs: { title: t.valueListInt[i.name].length === 1 ? t.getI18nTerm("form.clearField") || "Clear" : t.getI18nTerm("form.removeField", -1, { fieldType: t.getFieldName(i) }), role: "button", tabindex: "0", name: "remove" }, nativeOn: { click: function(s) {
        return t.removeField(i, n);
      }, keydown: function(s) {
        return !s.type.indexOf("key") && t._k(s.keyCode, "enter", 13, s.key, "Enter") ? null : t.removeField(i, n);
      } } }) : t._e(), r("base-icon", { staticClass: "base-form__inline-icon", attrs: { title: n !== t.valueListInt[i.name].length - 1 ? t.getI18nTerm("form.addGroupBelow", -1, {
        fieldType: t.getFieldName(i)
      }) : t.getI18nTerm("form.addGroup", -1, {
        fieldType: t.getFieldName(i)
      }), role: "button", tabindex: "0", name: "plus" }, nativeOn: { click: function(s) {
        return t.multiplyField(i, n);
      }, keydown: function(s) {
        return !s.type.indexOf("key") && t._k(s.keyCode, "enter", 13, s.key, "Enter") ? null : t.multiplyField(i, n);
      } } })], 1)] : t._e()], 2);
    }), t.multiplyButtonsInline(i) ? t._e() : r("div", { key: "multiplyButton" + o, staticClass: "group-multiply" }, [r("button", { staticClass: "field-group-button", attrs: { type: "button" }, on: { click: function(a) {
      return a.preventDefault(), t.multiplyField(i);
    } } }, [r("span", [t._v(" " + t._s(t.getI18nTerm("form.addGroup", -1, { fieldType: t.getFieldName(i) })) + " ")]), r("span", [r("base-icon", { staticClass: "field-group-icon", attrs: { name: "plus" } })], 1)])])] : r("BaseFormFieldCreator", t._b({ key: `${i.name}_${o}_${t.formId}`, class: [
      "base-form-field",
      t.formFieldsHalf.indexOf(i) >= 0 ? "base-form-field-half" : "base-form-field-full",
      { "base-form-field-top-margin": i.type === "boolean" && t.formFieldsHalf.indexOf(i) >= 0 },
      { "base-form-field-left-margin": t.isHalfFieldSecond(i) }
    ], on: { "field-value-changed": function(a) {
      return t.setFieldValue(a, i.name);
    }, "fetch-autocomplete": t.fetchAutocomplete }, scopedSlots: t._u([{ key: "label-addition", fn: function({ fieldName: a, groupNames: n }) {
      return [t._t("label-addition", null, { fieldName: a, groupNames: n })];
    } }, { key: "pre-input-field", fn: function({ fieldName: a, groupNames: n }) {
      return [t._t("pre-input-field", null, { fieldName: a, groupNames: n })];
    } }, { key: "input-field-addition-before", fn: function({ fieldName: a, groupNames: n }) {
      return [t._t("input-field-addition-before", null, { fieldName: a, groupNames: n })];
    } }, { key: "input-field-inline-before", fn: function({ fieldName: a, groupNames: n }) {
      return [t._t("input-field-inline-before", null, { fieldName: a, groupNames: n }), i["x-attrs"] && i["x-attrs"].text_before ? r("span", { staticClass: "base-form-field__text-before" }, [t._v(" " + t._s(i["x-attrs"].text_before) + " ")]) : t._e()];
    } }, { key: "input-field-addition-after", fn: function({ fieldName: a, groupNames: n }) {
      return [i["x-attrs"] && i["x-attrs"].text_after ? r("span", { staticClass: "base-form-field__text-after" }, [t._v(" " + t._s(i["x-attrs"].text_after) + " ")]) : t._e(), t._t("input-field-addition-after", null, { fieldName: a, groupNames: n })];
    } }, { key: "post-input-field", fn: function({ fieldName: a, groupNames: n }) {
      return [t._t("post-input-field", null, { fieldName: a, groupNames: n })];
    } }, { key: "error-icon", fn: function() {
      return [t._t("error-icon")];
    }, proxy: !0 }, { key: "remove-icon", fn: function() {
      return [t._t("remove-icon")];
    }, proxy: !0 }, { key: "below-input", fn: function({ fieldName: a, groupNames: n }) {
      return [t._t("below-input", null, { fieldName: a, groupNames: n })];
    } }], null, !0) }, "BaseFormFieldCreator", t.formFieldComponentProps(i, o), !1))];
  })], 2)], 1);
}, b = [], F = /* @__PURE__ */ h(
  g,
  y,
  b,
  !1,
  null,
  "fff90047",
  null,
  null
);
const O = F.exports;
export {
  O as default
};
