// base-ui-components@3.1.2/dist/components/BaseChipsInput/BaseChipsInput.es.js
import o from "../BaseIcon/BaseIcon.es.js";
import s from "../BaseChipsInputField/BaseChipsInputField.es.js";
import l from "../../mixins/i18n.es.js";
import r from "../../mixins/navigateList.es.js";
import "./BaseChipsInput.vue_vue_type_style_index_0_scoped_90d3c286_lang.es.js";
import a from "../../_virtual/_plugin-vue2_normalizer.es.js";
const p = {
  name: "BaseChipsInput",
  components: {
    BaseIcon: o,
    BaseDropDownList: () => import("../BaseDropDownList/BaseDropDownList.es.js").then((e) => e.default || e),
    BaseChipsInputField: s
  },
  mixins: [
    l,
    r
  ],
  model: {
    prop: "selectedList",
    event: "selected-changed"
  },
  props: {
    list: {
      type: Array,
      default: () => []
    },
    selectedList: {
      type: Array,
      default: () => []
    },
    label: {
      type: String,
      required: !0
    },
    showLabel: {
      type: Boolean,
      default: !0
    },
    placeholder: {
      type: String,
      default: null
    },
    dropDownNoOptionsInfo: {
      type: String,
      default: "No options available"
    },
    allowUnknownEntries: {
      type: Boolean,
      default: !1
    },
    allowMultipleEntries: {
      type: Boolean,
      default: !0
    },
    allowDynamicDropDownEntries: {
      type: Boolean,
      default: !1
    },
    addSelectedEntryDirectly: {
      type: Boolean,
      default: !0
    },
    showInputBorder: {
      type: Boolean,
      default: !0
    },
    sortable: {
      type: Boolean,
      default: !1
    },
    draggable: {
      type: Boolean,
      default: !1
    },
    alwaysLinked: {
      type: Boolean,
      default: !1
    },
    hoverboxContent: {
      type: Object,
      default: () => ({})
    },
    isLoading: {
      type: Boolean,
      default: !1
    },
    sortText: {
      type: String,
      default: "Sort A \u2013 Z"
    },
    sortName: {
      type: Boolean,
      default: !1
    },
    language: {
      type: String,
      default: ""
    },
    addNewChipText: {
      type: String,
      default: ""
    },
    id: {
      type: String,
      default: ""
    },
    identifierPropertyName: {
      type: String,
      default: ""
    },
    labelPropertyName: {
      type: String,
      default: ""
    },
    displayChipsInline: {
      type: Boolean,
      default: !0
    },
    required: {
      type: Boolean,
      default: !1
    },
    invalid: {
      type: Boolean,
      default: !1
    },
    disabled: {
      type: Boolean,
      default: !1
    },
    errorMessage: {
      type: String,
      default: ""
    },
    showErrorIcon: {
      type: Boolean,
      default: !1
    },
    clearable: {
      type: Boolean,
      default: !1
    },
    inputClass: {
      type: String,
      default: ""
    },
    chipsEditable: {
      type: Boolean,
      default: !1
    },
    assistiveText: {
      type: Object,
      default: () => ({})
    }
  },
  data() {
    return {
      input: "",
      selectedListInt: [],
      activeOptionIndex: 0,
      selectedOption: null,
      isActive: !1,
      dropDownActive: !1,
      returnAsString: !1,
      dropDownMinWidth: "100%",
      inputElem: null
    };
  },
  computed: {
    chipsFieldInputProps() {
      const e = { ...this.$props };
      return delete e.dropDownNoOptionsInfo, delete e.allowDynamicDropDownEntries, delete e.addNewChipText, delete e.list, e;
    },
    listInt() {
      let e = [];
      return this.list && this.list.length && typeof this.list[0] == "string" ? e = this.list.map((t) => ({
        [this.labelPropertyName]: t
      })) : e = [...this.list], this.allowUnknownEntries && this.input && e.unshift({
        [this.labelPropertyName]: this.language ? { [this.language]: this.input } : this.input
      }), this.selectedListInt && this.selectedListInt.length && (e = e.filter((t) => !this.selectedListInt.map((n) => n[this.identifierPropertyName] || n[this.labelPropertyName]).includes(t[this.identifierPropertyName] || t[this.labelPropertyName]))), this.input && !this.allowDynamicDropDownEntries ? e.filter((t) => this.getLangLabel(t[this.labelPropertyName]).toLowerCase().includes(this.input.toLowerCase())) : e;
    },
    chipsInputActive: {
      set(e) {
        this.dropDownActive = e, this.isActive = e;
      },
      get() {
        return this.isActive || this.dropDownActive;
      }
    },
    activeOption: {
      set(e) {
        this.activeOptionIndex = this.listInt.indexOf(e);
      },
      get() {
        return this.listInt[this.activeOptionIndex];
      }
    }
  },
  watch: {
    selectedOption(e) {
      e && (this.addSelectedOption(e), this.inputElem && this.allowMultipleEntries && this.inputElem.focus());
    },
    listInt: {
      handler(e) {
        e.length && !this.activeOption ? this.activeOptionIndex = 0 : !e.length && (!this.allowUnknownEntries || !this.input) && (this.activeOptionIndex = -1);
      },
      immediate: !0
    },
    list: {
      handler(e) {
        this.activeOptionIndex = e.length || this.allowUnknownEntries && this.input ? 0 : -1, !this.returnAsString && e && e.length && typeof e[0] == "string" && (this.returnAsString = !0);
      },
      immediate: !0
    },
    selectedList: {
      handler(e) {
        this.selectedListInt = [...e];
      },
      immediate: !0
    },
    selectedListInt(e) {
      JSON.stringify(e) !== JSON.stringify(this.selectedList) && this.updateParentSelectedList(e);
    },
    input(e) {
      this.allowDynamicDropDownEntries && this.$emit("fetch-dropdown-entries", { value: e, type: this.labelPropertyName });
    },
    chipsInputActive(e) {
      e && !this.allowDynamicDropDownEntries && !this.listInt.length && this.$emit("fetch-dropdown-entries", { value: this.input, type: this.labelPropertyName }), e && (this.calcDropDownMinWidth(), this.activeOptionIndex = 0);
    }
  },
  mounted() {
    const e = this.$el.getElementsByTagName("input");
    e && e.length && ([this.inputElem] = e);
  },
  methods: {
    addSelectedOption(e) {
      this.allowMultipleEntries ? this.selectedListInt.push(e) : (this.$set(this.selectedListInt, 0, e), this.dropDownActive = !1), this.updateParentSelectedList(this.selectedListInt), this.input = "", this.selectedOption = null, this.allowMultipleEntries || (this.chipsInputActive = !1, this.inputElem.blur());
    },
    updateParentSelectedList(e) {
      if (JSON.stringify(this.selectedList) !== JSON.stringify(e)) {
        let t = [...e];
        this.returnAsString && (t = t.map((n) => n[this.labelPropertyName])), this.$emit("selected-changed", t);
      }
    },
    onInputFocus() {
      this.chipsInputActive = !0;
    },
    onInputBlur() {
      this.chipsInputActive = !1, !this.dropDownActive && document.activeElement.tagName === "BODY" && (this.input = "");
    },
    checkKeyEvent(e) {
      e.key === "Tab" && (this.chipsInputActive = !1);
    },
    onEnter() {
      this.activeOption && this.addSelectedOption(this.activeOption);
    },
    onArrowKey(e) {
      this.listInt.length && (this.activeOption = this.navigate(
        this.listInt,
        e.key === "ArrowDown",
        this.activeOptionIndex,
        !0
      ));
    },
    hoverBoxActive(e, t) {
      this.$emit("hoverbox-active", { value: e, entry: t });
    },
    calcDropDownMinWidth() {
      const e = this.$refs.baseInput;
      e && e.$el && e.$el.clientWidth && (this.dropDownMinWidth = `${e.$el.clientWidth}px`);
    }
  }
};
var d = function() {
  var t = this, n = t._self._c;
  return n("div", { staticClass: "base-chips-input" }, [n("BaseChipsInputField", t._g(t._b({ ref: "baseInput", attrs: { "add-selected-entry-directly": !1, "selected-list": t.selectedListInt, "drop-down-list-id": t.id, "linked-list-option": t.activeOption ? t.activeOption[t.identifierPropertyName] : null, "return-as-string": t.returnAsString, "is-active": t.isActive, loadable: t.allowDynamicDropDownEntries }, on: { "update:selectedList": function(i) {
    t.selectedListInt = i;
  }, "update:selected-list": function(i) {
    t.selectedListInt = i;
  }, "update:isActive": function(i) {
    t.isActive = i;
  }, "update:is-active": function(i) {
    t.isActive = i;
  }, keydown: [function(i) {
    return !i.type.indexOf("key") && t._k(i.keyCode, "enter", 13, i.key, "Enter") ? null : (i.preventDefault(), t.onEnter.apply(null, arguments));
  }, function(i) {
    return !i.type.indexOf("key") && t._k(i.keyCode, "up", 38, i.key, ["Up", "ArrowUp"]) && t._k(i.keyCode, "down", 40, i.key, ["Down", "ArrowDown"]) ? null : (i.preventDefault(), t.onArrowKey.apply(null, arguments));
  }], "click-input-field": t.onInputFocus, "clicked-outside": t.onInputBlur }, scopedSlots: t._u([{ key: "below-input", fn: function() {
    return [t.isActive ? n("BaseDropDownList", { ref: "dropDownList", staticClass: "base-chips-input__drop-down", style: { "min-width": t.dropDownMinWidth }, attrs: { "drop-down-options": t.listInt, "active-option": t.activeOption, "selected-option": t.selectedOption, "identifier-property-name": t.identifierPropertyName, "label-property-name": t.labelPropertyName, "list-id": t.id, language: t.language, "drop-down-no-options-info": t.dropDownNoOptionsInfo }, on: { "update:activeOption": function(i) {
      t.activeOption = i;
    }, "update:active-option": function(i) {
      t.activeOption = i;
    }, "update:selectedOption": function(i) {
      t.selectedOption = i;
    }, "update:selected-option": function(i) {
      t.selectedOption = i;
    }, "within-drop-down": function(i) {
      t.dropDownActive = i;
    } }, nativeOn: { click: function(i) {
      i.stopPropagation();
    }, touchstart: function(i) {
      i.stopPropagation();
    } }, scopedSlots: t._u([{ key: "option", fn: function(i) {
      return [t.allowUnknownEntries && !i.option[t.identifierPropertyName] ? n("span", { key: i.option[t.identifierPropertyName], ref: "option" }, [t._v(" " + t._s(t.addNewChipText ? `${t.addNewChipText} ${t.getLangLabel(i.option[t.labelPropertyName], !0)} ...` : `${t.getI18nTerm("form.Add", -1, { value: t.getLangLabel(i.option[t.labelPropertyName], !0) })} ...`) + " ")]) : i ? [t._t("drop-down-entry", function() {
        return [t._v(" " + t._s(t.getLangLabel(i.option[t.labelPropertyName], !0)) + " ")];
      }, { item: i.option })] : t._e()];
    } }, { key: "no-options", fn: function() {
      return [t._t("no-options")];
    }, proxy: !0 }], null, !0) }) : t._e()];
  }, proxy: !0 }, { key: "label-addition", fn: function() {
    return [t._t("label-addition")];
  }, proxy: !0 }, { key: "pre-input-field", fn: function() {
    return [t._t("pre-input-field")];
  }, proxy: !0 }, { key: "input-field-addition-before", fn: function() {
    return [t._t("input-field-addition-before")];
  }, proxy: !0 }, { key: "input-field-inline-before", fn: function() {
    return [t._t("input-field-inline-before")];
  }, proxy: !0 }, { key: "input-field-addition-after", fn: function() {
    return [t._t("input-field-addition-after")];
  }, proxy: !0 }, { key: "post-input-field", fn: function() {
    return [t._t("post-input-field"), t.allowMultipleEntries ? t._e() : n("div", { staticClass: "base-chips-input__single-dropdown", on: { keydown: function(i) {
      if (!i.type.indexOf("key") && t._k(i.keyCode, "enter", 13, i.key, "Enter"))
        return null;
      i.stopPropagation(), t.isActive = !t.isActive;
    }, click: function(i) {
      i.stopPropagation(), t.isActive = !t.isActive;
    } } }, [n("BaseIcon", { class: [
      "base-chips-input__single-dropdown-icon",
      {
        "base-chips-input__single-dropdown-icon-rotated": t.isActive
      }
    ], attrs: { name: "drop-down" } })], 1)];
  }, proxy: !0 }, { key: "error-icon", fn: function() {
    return [t._t("error-icon")];
  }, proxy: !0 }, { key: "remove-icon", fn: function() {
    return [t._t("remove-icon")];
  }, proxy: !0 }], null, !0), model: { value: t.input, callback: function(i) {
    t.input = i;
  }, expression: "input" } }, "BaseChipsInputField", t.chipsFieldInputProps, !1), t.$listeners))], 1);
}, u = [], c = /* @__PURE__ */ a(
  p,
  d,
  u,
  !1,
  null,
  "90d3c286",
  null,
  null
);
const v = c.exports;
export {
  v as default
};
