// base-ui-components@3.1.2/dist/mixins/i18n.es.js
const a = {
  computed: {
    hasI18n: function() {
      return Boolean(this.$i18n);
    }
  },
  methods: {
    getI18nTerm: function(n) {
      var e = arguments.length > 1 && arguments[1] !== void 0 ? arguments[1] : -1, t = arguments.length > 2 && arguments[2] !== void 0 ? arguments[2] : {};
      if (!n)
        return "";
      if (this.hasI18n && this.$te(n))
        return e >= 0 ? this.$tc(n, e, t) : this.$t(n, t);
      var r = n.split(".");
      return r.pop() || n;
    },
    setLangLabels: function(n, e) {
      var t = this;
      return e.reduce(function(r, s) {
        return t.$set(r, s, t.$i18n ? t.$t(n, s) : n.split(".").pop()), r;
      }, {});
    },
    getLangLabel: function(n) {
      var e = arguments.length > 1 && arguments[1] !== void 0 ? arguments[1] : !1, t = this.hasI18n ? this.language || this.$i18n.locale : this.language;
      if (typeof n == "string")
        return n;
      if (n && t && n[t])
        return n[t];
      if (n && t && e) {
        var r = Object.keys(n).find(function(s) {
          return !!n[s];
        });
        return n[r] || n[t] || "";
      }
      return n;
    }
  }
};
export {
  a as default
};
